---
title: "Homology analysis"
author: "Nils Eling"
date: "9/14/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This script reproduces the homology analysis between the different chemokines.
We downloaded the data from www.ncbi.nlm.nih.gov and saved the transcript 
sequences. https://www.ebi.ac.uk/Tools/msa/clustalo/ was used to align all-vs-all
transcripts using the following call:

```
$APPBIN/clustal-omega-1.2.4/bin/clustalo --infile clustalo-E20210914-122047-0397-8283578-p2m.sequence --threads 8 --MAC-RAM 8000 --verbose --guidetree-out clustalo-E20210914-122047-0397-8283578-p2m.dnd --outfmt clustal --resno --outfile clustalo-E20210914-122047-0397-8283578-p2m.clustal_num --output-order tree-order --seqtype dna
```

We will first look at the identity matrix, inidcating the percentage of
sequence similarity.

```{r read-in-data, message=FALSE, warning=FALSE}
library(tidyverse)
library(SingleCellExperiment)

seq_similarity <- read.table("~/switchdrive/Institution/MelanomaMC/ClustalW_results/identity_matrix.txt")
rownames(seq_similarity) <- seq_similarity$V2
seq_similarity <- seq_similarity[,-c(1,2)]

# Remove non coding transcript variants
seq_similarity <- seq_similarity[!grepl("XR_|NR_", rownames(seq_similarity)),
                                 !grepl("XR_|NR_", rownames(seq_similarity))]
rownames(seq_similarity) <- sub("\\.[0-9]*", "", rownames(seq_similarity))

library(ape)
phylogenetic_tree <- read.tree("~/switchdrive/Institution/MelanomaMC/ClustalW_results/phylogenetic_tree.txt")
plot(phylogenetic_tree)
```

Now, we will map between RefSeq transcript ids, ensemble transcript ids and
gene names.

```{r biomart}
library(biomaRt)
ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
cur_tab <- getBM(attributes=c("refseq_mrna", "ensembl_gene_id", "hgnc_symbol"), 
      filters = "refseq_mrna", values = rownames(seq_similarity), 
      mart = ensembl, uniqueRows = FALSE)
cur_tab <- cur_tab[match(rownames(seq_similarity), cur_tab$refseq_mrna),]

rownames(seq_similarity)  <- colnames(seq_similarity) <- 
    paste0(cur_tab$refseq_mrna, "_", cur_tab$hgnc_symbol)
```

We now visualize the similarity as heatmap.

```{r viz-similarity-heatmap, fig.height=7, fig.width=7}
library(pheatmap)
library(viridis)
pheatmap(seq_similarity, cluster_cols = FALSE,
         cluster_rows = FALSE, color = inferno(100), breaks = 0:100)
```

And we compare it to correlation in expression across all cells.

```{r cor-expression, fig.height=5, fig.width=5}
final_sce <- readRDS("~/Desktop/REDSEA_test/sce_RNA.rds")
# select only the cells that express chemokines
for_analysis <- final_sce[,final_sce$expressor != "NA"]

cur_cor <- cor(t(assay(for_analysis, "asinh")[c("T5_CCL4", "T7_CCL18", "T1_CXCL8",
                                             "T4_CXCL10", "T3_CXCL12", "T8_CXCL13",
                                             "T12_CCL2", "T2_CCL22",
                                             "T9_CXCL9", "T11_CCL8", "T10_CCL19"),]), 
                                           method = "spearman")

pheatmap(cur_cor, color = colorRampPalette(c("dark blue", "white", "dark red"))(100), 
         breaks = seq(-1, 1, length.out = 100))
```

```{r viz-correlation}
cor_tibbble <- cur_cor %>%
    as_tibble() %>% 
    mutate(probe = rownames(cur_cor)) %>% 
    pivot_longer(cols = 1:ncol(cur_cor)) %>%
    mutate(probe = str_split(probe, pattern = "_", simplify = TRUE)[,2],
           name = str_split(name, pattern = "_", simplify = TRUE)[,2]) %>%
    arrange(probe, name) %>%
    filter(probe != name)

sim_tibble <- seq_similarity %>%
    as_tibble() %>% 
    mutate(probe = rownames(seq_similarity)) %>% 
    pivot_longer(cols = 1:ncol(seq_similarity)) %>%
    mutate(from_chemo = str_split(probe, pattern = "_", simplify = TRUE)[,3],
           to_chemo = str_split(name, pattern = "_", simplify = TRUE)[,3]) %>%
    group_by(from_chemo, to_chemo) %>%
    summarize(mean_similarity = mean(value, na.rm=TRUE)) %>%
    arrange(from_chemo, to_chemo) %>%
    filter(from_chemo != to_chemo)

all.equal(paste(cor_tibbble$probe, cor_tibbble$name),
          paste(sim_tibble$from_chemo, sim_tibble$to_chemo))

ggplot(data.frame(similarity = sim_tibble$mean_similarity,
                  correlation = cor_tibbble$value)) +
    geom_point(aes(similarity, correlation))

cor.test(sim_tibble$mean_similarity,
         cor_tibbble$value)
```

## Number of co-expressors

Finally, we will compare the sequence similarity to the jaccard index 
of chemokine expressors.

```{r}
# Define the jaccard similarity
jaccard <- function(x,y){
    intersection <- length(intersect(x,y))
    union = length(x) + length(y) - intersection
    return (intersection/union)
}

# We will pass the unique cell ids into this function
cur_out <- lapply(seq_len(nrow(sim_tibble)),
       function(x){
           from_chemo_cells <- colnames(final_sce)[colData(final_sce)[[sim_tibble$from_chemo[x]]] != 0]
           to_chemo_cells <- colnames(final_sce)[colData(final_sce)[[sim_tibble$to_chemo[x]]] != 0]
           return(jaccard(from_chemo_cells, to_chemo_cells))
       })
sim_tibble$jaccard_sim <- unlist(cur_out)

ggplot(data.frame(similarity = sim_tibble$mean_similarity,
                  jaccard_sim = sim_tibble$jaccard_sim)) +
    geom_point(aes(similarity, jaccard_sim))

cor.test(sim_tibble$mean_similarity,
         sim_tibble$jaccard_sim)
```

