<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Nils Eling/Daniel Schulz/Tobias Hoch" />

<meta name="date" content="2020-07-28" />

<title>03_RNA_quality_control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Score.html">Bcell Score</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_1.html">Supp-Figure 1</a>
    </li>
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="XX_hazard_ratio.html">Hazard Ratio Analysis</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">03_RNA_quality_control</h1>
<h4 class="author">Nils Eling/Daniel Schulz/Tobias Hoch</h4>
<h4 class="date">2020-07-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>melanoma_publication_old_data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtreeee1595de9ec17ee414a5beafb9ed08d174e12139targetblankee1595da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/ee1595de9ec17ee414a5beafb9ed08d174e12139" target="_blank">ee1595d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtreeee1595de9ec17ee414a5beafb9ed08d174e12139targetblankee1595da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/ee1595de9ec17ee414a5beafb9ed08d174e12139" target="_blank">ee1595d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/._clinical metadata preparation.Rmd
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._protein_neutrophil.png
    Ignored:    output/._rna_neutrophil.png
    Ignored:    output/PSOCKclusterOut/
    Ignored:    output/bcell_grouping.png
    Ignored:    output/dysfunction_correlation.pdf

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/04_1_Protein_celltype_classification.rmd
    Modified:   analysis/04_1_RNA_celltype_classification.rmd
    Modified:   analysis/04_2_RNA_classification_subclustering.rmd
    Modified:   analysis/04_2_protein_classification_subclustering.rmd
    Modified:   analysis/07_TCF7_PD1_gating.rmd
    Modified:   analysis/08_color_vectors.rmd
    Modified:   analysis/09_Tcell_Score.Rmd
    Modified:   analysis/10_Dysfunction_Score.rmd
    Modified:   analysis/11_Bcell_Score.Rmd
    Modified:   analysis/Figure_1.rmd
    Modified:   analysis/Figure_2.rmd
    Modified:   analysis/Figure_3.rmd
    Modified:   analysis/Figure_4.rmd
    Modified:   analysis/Figure_5.rmd
    Modified:   analysis/Supp-Figure_1.rmd
    Modified:   analysis/Supp-Figure_2.rmd
    Modified:   analysis/Supp-Figure_3.rmd
    Modified:   analysis/Supp-Figure_4.rmd
    Modified:   analysis/Supp-Figure_5.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03_RNA_quality_control.rmd</code>) and HTML (<code>docs/03_RNA_quality_control.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/ee1595de9ec17ee414a5beafb9ed08d174e12139/analysis/03_RNA_quality_control.rmd" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/ee1595de9ec17ee414a5beafb9ed08d174e12139/docs/03_RNA_quality_control.html" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/2e443a5cda5affb9de9ca1eb678438325319a857/analysis/03_RNA_quality_control.rmd" target="_blank">2e443a5</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
remove files that are not needed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/3f5af3f59358eebb33aa2a8a06fddca34dad6a6b/docs/03_RNA_quality_control.html" target="_blank">3f5af3f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
add .html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/f9bb33a83d519d86072a7e716214ff78bce022a0/analysis/03_RNA_quality_control.rmd" target="_blank">f9bb33a</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-04
</td>
<td>
new Figure 5 and minor changes in figure order
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/2ac183310c0cc9a7a420897f1b99752b1aae10a7/analysis/03_RNA_quality_control.rmd" target="_blank">2ac1833</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-01-08
</td>
<td>
changes to Figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/d8819f2b618d0620606a5e5b181d3b5340469763/analysis/03_RNA_quality_control.rmd" target="_blank">d8819f2</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-08
</td>
<td>
read new data (nuclei expansion) and adapt scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/fb0f7cb1d0385419cfd271f1d5bb83f595dae0f7/analysis/03_RNA_quality_control.rmd" target="_blank">fb0f7cb</a>
</td>
<td>
SchulzDan
</td>
<td>
2020-08-24
</td>
<td>
more paths adapted
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/2c11d5cd8e42b3be31d1c1f512864b531f93fd7c/analysis/03_RNA_quality_control.rmd" target="_blank">2c11d5c</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
add new scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/e9cc8b2c0d594de3655751cc57ee9b3f859a3fce/analysis/03_RNA_quality_control.rmd" target="_blank">e9cc8b2</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-07-30
</td>
<td>
rename files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions/calculateSummary.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/censor_dat.R
value   ?                                 
visible FALSE                             
        code/helper_functions/detect_mRNA_expression.R
value   ?                                             
visible FALSE                                         
        code/helper_functions/DistanceToClusterCenter.R
value   ?                                              
visible FALSE                                          
        code/helper_functions/findMilieu.R code/helper_functions/findPatch.R
value   ?                                  ?                                
visible FALSE                              FALSE                            
        code/helper_functions/getInfoFromString.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/getSpotnumber.R
value   ?                                    
visible FALSE                                
        code/helper_functions/plotCellCounts.R
value   ?                                     
visible FALSE                                 
        code/helper_functions/plotCellFractions.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/plotDist.R
value   ?                               
visible FALSE                           
        code/helper_functions/scatter_function.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/sceChecks.R
value   ?                                
visible FALSE                            
        code/helper_functions/validityChecks.R
value   ?                                     
visible FALSE                                 </code></pre>
<pre class="r"><code>library(SingleCellExperiment)
library(dplyr)
library(ggplot2)
library(scater)
library(CATALYST)
library(reshape2)
library(viridis)
library(ggridges)
library(cowplot)
library(BiocParallel)
library(dittoSeq)</code></pre>
</div>
<div id="load-the-single-cell-experiment-object-and-the-image-metadata" class="section level2">
<h2>Load the single cell experiment object and the image metadata</h2>
<pre class="r"><code>sce &lt;- readRDS(file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
</div>
</div>
<div id="assay" class="section level1">
<h1>Assay</h1>
<div id="add-different-assays" class="section level2">
<h2>Add different assays</h2>
<pre class="r"><code>assay(sce, &quot;scaled_counts&quot;) &lt;- t(scale(t(assay(sce, &quot;counts&quot;))))
assay(sce, &quot;scaled_asinh&quot;) &lt;- t(scale(t(assay(sce, &quot;asinh&quot;))))</code></pre>
</div>
</div>
<div id="general-plots" class="section level1">
<h1>General Plots</h1>
<div id="plot-cell-counts-for-every-image" class="section level2">
<h2>Plot Cell Counts for every Image</h2>
<pre class="r"><code># this function takes all the column metadata from the sce and plots parts thereof
plotCellCounts(sce, colour_by = &quot;Location&quot;, split_by = &quot;ImageNumber&quot;, imageID = &quot;ImageNumber&quot;)</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/cell_counts_over_samples-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="image-57-has-only-a-few-cells-and-should-probably-be-excluded." class="section level2">
<h2>Image 57 has only a few cells and should probably be excluded.</h2>
<p>will be flagged below</p>
<pre class="r"><code>cur_sce &lt;- data.frame(colData(sce))
# show images with less than 500 cells
cur_sce %&gt;%
  group_by(ImageNumber) %&gt;%
  summarise(n=n()) %&gt;%
  filter(n&lt;500)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code># A tibble: 1 x 2
  ImageNumber     n
        &lt;int&gt; &lt;int&gt;
1          57   136</code></pre>
</div>
<div id="flag-image-in-sce-object-for-future-exclusion" class="section level2">
<h2>Flag image in sce object for future exclusion</h2>
<pre class="r"><code># define vector for each single cell whether to keep (TRUE) or not (FALSE)
includeImage &lt;- colData(sce)$ImageNumber != 57
sce$includeImage &lt;- includeImage</code></pre>
</div>
<div id="mean-intensity-of-markers-per-image" class="section level2">
<h2>Mean intensity of markers per image</h2>
<pre class="r"><code># we use a function from Nils. This function makes use of the aggregate function to calculate the mean for each channel over all specified groups
mean_sce &lt;- calculateSummary(sce, split_by = c(&quot;ImageNumber&quot;, &quot;BlockID&quot;, &quot;Location&quot;,&quot;Mutation&quot;,&quot;Cancer_Stage&quot;, &quot;Status_at_3m&quot;,&quot;E_I_D&quot;,&quot;Adjuvant&quot;), exprs_values = &quot;counts&quot;)</code></pre>
</div>
<div id="transform-data" class="section level2">
<h2>Transform data</h2>
<pre class="r"><code>assay(mean_sce, &quot;asinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;))
assay(mean_sce, &quot;asinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))</code></pre>
</div>
<div id="plot-the-mean-data" class="section level2">
<h2>Plot the mean data</h2>
<pre class="r"><code># first we define a vector of markers that we want to plot
plot_targets &lt;- rownames(sce)
plot_targets &lt;- plot_targets[! plot_targets %in% c(&quot;DNA1&quot;,&quot;DNA2&quot;,&quot;HistoneH3&quot;)]

# now we plot the heatmap
plotHeatmap(mean_sce,features = plot_targets  ,exprs_values = &quot;asinh&quot;,colour_columns_by = &quot;ImageNumber&quot;,color = viridis(100))</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/mean_per_image-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-the-scaled-data" class="section level2">
<h2>Plot the scaled data</h2>
<pre class="r"><code># now we plot the scaled heatmap
plotHeatmap(mean_sce,features = plot_targets, exprs_values = &quot;asinh_scaled&quot;, colour_columns_by = c(&quot;ImageNumber&quot;), zlim = c(-3,3),
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100))</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/mean_per_image_scaled-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-level-quality-control" class="section level2">
<h2>Cell level quality control</h2>
<p>here we plot the marker intensity distributions for all images. since we have too many images we make groups of 10.</p>
<pre class="r"><code>y &lt;- c(rep(1:10,16),rep(11,7))

# add the group information to the sce object
sce$groups &lt;- y[colData(sce)$ImageNumber]

# now we use the function written by Nils
plotDist(sce, plot_type = &quot;ridges&quot;, 
         colour_by = &quot;groups&quot;, split_by = &quot;rows&quot;, 
         exprs_values = &quot;asinh&quot;) + 
  theme_minimal(base_size = 15)</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/density_plots_all_markers-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code> # the distributions look very even across images indicating that we have no major batch effects.</code></pre>
</div>
<div id="define-markers-which-had-poor-staining" class="section level2">
<h2>Define markers which had poor staining</h2>
<p>By visual inspection, we defined bad markers</p>
<pre class="r"><code>rowData(sce)$good_marker &lt;- ! grepl(&quot;DNA|Histone|Vimentin|PD1|PDL1|CCR2|CK5&quot;,rownames(sce))

# good_marker without RNA channels
rowData(sce)$noRNA_marker &lt;- ! grepl(&quot;DNA|Histone|Vimentin|PD1|PDL1|CCR2|CK5|CXCL|CCL|DapB&quot;,rownames(sce))</code></pre>
</div>
<div id="calculate-umap" class="section level2">
<h2>Calculate UMAP</h2>
<pre class="r"><code>set.seed(12345)

# UMAP
start = Sys.time()
sce &lt;- runUMAP(sce, exprs_values = &quot;scaled_counts&quot;, 
                     subset_row = rowData(sce)$good_marker)
end = Sys.time()
print(end-start)</code></pre>
<pre><code>Time difference of 8.003067 mins</code></pre>
</div>
<div id="subset-sce-for-umap-visualization" class="section level2">
<h2>Subset SCE for UMAP visualization</h2>
<pre class="r"><code>cur_sce &lt;- sce[, colnames(sce) %in% sample(sce$cellID, round(length(sce$cellID)*0.05))]
cur_sce$ImageNumber &lt;- as.character(cur_sce$ImageNumber)</code></pre>
</div>
<div id="visualize-features-on-and-umap" class="section level2">
<h2>Visualize features on and UMAP</h2>
<p>Next, we will visualize different quality features on these representations.</p>
</div>
<div id="umap" class="section level2">
<h2>UMAP</h2>
<pre class="r"><code># Select plots in list
p.list &lt;- list()

# 
p.list$ImageNumber &lt;- dittoDimPlot(cur_sce, var = &quot;ImageNumber&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5, legend.show = FALSE) 
p.list$Mutation &lt;- dittoDimPlot(cur_sce, var = &quot;Mutation&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$Cancer_Stage &lt;- dittoDimPlot(cur_sce, var = &quot;Cancer_Stage&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$relapse &lt;- dittoDimPlot(cur_sce, var = &quot;relapse&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$Location &lt;- dittoDimPlot(cur_sce, var = &quot;Location&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$TissueType &lt;- dittoDimPlot(cur_sce, var = &quot;TissueType&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$MM_location_simplified &lt;- dittoDimPlot(cur_sce, var = &quot;MM_location_simplified&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)
p.list$treatment_group_before_surgery &lt;- dittoDimPlot(cur_sce, var = &quot;treatment_group_before_surgery&quot;, reduction.use = &quot;UMAP&quot;, size = 0.5)


plot_grid(plotlist = p.list, ncol = 4, rel_widths = c(1.5, 1, 1, 1))</code></pre>
<pre><code>Warning: Removed 1183 rows containing missing values (geom_point).

Warning: Removed 1183 rows containing missing values (geom_point).

Warning: Removed 1183 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/UMAP_clinical_features-1.png" width="3360" style="display: block; margin: auto;" /></p>
</div>
<div id="marker-expression-on-umap" class="section level2">
<h2>Marker Expression on UMAP</h2>
<pre class="r"><code>p.list &lt;- list()
for(i in rownames(sce)[rowData(cur_sce)$good_marker]){
  p.list[[i]] &lt;- plotUMAP(cur_sce, colour_by = i, by_exprs_values = &quot;asinh&quot;)
}

plot_grid(plotlist = p.list, ncol = 7)</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/asinh_UMAP_marker-1.png" width="3360" style="display: block; margin: auto;" /></p>
</div>
<div id="scaled-expression-umap" class="section level2">
<h2>Scaled Expression UMAP</h2>
<pre class="r"><code>p.list &lt;- list()
for(i in rownames(sce)[rowData(cur_sce)$good_marker]){
  p.list[[i]] &lt;- plotUMAP(cur_sce, colour_by = i, by_exprs_values = &quot;scaled_asinh&quot;)
}


plot_grid(plotlist = p.list, ncol = 7)</code></pre>
<p><img src="figure/03_RNA_quality_control.rmd/scaled_asinh_UMAP_marker-1.png" width="3360" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-data" class="section level1">
<h1>Save data</h1>
<div id="save-updated-sce-object" class="section level2">
<h2>Save updated SCE object</h2>
<pre class="r"><code>saveRDS(sce, file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] dittoSeq_1.0.2              BiocParallel_1.22.0        
 [3] cowplot_1.1.1               ggridges_0.5.3             
 [5] viridis_0.5.1               viridisLite_0.3.0          
 [7] reshape2_1.4.4              CATALYST_1.12.2            
 [9] scater_1.16.2               ggplot2_3.3.3              
[11] dplyr_1.0.2                 SingleCellExperiment_1.12.0
[13] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[15] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[17] IRanges_2.24.1              S4Vectors_0.28.1           
[19] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[21] matrixStats_0.57.0          workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                circlize_0.4.12            
  [3] drc_3.0-1                   plyr_1.8.6                 
  [5] igraph_1.2.6                ConsensusClusterPlus_1.52.0
  [7] splines_4.0.3               flowCore_2.0.1             
  [9] TH.data_1.0-10              digest_0.6.27              
 [11] htmltools_0.5.0             fansi_0.4.1                
 [13] magrittr_2.0.1              CytoML_2.0.5               
 [15] cluster_2.1.0               limma_3.44.3               
 [17] openxlsx_4.2.3              ComplexHeatmap_2.4.3       
 [19] RcppParallel_5.0.2          sandwich_3.0-0             
 [21] flowWorkspace_4.0.6         cytolib_2.0.3              
 [23] jpeg_0.1-8.1                colorspace_2.0-0           
 [25] ggrepel_0.9.0               haven_2.3.1                
 [27] xfun_0.20                   crayon_1.3.4               
 [29] RCurl_1.98-1.2              jsonlite_1.7.2             
 [31] hexbin_1.28.2               graph_1.66.0               
 [33] survival_3.2-7              zoo_1.8-8                  
 [35] glue_1.4.2                  gtable_0.3.0               
 [37] nnls_1.4                    zlibbioc_1.36.0            
 [39] XVector_0.30.0              GetoptLong_1.0.5           
 [41] DelayedArray_0.16.0         ggcyto_1.16.0              
 [43] car_3.0-10                  BiocSingular_1.4.0         
 [45] Rgraphviz_2.32.0            shape_1.4.5                
 [47] abind_1.4-5                 scales_1.1.1               
 [49] pheatmap_1.0.12             mvtnorm_1.1-1              
 [51] edgeR_3.30.3                Rcpp_1.0.5                 
 [53] plotrix_3.7-8               clue_0.3-58                
 [55] foreign_0.8-81              rsvd_1.0.3                 
 [57] FlowSOM_1.20.0              tsne_0.1-3                 
 [59] RColorBrewer_1.1-2          ellipsis_0.3.1             
 [61] farver_2.0.3                pkgconfig_2.0.3            
 [63] XML_3.99-0.5                uwot_0.1.10                
 [65] utf8_1.1.4                  locfit_1.5-9.4             
 [67] labeling_0.4.2              tidyselect_1.1.0           
 [69] rlang_0.4.10                later_1.1.0.1              
 [71] munsell_0.5.0               cellranger_1.1.0           
 [73] tools_4.0.3                 cli_2.2.0                  
 [75] generics_0.1.0              evaluate_0.14              
 [77] stringr_1.4.0               yaml_2.2.1                 
 [79] knitr_1.30                  fs_1.5.0                   
 [81] zip_2.1.1                   purrr_0.3.4                
 [83] RBGL_1.64.0                 whisker_0.4                
 [85] xml2_1.3.2                  compiler_4.0.3             
 [87] rstudioapi_0.13             beeswarm_0.2.3             
 [89] curl_4.3                    png_0.1-7                  
 [91] tibble_3.0.4                stringi_1.5.3              
 [93] RSpectra_0.16-0             forcats_0.5.0              
 [95] lattice_0.20-41             Matrix_1.3-2               
 [97] vctrs_0.3.6                 pillar_1.4.7               
 [99] lifecycle_0.2.0             GlobalOptions_0.1.2        
[101] RcppAnnoy_0.0.18            BiocNeighbors_1.6.0        
[103] data.table_1.13.6           bitops_1.0-6               
[105] irlba_2.3.3                 httpuv_1.5.4               
[107] R6_2.5.0                    latticeExtra_0.6-29        
[109] promises_1.1.1              gridExtra_2.3              
[111] RProtoBufLib_2.0.0          rio_0.5.16                 
[113] vipor_0.4.5                 codetools_0.2-18           
[115] assertthat_0.2.1            MASS_7.3-53                
[117] gtools_3.8.2                rprojroot_2.0.2            
[119] rjson_0.2.20                withr_2.3.0                
[121] multcomp_1.4-15             GenomeInfoDbData_1.2.4     
[123] hms_0.5.3                   ncdfFlow_2.34.0            
[125] grid_4.0.3                  rmarkdown_2.6              
[127] DelayedMatrixStats_1.10.1   carData_3.0-4              
[129] Rtsne_0.15                  git2r_0.28.0               
[131] base64enc_0.1-3             ggbeeswarm_0.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
