<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Daniel" />


<title>subluclustering and cell type definition</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Score.html">Bcell Score</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_1.html">Supp-Figure 1</a>
    </li>
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="XX_hazard_ratio.html">Hazard Ratio Analysis</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">subluclustering and cell type definition</h1>
<h4 class="author">Daniel</h4>
<h4 class="date">19 12 2019</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>melanoma_publication_old_data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree2e443a5cda5affb9de9ca1eb678438325319a857targetblank2e443a5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/2e443a5cda5affb9de9ca1eb678438325319a857" target="_blank">2e443a5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree2e443a5cda5affb9de9ca1eb678438325319a857targetblank2e443a5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/2e443a5cda5affb9de9ca1eb678438325319a857" target="_blank">2e443a5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/._clinical metadata preparation.Rmd
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._protein_neutrophil.png
    Ignored:    output/._rna_neutrophil.png
    Ignored:    output/PSOCKclusterOut/
    Ignored:    output/bcell_grouping.png
    Ignored:    output/dysfunction_correlation.pdf

Untracked files:
    Untracked:  analysis/00_prepare_clinical_dat.rmd
    Untracked:  code/helper_functions/findMilieu.R
    Untracked:  code/helper_functions/findPatch.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/01_Protein_read_data.rmd
    Modified:   analysis/01_RNA_read_data.rmd
    Modified:   analysis/02_Protein_annotations.rmd
    Modified:   analysis/02_RNA_annotations.rmd
    Modified:   analysis/03_Protein_quality_control.rmd
    Modified:   analysis/03_RNA_quality_control.rmd
    Modified:   analysis/04_1_Protein_celltype_classification.rmd
    Modified:   analysis/04_1_RNA_celltype_classification.rmd
    Modified:   analysis/04_2_RNA_classification_subclustering.rmd
    Modified:   analysis/04_2_protein_classification_subclustering.rmd
    Modified:   analysis/05_RNA_chemokine_expressing_cells.rmd
    Modified:   analysis/06_RNA_chemokine_patch_detection.rmd
    Modified:   analysis/07_TCF7_PD1_gating.rmd
    Modified:   analysis/08_color_vectors.rmd
    Modified:   analysis/09_Tcell_Score.Rmd
    Modified:   analysis/10_Dysfunction_Score.rmd
    Modified:   analysis/11_Bcell_Score.Rmd
    Modified:   analysis/Figure_1.rmd
    Modified:   analysis/Figure_2.rmd
    Modified:   analysis/Figure_3.rmd
    Modified:   analysis/Figure_4.rmd
    Modified:   analysis/Figure_5.rmd
    Modified:   analysis/Summary_Statistics.rmd
    Modified:   analysis/Supp-Figure_1.rmd
    Modified:   analysis/Supp-Figure_2.rmd
    Modified:   analysis/Supp-Figure_3.rmd
    Modified:   analysis/Supp-Figure_4.rmd
    Modified:   analysis/Supp-Figure_5.rmd
    Modified:   analysis/XX_hazard_ratio.rmd
    Deleted:    code/findPackages.R
    Deleted:    code/helper_functions/findClusters.R
    Deleted:    code/helper_functions/findCommunity.R
    Deleted:    code/helper_functions/getCellCount.R
    Deleted:    code/helper_functions/plotBarFracCluster.R
    Deleted:    code/helper_functions/plotCellFrac.R
    Deleted:    code/helper_functions/plotCellFracGroups.R
    Deleted:    code/helper_functions/plotCellFracGroupsSubset.R
    Modified:   code/helper_functions/scatter_function.R
    Modified:   code/helper_functions/validityChecks.R
    Deleted:    data/mask_comparison/20190809_ZTMA256.1_slide2_TH_s1_p1_r15_a15_ac_full.tiff
    Deleted:    data/mask_comparison/20190809_ZTMA256.1_slide2_TH_s1_p1_r15_a15_ac_ilastik_s2_Probabilities_equalized_cellmask.tiff
    Deleted:    data/mask_comparison/20191023_ZTMA256.1_slide3_TH_s0_p10_r4_a4_ac_full.tiff
    Deleted:    data/mask_comparison/20191023_ZTMA256.1_slide3_TH_s0_p10_r4_a4_ac_ilastik_s2_Probabilities_equalized_cellmask.tiff

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>load packages</h1>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions/&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions//calculateSummary.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//censor_dat.R
value   ?                                  
visible FALSE                              
        code/helper_functions//detect_mRNA_expression.R
value   ?                                              
visible FALSE                                          
        code/helper_functions//DistanceToClusterCenter.R
value   ?                                               
visible FALSE                                           
        code/helper_functions//findMilieu.R code/helper_functions//findPatch.R
value   ?                                   ?                                 
visible FALSE                               FALSE                             
        code/helper_functions//getInfoFromString.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//getSpotnumber.R
value   ?                                     
visible FALSE                                 
        code/helper_functions//plotCellCounts.R
value   ?                                      
visible FALSE                                  
        code/helper_functions//plotCellFractions.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//plotDist.R
value   ?                                
visible FALSE                            
        code/helper_functions//scatter_function.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//sceChecks.R
value   ?                                 
visible FALSE                             
        code/helper_functions//validityChecks.R
value   ?                                      
visible FALSE                                  </code></pre>
<pre class="r"><code>library(LSD)
library(SingleCellExperiment)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre class="r"><code>library(ggplot2)
library(scater)
library(viridis)</code></pre>
<pre><code>Loading required package: viridisLite</code></pre>
<pre class="r"><code>library(Rphenograph)
library(igraph)</code></pre>
<pre><code>
Attaching package: &#39;igraph&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:scater&#39;:

    normalize</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    union</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    union</code></pre>
<pre><code>The following object is masked from &#39;package:S4Vectors&#39;:

    union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    normalize, path, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    decompose, spectrum</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    union</code></pre>
<pre class="r"><code>library(CATALYST)
library(workflowr)</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>load data</h1>
<pre class="r"><code>sce = readRDS(file = &quot;data/data_for_analysis/sce_protein.rds&quot;)</code></pre>
<pre class="r"><code>for(i in 1:10){
scatter_x_y(sce,x = &quot;CD8&quot;,y = &quot;pERK&quot;,imagenumber =i, xlim = c(0,5), ylim = c(0,5))
  abline(h = 1.25)
  abline(v=1.5)
  abline(a = 0, b= 1)
}</code></pre>
<p><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_2_Protein_classification_subclustering.rmd/scatter%20plot%20to%20check%20cut%20offs%20for%20single%20markers-10.png" width="672" style="display: block; margin: auto;" /></p>
<div id="mean-expression-of-markers-per-celltype" class="section level2">
<h2>mean expression of markers per celltype</h2>
<pre class="r"><code># Aggregate the counts. calculateSummary is a function written from Nils
mean_sce &lt;- calculateSummary(sce, split_by = &quot;celltype&quot;, 
                             exprs_values = &quot;counts&quot;)

# Exclude DNA and Histone
mean_sce &lt;- mean_sce[!grepl(&quot;DNA|Histone&quot;, rownames(mean_sce)),]

# Transform and scale
assay(mean_sce, &quot;arcsinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;)) 
assay(mean_sce, &quot;arcsinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))

# Linear scale 
plotHeatmap(mean_sce, exprs_values = &quot;arcsinh&quot;, 
            features = c(rownames(mean_sce)), 
            colour_columns_by = &quot;celltype&quot;, 
            color = viridis(100), cluster_rows = FALSE, gaps_row = 10)</code></pre>
<p><img src="figure/04_2_Protein_classification_subclustering.rmd/heatmap%20mean%20per%20celltype%20asinh-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="scaled-mean-expression-of-markers-per-celltype" class="section level2">
<h2>scaled mean expression of markers per celltype</h2>
<pre class="r"><code># Linear scale 
plotHeatmap(mean_sce, exprs_values = &quot;arcsinh_scaled&quot;, 
            features = c(rownames(mean_sce)), 
            colour_columns_by = &quot;celltype&quot;, 
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            zlim = c(-3,3), cluster_rows = TRUE, gaps_row = 10)</code></pre>
<p><img src="figure/04_2_Protein_classification_subclustering.rmd/heatmap%20mean%20per%20celltype%20asinh%20scaled-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="definition-of-markers-used-for-subclustering-the-super-classes" class="section level2">
<h2>definition of markers used for subclustering the super-classes</h2>
<pre class="r"><code># Select markers
marker_list &lt;- list()

marker_list$Tumor &lt;- c(&quot;Ki67&quot;, &quot;PDL1&quot;, &quot;PARP&quot;, &quot;H3K27me3&quot;,&quot;CXCR2&quot;,&quot;HLADR&quot;,&quot;S100&quot;,&quot;Sox9&quot;,&quot;pERK&quot;,&quot;CD36&quot;,
                       &quot;p75&quot;,&quot;MiTF&quot;,&quot;bCatenin&quot;,&quot;Collagen1&quot;,&quot;pS6&quot;,&quot;IDO1&quot;,&quot;SOX10&quot;)

marker_list$Neutrophil &lt;- c(&quot;CD15&quot;, &quot;PDL1&quot;, &quot;MPO&quot;, &quot;CD11b&quot;)


marker_list$Tregulatory &lt;- c(&quot;IDO1&quot;, &quot;PD1&quot;, &quot;FOXP3&quot;, &quot;ICOS&quot;, &quot;CD4&quot;, &quot;Ki67&quot;, &quot;GrzB&quot;,
                        &quot;CD45RA&quot;, &quot;CD45RO&quot;)

marker_list$Tcytotoxic &lt;- c(&quot;IDO1&quot;, &quot;TOX1&quot;, &quot;PD1&quot;, &quot;TCF7&quot;, &quot;ICOS&quot;, &quot;CD8&quot;,&quot;Ki67&quot;, &quot;GrzB&quot;,
                        &quot;CD45RA&quot;, &quot;CD45RO&quot;)

marker_list$Thelper &lt;- c(&quot;IDO1&quot;, &quot;TOX1&quot;, &quot;PD1&quot;, &quot;TCF7&quot;, 
                        &quot;FOXP3&quot;, &quot;ICOS&quot;, &quot;CD4&quot;, &quot;Ki67&quot;, &quot;GrzB&quot;,
                        &quot;CD45RA&quot;, &quot;CD45RO&quot;)

marker_list$BnTcell &lt;- c(&quot;IDO1&quot;, &quot;TOX1&quot;, &quot;PD1&quot;, &quot;TCF7&quot;, 
                        &quot;FOXP3&quot;, &quot;ICOS&quot;, &quot;CD8&quot;, &quot;CD4&quot;, &quot;Ki67&quot;, &quot;GrzB&quot;,
                        &quot;CD45RA&quot;, &quot;CD45RO&quot;, &quot;CD20&quot;,&quot;HLADR&quot;,&quot;H3K27me3&quot;,&quot;pS6&quot;,&quot;pERK&quot;, &quot;CD7&quot;)

marker_list$pDC &lt;- c(&quot;CD303&quot;,&quot;IDO1&quot;,&quot;GrzB&quot;,&quot;CXCR2&quot;,&quot;CD11b&quot;)

marker_list$Bcell &lt;- c(&quot;Ki67&quot;,&quot;CD45RA&quot;, &quot;CD45RO&quot;, &quot;CD20&quot;,&quot;HLADR&quot;,&quot;H3K27me3&quot;,&quot;pS6&quot;,&quot;pERK&quot;)

marker_list$Macrophage &lt;- c(&quot;Caveolin1&quot;, &quot;PDL1&quot;, &quot;PARP&quot;, &quot;H3K27me3&quot;,&quot;CXCR2&quot;,&quot;HLADR&quot;,&quot;CD45RO&quot;,&quot;CD45RA&quot;,&quot;CD68&quot;,&quot;pERK&quot;,&quot;CD36&quot;, &quot;Collagen1&quot;,&quot;CD11c&quot;,&quot;pS6&quot;,&quot;IDO1&quot;,&quot;CD206&quot;)

marker_list$Stroma &lt;- c(&quot;Caveolin1&quot;, &quot;CD68&quot;,&quot;CD36&quot;, &quot;Collagen1&quot;,&quot;SMA&quot;)

marker_list$unknown &lt;- rownames(sce[rowData(sce)$good_marker,])</code></pre>
</div>
<div id="cluster-the-whole-dataset-with-flowsom" class="section level2">
<h2>cluster the whole dataset with FlowSOM</h2>
<p>FlowSOM first because it is faster</p>
<pre class="r"><code>## the FlowSOM function from CATALYST needs an another column in the rowData of the sce to work properly:
rowData(sce)$marker_class &lt;- &quot;state&quot;

# vector for clustering
fs_clustering &lt;- vector(length = ncol(sce))

# create the &quot;exprs&quot; slot in the assay data (needed for CATALYST)
assay(sce, &quot;exprs&quot;) &lt;- assay(sce,&quot;asinh&quot;)

# Macrophage, Bcells, Thelper, Tcytotoxic, Tother and BnT cells will be clustered for a total of 6 clustes each
set.seed(12345)
for(i in c(&quot;Macrophage&quot;,&quot;Thelper&quot;,&quot;Tcytotoxic&quot;)){
  cur_sce &lt;- sce[,sce$celltype == i]
  
  cur_sce &lt;- cluster(cur_sce,features = marker_list[i][[1]],ydim = 2,xdim = 3,maxK = 4)
  
  fs_clustering[sce$celltype == i] &lt;- cur_sce$cluster_id
}</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre class="r"><code># pDCs and Neutrophils will be clustered to 4 clusteres
for(i in c(&quot;pDC&quot;,&quot;Neutrophil&quot;,&quot;BnTcell&quot;,&quot;Bcell&quot;,&quot;Tregulatory&quot;,&quot;Stroma&quot;, &quot;unknown&quot;)){
  cur_sce &lt;- sce[,sce$celltype == i]
  
  cur_sce &lt;- cluster(cur_sce,features = marker_list[i][[1]],ydim = 2,xdim = 2,maxK = 3)
  
  fs_clustering[sce$celltype == i] &lt;- cur_sce$cluster_id
}</code></pre>
<pre><code>o running FlowSOM clustering...
o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre><code>o running FlowSOM clustering...</code></pre>
<pre><code>o running ConsensusClusterPlus metaclustering...</code></pre>
<pre class="r"><code># Tumor will be clustered into 9 clusters
cur_sce &lt;- sce[,sce$celltype == &quot;Tumor&quot;]
  
cur_sce &lt;- cluster(cur_sce,features = marker_list[&quot;Tumor&quot;][[1]],ydim = 3,xdim = 3,maxK = 7)</code></pre>
<pre><code>o running FlowSOM clustering...
o running ConsensusClusterPlus metaclustering...</code></pre>
<pre class="r"><code>fs_clustering[sce$celltype == &quot;Tumor&quot;] &lt;- cur_sce$cluster_id

# Save in SCE object
colData(sce)$celltype_clustered &lt;- as.factor(fs_clustering)

sce$celltype_clustered &lt;- paste0(sce$celltype, &quot;_&quot;, sce$celltype_clustered)</code></pre>
</div>
<div id="visulalize-clustering-results-for-flowsom" class="section level2">
<h2>visulalize clustering results for FlowSOM</h2>
<pre class="r"><code># Aggregate the counts
mean_sce &lt;- calculateSummary(sce, split_by = c(&quot;celltype&quot;,&quot;celltype_clustered&quot;), 
                             exprs_values = &quot;counts&quot;)

# Exclude DNA and Histone
mean_sce &lt;- mean_sce[!grepl(&quot;DNA|Histone&quot;, rownames(mean_sce)),]

# Transform and scale
assay(mean_sce, &quot;arcsinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;)) 
assay(mean_sce, &quot;arcsinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))

# Linear scale 
plotHeatmap(mean_sce, exprs_values = &quot;arcsinh&quot;, 
            colour_columns_by = c(&quot;celltype&quot;), 
            color = viridis(100), labels_col = mean_sce$celltype_clustered,
            show_colnames = TRUE, annotation_legend = FALSE, borders_color = NA)</code></pre>
<p><img src="figure/04_2_Protein_classification_subclustering.rmd/heatmap%20mean%20expression%20per%20celltype%20subclustered%20asinh-1.png" width="1920" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Scaled
plotHeatmap(mean_sce, exprs_values = &quot;arcsinh_scaled&quot;, 
            colour_columns_by = c(&quot;celltype&quot;), 
            labels_col = mean_sce$celltype_clustered,
            show_colnames = TRUE, annotation_legend = TRUE, borders_color = NA,
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            zlim = c(-4, 4),legend = TRUE)</code></pre>
<p><img src="figure/04_2_Protein_classification_subclustering.rmd/heatmap%20mean%20expression%20per%20celltype%20subclustered%20asinh%20scaled-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="numbers-of-cells-per-cluster" class="section level2">
<h2>numbers of cells per cluster</h2>
<pre class="r"><code>table(sce$celltype_clustered)</code></pre>
<pre><code>
      Bcell_1       Bcell_2       Bcell_3       Bcell_4     BnTcell_1 
        22407          4277         13208         10504          6449 
    BnTcell_2     BnTcell_3     BnTcell_4  Macrophage_1  Macrophage_2 
         6905          8722          9202         12808         14507 
 Macrophage_3  Macrophage_4  Macrophage_5  Macrophage_6  Neutrophil_1 
        11276          7769          7634          8465          3483 
 Neutrophil_2  Neutrophil_3  Neutrophil_4         pDC_1         pDC_2 
         2949          2105          2479          2667          1837 
        pDC_3         pDC_4      Stroma_1      Stroma_2      Stroma_3 
         1816           943          9888         14920         17433 
     Stroma_4  Tcytotoxic_1  Tcytotoxic_2  Tcytotoxic_3  Tcytotoxic_4 
        18504          7484          9623          6782          8039 
 Tcytotoxic_5  Tcytotoxic_6     Thelper_1     Thelper_2     Thelper_3 
        10739         10535         11094          9694         11068 
    Thelper_4     Thelper_5     Thelper_6 Tregulatory_1 Tregulatory_2 
         3019         12192         12970          3081          2345 
Tregulatory_3 Tregulatory_4       Tumor_1       Tumor_2       Tumor_3 
         2705          1275         76990         78446         90295 
      Tumor_4       Tumor_5       Tumor_6       Tumor_7       Tumor_8 
        52280         72756         79784         45770         53527 
      Tumor_9     unknown_1     unknown_2     unknown_3     unknown_4 
        74583          2672          1570         10194          4735 </code></pre>
</div>
<div id="assign-names-to-clusters" class="section level2">
<h2>assign names to clusters</h2>
<pre class="r"><code>annotations &lt;- sce$celltype_clustered

# annotations[annotations == &quot;other_4&quot;] &lt;- &quot;Other_CD206Low_pS6Low&quot;
# annotations[annotations == &quot;other_2&quot;] &lt;- &quot;Tumor_SOX9Low_SOX10+_MITF+_S100+_bCatenin++&quot;
# annotations[annotations == &quot;other_3&quot;] &lt;- &quot;Other_SOX10Low_MITFLow_bCatenin+&quot;
# annotations[annotations == &quot;other_1&quot;] &lt;- &quot;Other_CaveolinLow&quot;
# annotations[annotations == &quot;other_5&quot;] &lt;- &quot;Vasculature_SMA++_Collagen+_Caveolin+&quot;
# annotations[annotations == &quot;other_6&quot;] &lt;- &quot;Stroma_SMA+_Collagen++_Caveolin++_CD36++&quot;
# 
# annotations[annotations == &quot;Tumor_2&quot;] &lt;- &quot;Tumor_SOX9++_p75+&quot;
# annotations[annotations == &quot;Tumor_3&quot;] &lt;- &quot;Tumor_SOX10+_MITFLow_CD36Low&quot;
# annotations[annotations == &quot;Tumor_4&quot;] &lt;- &quot;Tumor_SOX10Low_S100+&quot;
# annotations[annotations == &quot;Tumor_5&quot;] &lt;- &quot;Tumor_SOX9++_SOX10+_MITF+_S100Low&quot;
# annotations[annotations == &quot;Tumor_8&quot;] &lt;- &quot;Tumor_SOX9Low_SOX10Low_MITFLow_p75+&quot;
# annotations[annotations == &quot;Tumor_9&quot;] &lt;- &quot;Tumor_SOX10+_MITF+_S100++&quot;
# annotations[annotations == &quot;Tumor_6&quot;] &lt;- &quot;Tumor_S100+&quot;
# annotations[annotations == &quot;Tumor_1&quot;] &lt;- &quot;Tumor_SOX10Low_MITFLow_HLADR+&quot;
# annotations[annotations == &quot;Tumor_7&quot;] &lt;- &quot;Tumor_SOX10+_MITF++_S100+_pS6+&quot;
# 
# annotations[annotations == &quot;Bcells_1&quot;] &lt;- &quot;Bcells_pErkLow_H3K27meLow&quot;
# annotations[annotations == &quot;Bcells_4&quot;] &lt;- &quot;Bcells_pErkLow_H3K27meLow&quot;
# annotations[annotations == &quot;Bcells_2&quot;] &lt;- &quot;Bcells_H3K27me_LowpS6Low&quot;
# annotations[annotations == &quot;Bcells_3&quot;] &lt;- &quot;Bcells_CD19++_pErkLow_H3K27meLow&quot;
# annotations[annotations == &quot;Bcells_5&quot;] &lt;- &quot;Bcells_Ki67++&quot;
# annotations[annotations == &quot;Bcells_6&quot;] &lt;- &quot;Bcells_SMALow&quot;
# 
# annotations[annotations == &quot;BnT_1&quot;] &lt;- &quot;Bcells_CD3Low_CD4Low&quot;
# annotations[annotations == &quot;BnT_2&quot;] &lt;- &quot;Bcells_pErkLow_H3K27meLow&quot;
# annotations[annotations == &quot;BnT_6&quot;] &lt;- &quot;BnT_CD8+&quot;
# annotations[annotations == &quot;BnT_4&quot;] &lt;- &quot;BnT_CD8+&quot;
# annotations[annotations == &quot;BnT_5&quot;] &lt;- &quot;BnT_CD4+_TCF7+&quot;
# annotations[annotations == &quot;BnT_3&quot;] &lt;- &quot;BnT_CD4+_ICOS++_PD1++_Ki67+&quot;
# 
# annotations[annotations == &quot;T_other_4&quot;] &lt;- &quot;Tcells_undefined&quot;
# annotations[annotations == &quot;T_other_5&quot;] &lt;- &quot;Tcells_undefined&quot;
# annotations[annotations == &quot;T_other_3&quot;] &lt;- &quot;Tcells_CD15_LowGrzB+&quot;
# annotations[annotations == &quot;T_other_1&quot;] &lt;- &quot;Tcells_Ki67++&quot;
# annotations[annotations == &quot;T_other_2&quot;] &lt;- &quot;Tcells_CaveolinLow_SMALow&quot;
# annotations[annotations == &quot;T_other_6&quot;] &lt;- &quot;Tcells_undefined&quot;
# 
# annotations[annotations == &quot;T_helper_2&quot;] &lt;- &quot;Thelper_CD11cLow_HLADRLow&quot;
# annotations[annotations == &quot;T_helper_5&quot;] &lt;- &quot;Thelper_SMALow_CaveolinLow&quot;
# annotations[annotations == &quot;T_helper_1&quot;] &lt;- &quot;Thelper_TCF7++&quot;
# annotations[annotations == &quot;T_helper_6&quot;] &lt;- &quot;Thelper_FoxP3++_ICOS+_CD11cLow_CD206Low&quot;
# annotations[annotations == &quot;T_helper_3&quot;] &lt;- &quot;Thelper_FoxP3Low_pERKLow&quot;
# annotations[annotations == &quot;T_helper_4&quot;] &lt;- &quot;Thelper_FoxP3Low&quot;
# 
# annotations[annotations == &quot;T_cytotoxic_3&quot;] &lt;- &quot;Tcytotoxic_PD1+_TOX1+_GrzBLow&quot;
# annotations[annotations == &quot;T_cytotoxic_5&quot;] &lt;- &quot;Tcytotoxic_TCF7+_H3K27meLow&quot;
# annotations[annotations == &quot;T_cytotoxic_6&quot;] &lt;- &quot;Tcytotoxic_PD1+_TOX1+_GrzB+_ICOSLow_Ki67+&quot;
# annotations[annotations == &quot;T_cytotoxic_1&quot;] &lt;- &quot;Tcytotoxic_GrzBLow_SMALow_CaveolinLow&quot;
# annotations[annotations == &quot;T_cytotoxic_2&quot;] &lt;- &quot;Tcytotoxic_GrzBLow_pS6Low&quot;
# annotations[annotations == &quot;T_cytotoxic_4&quot;] &lt;- &quot;Tcytotoxic_PD1+_TOX1+_GrzBLow_ICOSLow_CD11cLow_CD206Low&quot;
# 
# annotations[annotations == &quot;Macrophage_4&quot;] &lt;- &quot;Macrophage_CD68Low_CaveolinLow&quot;
# annotations[annotations == &quot;Macrophage_6&quot;] &lt;- &quot;Tumor_CD68Low_S100Low_bCateninLow_MITFLow_SOX10Low_HLADR+&quot;
# annotations[annotations == &quot;Macrophage_2&quot;] &lt;- &quot;Macrophage_CD68Low_CD206++_CD36Low&quot;
# annotations[annotations == &quot;Macrophage_1&quot;] &lt;- &quot;Macrophage_CD68Low_pS6Low_S100Low&quot;
# annotations[annotations == &quot;Macrophage_3&quot;] &lt;- &quot;Macrophage_CD68+_CD11c+_CD206++_CXCR2+_PDL1+&quot;
# annotations[annotations == &quot;Macrophage_5&quot;] &lt;- &quot;Macrophage_CD68++_CD11c+_CD36+_Caveolin+_Collagen+&quot;
# 
# annotations[annotations == &quot;pDC_1&quot;] &lt;- &quot;pDC_IDO1+_GrzB+&quot;
# annotations[annotations == &quot;pDC_2&quot;] &lt;- &quot;pDC_IDO1+_TOX1+_CD11b+&quot;
# annotations[annotations == &quot;pDC_3&quot;] &lt;- &quot;pDC_IDO1++_GrzB++&quot;
# annotations[annotations == &quot;pDC_4&quot;] &lt;- &quot;pDC_Ido1+_PARP++&quot;
# 
# annotations[annotations == &quot;Neutrophil_2&quot;] &lt;- &quot;Neutrophil_CD11b+_CD15++_MPO+_PDL1++&quot;
# annotations[annotations == &quot;Neutrophil_4&quot;] &lt;- &quot;Neutrophil_CD11b+_CD15+_MPO+_PDL1Low&quot;
# annotations[annotations == &quot;Neutrophil_3&quot;] &lt;- &quot;Neutrophil_CD11b++_CD15++_MPO++_CXCR2Low_PDL1+&quot;
# annotations[annotations == &quot;Neutrophil_1&quot;] &lt;- &quot;Neutrophil_CD11b+_CD15+_MPO++&quot;

sce$cellAnnotation &lt;- annotations

mean_sce &lt;- calculateSummary(sce, split_by = c(&quot;celltype&quot;,&quot;cellAnnotation&quot;), 
                             exprs_values = &quot;counts&quot;)

# Exclude DNA and Histone
mean_sce &lt;- mean_sce[!grepl(&quot;DNA|Histone&quot;, rownames(mean_sce)),]

# Transform and scale
assay(mean_sce, &quot;arcsinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;)) 
assay(mean_sce, &quot;arcsinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))

plotHeatmap(mean_sce, exprs_values = &quot;arcsinh_scaled&quot;, 
            colour_columns_by = c(&quot;celltype&quot;), 
            labels_col = mean_sce$cellAnnotation,
            show_colnames = TRUE, annotation_legend = TRUE, borders_color = NA,
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            zlim = c(-4, 4),legend = TRUE)</code></pre>
<p><strong>clustering does not really pick up PDL1 positive Tumor cells (which definitely exist). Potentially we have to investigate this with a manual cut off. arcsinh &gt; 1.75 seems to be reasonable</strong></p>
<p><strong>Additionally, the clustering did not pick up proliferating and non prolifertating tumor clusters. this should also be investigated by manual cut offs. arcsinh &gt; 1.25 seems reasonable.</strong></p>
<pre class="r"><code>celltype_counts &lt;- sce$celltype

table(celltype_counts)</code></pre>
</div>
<div id="save-the-single-cell-object-with-the-clusters-assigned-to-the-cells" class="section level2">
<h2>save the single cell object with the clusters assigned to the cells</h2>
<pre class="r"><code># delete the &quot;exprs&quot; slot from the single cell experiment again.
assay(sce,&quot;exprs&quot;) &lt;- NULL

saveRDS(sce,file = &quot;data/data_for_analysis/sce_protein.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] CATALYST_1.12.2             igraph_1.2.6               
 [3] Rphenograph_0.99.1.9003     viridis_0.5.1              
 [5] viridisLite_0.3.0           scater_1.16.2              
 [7] ggplot2_3.3.3               SingleCellExperiment_1.12.0
 [9] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[11] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[13] IRanges_2.24.1              S4Vectors_0.28.1           
[15] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[17] matrixStats_0.57.0          LSD_4.1-0                  
[19] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                circlize_0.4.12            
  [3] drc_3.0-1                   plyr_1.8.6                 
  [5] ConsensusClusterPlus_1.52.0 splines_4.0.3              
  [7] flowCore_2.0.1              BiocParallel_1.22.0        
  [9] TH.data_1.0-10              digest_0.6.27              
 [11] htmltools_0.5.0             magrittr_2.0.1             
 [13] CytoML_2.0.5                cluster_2.1.0              
 [15] openxlsx_4.2.3              ComplexHeatmap_2.4.3       
 [17] RcppParallel_5.0.2          sandwich_3.0-0             
 [19] flowWorkspace_4.0.6         cytolib_2.0.3              
 [21] jpeg_0.1-8.1                colorspace_2.0-0           
 [23] ggrepel_0.9.0               haven_2.3.1                
 [25] xfun_0.20                   dplyr_1.0.2                
 [27] crayon_1.3.4                RCurl_1.98-1.2             
 [29] jsonlite_1.7.2              hexbin_1.28.2              
 [31] graph_1.66.0                survival_3.2-7             
 [33] zoo_1.8-8                   glue_1.4.2                 
 [35] gtable_0.3.0                nnls_1.4                   
 [37] zlibbioc_1.36.0             XVector_0.30.0             
 [39] GetoptLong_1.0.5            DelayedArray_0.16.0        
 [41] ggcyto_1.16.0               car_3.0-10                 
 [43] BiocSingular_1.4.0          Rgraphviz_2.32.0           
 [45] shape_1.4.5                 abind_1.4-5                
 [47] scales_1.1.1                pheatmap_1.0.12            
 [49] mvtnorm_1.1-1               Rcpp_1.0.5                 
 [51] plotrix_3.7-8               clue_0.3-58                
 [53] foreign_0.8-81              rsvd_1.0.3                 
 [55] FlowSOM_1.20.0              tsne_0.1-3                 
 [57] RColorBrewer_1.1-2          ellipsis_0.3.1             
 [59] farver_2.0.3                pkgconfig_2.0.3            
 [61] XML_3.99-0.5                reshape2_1.4.4             
 [63] tidyselect_1.1.0            rlang_0.4.10               
 [65] later_1.1.0.1               munsell_0.5.0              
 [67] cellranger_1.1.0            tools_4.0.3                
 [69] generics_0.1.0              ggridges_0.5.3             
 [71] evaluate_0.14               stringr_1.4.0              
 [73] yaml_2.2.1                  knitr_1.30                 
 [75] fs_1.5.0                    zip_2.1.1                  
 [77] purrr_0.3.4                 RANN_2.6.1                 
 [79] RBGL_1.64.0                 xml2_1.3.2                 
 [81] compiler_4.0.3              rstudioapi_0.13            
 [83] beeswarm_0.2.3              curl_4.3                   
 [85] png_0.1-7                   tibble_3.0.4               
 [87] stringi_1.5.3               forcats_0.5.0              
 [89] lattice_0.20-41             Matrix_1.3-2               
 [91] vctrs_0.3.6                 pillar_1.4.7               
 [93] lifecycle_0.2.0             GlobalOptions_0.1.2        
 [95] BiocNeighbors_1.6.0         data.table_1.13.6          
 [97] cowplot_1.1.1               bitops_1.0-6               
 [99] irlba_2.3.3                 httpuv_1.5.4               
[101] R6_2.5.0                    latticeExtra_0.6-29        
[103] promises_1.1.1              gridExtra_2.3              
[105] RProtoBufLib_2.0.0          rio_0.5.16                 
[107] vipor_0.4.5                 codetools_0.2-18           
[109] MASS_7.3-53                 gtools_3.8.2               
[111] rprojroot_2.0.2             rjson_0.2.20               
[113] withr_2.3.0                 multcomp_1.4-15            
[115] GenomeInfoDbData_1.2.4      hms_0.5.3                  
[117] ncdfFlow_2.34.0             grid_4.0.3                 
[119] rmarkdown_2.6               DelayedMatrixStats_1.10.1  
[121] carData_3.0-4               Rtsne_0.15                 
[123] git2r_0.28.0                base64enc_0.1-3            
[125] ggbeeswarm_0.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
