<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="toobiwankenobi" />

<meta name="date" content="2020-07-28" />

<title>04_2_RNA_classification_subclustering</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Patch_Detection.html">Bcell Patch Detection</a>
    </li>
    <li>
      <a href="12_Bcell_Score.html">Bcell Score</a>
    </li>
    <li>
      <a href="13_Dysfunction_Stain.html">Dysfunction Validation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplementary Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
    <li>
      <a href="Supp-Figure_6.html">Supp-Figure 6</a>
    </li>
    <li>
      <a href="Supp-Figure_7.html">Supp-Figure 7</a>
    </li>
    <li>
      <a href="Supp-Figure_8.html">Supp-Figure 8</a>
    </li>
    <li>
      <a href="Supp-Figure_9.html">Supp-Figure 9</a>
    </li>
    <li>
      <a href="Supp-Figure_10.html">Supp-Figure 10</a>
    </li>
    <li>
      <a href="Supp-Figure_11.html">Supp-Figure 11</a>
    </li>
    <li>
      <a href="Supp-Figure_12.html">Supp-Figure 12</a>
    </li>
    <li>
      <a href="Supp-Figure_13.html">Supp-Figure 13</a>
    </li>
    <li>
      <a href="Table_ClinicalData.html">Clinical Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">04_2_RNA_classification_subclustering</h1>
<h4 class="author">toobiwankenobi</h4>
<h4 class="date">2020-07-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MelanomaIMC/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtreea2860df480bbbf6833e9812ecd3c9f60a9b68e66targetblanka2860dfa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/a2860df480bbbf6833e9812ecd3c9f60a9b68e66" target="_blank">a2860df</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtreea2860df480bbbf6833e9812ecd3c9f60a9b68e66targetblanka2860dfa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/a2860df480bbbf6833e9812ecd3c9f60a9b68e66" target="_blank">a2860df</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    Table_S4.csv
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/._.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/

Unstaged changes:
    Modified:   analysis/Figure_5.rmd
    Modified:   analysis/Supp-Figure_12.rmd
    Modified:   analysis/Supp-Figure_13.rmd
    Modified:   analysis/Supp-Figure_5.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/04_2_RNA_classification_subclustering.rmd</code>) and HTML (<code>docs/04_2_RNA_classification_subclustering.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/b24a994b61b316018cfcb359fff9c56402179f24/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">b24a994</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-08
</td>
<td>
update code to work under R 4.1.2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/f9a3a83c8bd4e7c866f7199a6bc59723b5f11a35/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">f9a3a83</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-08
</td>
<td>
clean repo for release
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/MelanomaIMC/4109ff15127b705dbe3fb4b6916895cd1220b29b/docs/04_2_RNA_classification_subclustering.html" target="_blank">4109ff1</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-07-07
</td>
<td>
delete html files and adapt gitignore
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/32038914d8c7a9d59102c5e037cad3afae9ac319/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">3203891</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-19
</td>
<td>
change celltype names
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/MelanomaIMC/32038914d8c7a9d59102c5e037cad3afae9ac319/docs/04_2_RNA_classification_subclustering.html" target="_blank">3203891</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-19
</td>
<td>
change celltype names
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/ee1595de9ec17ee414a5beafb9ed08d174e12139/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/MelanomaIMC/ee1595de9ec17ee414a5beafb9ed08d174e12139/docs/04_2_RNA_classification_subclustering.html" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/MelanomaIMC/3f5af3f59358eebb33aa2a8a06fddca34dad6a6b/docs/04_2_RNA_classification_subclustering.html" target="_blank">3f5af3f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
add .html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/9442cb9048bc92f3320eda95d2b40ce49d09d10e/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">9442cb9</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-12-22
</td>
<td>
add all new files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/d8819f2b618d0620606a5e5b181d3b5340469763/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">d8819f2</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-08
</td>
<td>
read new data (nuclei expansion) and adapt scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/a21c858d4863579f86e8afbae8e85cff16d1a46f/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">a21c858</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-06
</td>
<td>
adapt pipeline
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/4fc7560e9bd6c88e6f5a452bb1ab89649f7c76a9/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">4fc7560</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
Update 04_2_RNA_classification_subclustering.rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/402481d55592414c260088fccb5a1b6ca1170ab8/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">402481d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
Update 04_2_RNA_classification_subclustering.rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/8babce78447b577f0d38e58e9ea69e81d53ff847/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">8babce7</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
update subclustering
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/f20c4acfea6bb9d9f1480e3c29633533b9aa71d8/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">f20c4ac</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
Update 04_2_RNA_classification_subclustering.rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/2c11d5cd8e42b3be31d1c1f512864b531f93fd7c/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">2c11d5c</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-08-05
</td>
<td>
add new scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/73881bb7aa187ca60764b60dd8e7b530965b7655/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">73881bb</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-07-30
</td>
<td>
add code to files from old repo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/cf46cfa8b12f4f0a91b191b1049f877eac74db54/analysis/04_2_RNA_classification_subclustering.rmd" target="_blank">cf46cfa</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-07-28
</td>
<td>
create files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In this script we try to set global thresholds for some key markers (CD3 etc) and then sub-cluster the respective cells with markers that we know are expressed on those cells.</p>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message= FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())</code></pre>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions/&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions//calculateSummary.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//censor_dat.R
value   ?                                  
visible FALSE                              
        code/helper_functions//detect_mRNA_expression.R
value   ?                                              
visible FALSE                                          
        code/helper_functions//DistanceToClusterCenter.R
value   ?                                               
visible FALSE                                           
        code/helper_functions//findMilieu.R code/helper_functions//findPatch.R
value   ?                                   ?                                 
visible FALSE                               FALSE                             
        code/helper_functions//getInfoFromString.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//getSpotnumber.R
value   ?                                     
visible FALSE                                 
        code/helper_functions//plotCellCounts.R
value   ?                                      
visible FALSE                                  
        code/helper_functions//plotCellFractions.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//plotDist.R code/helper_functions//read_Data.R
value   ?                                 ?                                 
visible FALSE                             FALSE                             
        code/helper_functions//scatter_function.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//sceChecks.R
value   ?                                 
visible FALSE                             
        code/helper_functions//validityChecks.R
value   ?                                      
visible FALSE                                  </code></pre>
<pre class="r"><code>sapply(list.files(&quot;/Volumes/server_homes/daniels/Git/imcRtools/R/&quot;, full.names = TRUE), source)</code></pre>
<pre><code>named list()</code></pre>
<pre class="r"><code>library(LSD)
library(SingleCellExperiment)
library(ggplot2)
library(scater)
library(viridis)
library(igraph)
library(CATALYST)
library(workflowr)
library(gridExtra)
library(dplyr)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<pre class="r"><code>sce &lt;- readRDS(file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
</div>
</div>
<div id="marker-for-clustering" class="section level1">
<h1>Marker for Clustering</h1>
<div id="define-superclasses-for-subclustering" class="section level2">
<h2>Define Superclasses for subclustering</h2>
<pre class="r"><code># store labels from randomForest separately
sce$celltype_rf &lt;- sce$celltype

# assign superclasses
sce[,sce$celltype_rf %in% c(&quot;exhausted Tcytotoxic&quot;, &quot;Tcytotoxic&quot;)]$celltype &lt;- &quot;CD8+ T cell&quot;
sce[,sce$celltype_rf %in% c(&quot;CD134+ Tcell&quot;, &quot;CD134- Tcell&quot;)]$celltype &lt;- &quot;CD8- T cell&quot;</code></pre>
</div>
<div id="definition-of-markers-used-for-subclustering-the-classified-cells" class="section level2">
<h2>Definition of markers used for subclustering the classified cells</h2>
<pre class="r"><code>marker_list &lt;- list()

marker_list$Stroma &lt;- c(&quot;SMA&quot;, &quot;CK5&quot;, &quot;Cadherin11&quot;, &quot;FAP&quot;,&quot;pRB&quot;,&quot;B2M&quot;,&quot;GLUT1&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;CCR2&quot;,&quot;T1_CXCL8&quot;,
                        &quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,&quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;CD31&quot;,
                        &quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;)

marker_list$Tumor &lt;- c(&quot;HLADR&quot;, &quot;S100&quot;,&quot;B2M&quot;,&quot;GLUT1&quot;,&quot;PDL1&quot;,&quot;SOX10&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;CCR2&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;,
                       &quot;Mart1&quot;,&quot;pRB&quot;)

marker_list$Neutrophil &lt;- c(&quot;CD15&quot;, &quot;PDL1&quot;,&quot;GLUT1&quot;, &quot;MPO&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)


marker_list$`CD8- T cell` &lt;- c(&quot;CD3&quot;, &quot;CD8&quot;, &quot;PD1&quot;, &quot;Lag3&quot;, &quot;CCR2&quot;,&quot;CD38&quot;,&quot;PDL1&quot;, &quot;HLADR&quot;, &quot;CD134&quot;,&quot;T5_CCL4&quot;, &quot;B2M&quot;, &quot;CD31&quot;,
                       &quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,&quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,
                       &quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)

marker_list$`CD8+ T cell` &lt;- c(&quot;CD3&quot;, &quot;CD8&quot;, &quot;PD1&quot;, &quot;Lag3&quot;, &quot;CCR2&quot;,&quot;CD38&quot;,&quot;PDL1&quot;, &quot;HLADR&quot;, &quot;CD134&quot;,&quot;T5_CCL4&quot;, &quot;B2M&quot;, &quot;CD31&quot;,
                       &quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,&quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,
                       &quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)


marker_list$Macrophage &lt;- c(&quot;CD68&quot;, &quot;CD163&quot;, &quot;HLADR&quot;, &quot;PDL1&quot;, &quot;CD3&quot;, &quot;CCR2&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)

marker_list$Vasculature &lt;- c(&quot;CD31&quot;, &quot;SMA&quot;,&quot;CK5&quot;,&quot;pRB&quot;, &quot;Cadherin11&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)

marker_list$`HLA-DR` &lt;- c(&quot;HLADR&quot;,&quot;CD68&quot;,&quot;CD3&quot;,&quot;CCR2&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)

marker_list$CD38 &lt;- c(&quot;HLADR&quot;,&quot;CD38&quot;,&quot;CD3&quot;,&quot;CCR2&quot;,&quot;CD163&quot;,&quot;CD68&quot;,&quot;T5_CCL4&quot;,&quot;T7_CCL18&quot;,&quot;T1_CXCL8&quot;,&quot;T4_CXCL10&quot;,&quot;T8_CXCL13&quot;,
                       &quot;T3_CXCL12&quot;,&quot;T12_CCL2&quot;,&quot;T2_CCL22&quot;,&quot;T9_CXCL9&quot;,&quot;T11_CCL8&quot;,&quot;T10_CCL19&quot;,&quot;cleavedPARP&quot;, &quot;pRB&quot;)

marker_list$unknown &lt;- rownames(sce[rowData(sce)$good_marker,])

# check if the names of the marker list match the names of the classified cells
all(names(marker_list) %in% unique(sce$celltype))</code></pre>
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div id="clustering" class="section level1">
<h1>Clustering</h1>
<div id="sub-cluster-the-whole-dataset-with-flowsom" class="section level2">
<h2>Sub-cluster the whole dataset with FlowSOM</h2>
<p>FlowSOM first because it is faster</p>
<pre class="r"><code>## the FlowSOM function from CATALYST needs an another column in the rowData of the sce to work properly:
rowData(sce)$marker_class &lt;- &quot;state&quot;

# vector for clustering
fs_clustering &lt;- vector(length = ncol(sce))

# Macrophage, CD8- T cell, CD8+ T cell, Tumor will be clustered for a total of 6 clustes each
set.seed(12345)

# 6 clusters
for(i in c(&quot;CD8- T cell&quot;,&quot;Macrophage&quot;,&quot;CD8+ T cell&quot;,&quot;Tumor&quot;)){
  #subset cells for clustering
  cur_sce &lt;- sce[,sce$celltype == i]
  names(assays(cur_sce)) &lt;- c(&quot;counts&quot;, &quot;exprs&quot;,&quot;scaled_counts&quot;, &quot;scaled_asinh&quot;)
  cur_sce &lt;- CATALYST::cluster(cur_sce, features = marker_list[i][[1]], ydim = 2, xdim = 3, maxK = 4)
  fs_clustering[sce$celltype == i] &lt;- cur_sce$cluster_id
}

# 4 clusters
for(i in c(&quot;Neutrophil&quot;,&quot;CD38&quot;,&quot;Stroma&quot;,&quot;Vasculature&quot;,&quot;HLA-DR&quot;, &quot;unknown&quot;)){
  cur_sce &lt;- sce[,sce$celltype == i]
  names(assays(cur_sce)) &lt;-  c(&quot;counts&quot;, &quot;exprs&quot;,&quot;scaled_counts&quot;, &quot;scaled_asinh&quot;)
  cur_sce &lt;- CATALYST::cluster(cur_sce, features = marker_list[i][[1]], ydim = 2, xdim = 2, maxK = 3)
  fs_clustering[sce$celltype == i] &lt;- cur_sce$cluster_id
}

# Save in SCE object
colData(sce)$celltype_clustered &lt;- as.factor(fs_clustering)

sce$celltype_clustered &lt;- paste0(sce$celltype, &quot;_&quot;, sce$celltype_clustered)</code></pre>
</div>
</div>
<div id="visualization" class="section level1">
<h1>Visualization</h1>
<div id="visulalize-clustering-results-for-flowsom" class="section level2">
<h2>Visulalize clustering results for FlowSOM</h2>
<pre class="r"><code># Aggregate the counts
mean_sce &lt;- calculateSummary(sce, split_by = c(&quot;celltype_clustered&quot;, &quot;celltype&quot;), 
                             exprs_values = &quot;counts&quot;)
# Exclude bad markers
mean_sce &lt;- mean_sce[rowData(sce)$good_marker,]

# Transform and scale
assay(mean_sce, &quot;arcsinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;)) 
assay(mean_sce, &quot;arcsinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))

# Scaled
plotHeatmap(mean_sce, exprs_values = &quot;arcsinh_scaled&quot;, 
            features = rownames(sce)[rowData(sce)$good_marker],
            colour_columns_by = &quot;celltype&quot;, 
            labels_col = mean_sce$celltype_clustered,
            show_colnames = TRUE, annotation_legend = TRUE, borders_color = NA,
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            zlim = c(-4, 4),legend = TRUE)</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/heatmap_mean_expression_per_celltypes_layer1_subclustered_asinh_scaled-1.png" width="1920" style="display: block; margin: auto;" /></p>
</div>
<div id="numbers-of-cells-per-cluster" class="section level2">
<h2>Numbers of cells per cluster</h2>
<pre class="r"><code>celltype_counts &lt;- sce$celltype_clustered

table(celltype_counts)</code></pre>
<pre><code>celltype_counts
       CD38_1        CD38_2        CD38_3        CD38_4 CD8- T cell_1 
         1703          2323          3115          4049         24011 
CD8- T cell_2 CD8- T cell_3 CD8- T cell_4 CD8- T cell_5 CD8- T cell_6 
        14807         13230         12678         24222          8924 
CD8+ T cell_1 CD8+ T cell_2 CD8+ T cell_3 CD8+ T cell_4 CD8+ T cell_5 
         6013          7272          5321         10216          5908 
CD8+ T cell_6      HLA-DR_1      HLA-DR_2      HLA-DR_3      HLA-DR_4 
        10474         13176          4636          5941         12700 
 Macrophage_1  Macrophage_2  Macrophage_3  Macrophage_4  Macrophage_5 
         7513         13380         11788         13106          8945 
 Macrophage_6  Neutrophil_1  Neutrophil_2  Neutrophil_3  Neutrophil_4 
        10696           622          1223          1701          1508 
     Stroma_1      Stroma_2      Stroma_3      Stroma_4       Tumor_1 
         2325          3956          7841         10868         72776 
      Tumor_2       Tumor_3       Tumor_4       Tumor_5       Tumor_6 
        94020         69388         89337        108154        119002 
    unknown_1     unknown_2     unknown_3     unknown_4 Vasculature_1 
         1869           549           452           313         10360 
Vasculature_2 Vasculature_3 Vasculature_4 
         5568          4860          1424 </code></pre>
</div>
</div>
<div id="cluster-names" class="section level1">
<h1>Cluster Names</h1>
<div id="assign-names-to-clusters" class="section level2">
<h2>Assign names to clusters</h2>
<pre class="r"><code>annotations &lt;- sce$celltype_clustered

# add annotation if wanted
#annotations[annotations == &quot;Tumor_1&quot;] &lt;- &quot;Tumor1_&quot;
#annotations[annotations == &quot;Tumor_2&quot;] &lt;- &quot;Tumor2_&quot;
#annotations[annotations == &quot;Tumor_3&quot;] &lt;- &quot;Tumor3_&quot;
#annotations[annotations == &quot;Tumor_4&quot;] &lt;- &quot;Tumor4_&quot;
#annotations[annotations == &quot;Tumor_5&quot;] &lt;- &quot;Tumor5_&quot;
#annotations[annotations == &quot;Tumor_6&quot;] &lt;- &quot;Tumor6_&quot;

sce$cellAnnotation &lt;- annotations

mean_sce &lt;- calculateSummary(sce, split_by = c(&quot;celltype_clustered&quot;,&quot;celltype&quot;), 
                             exprs_values = &quot;counts&quot;)

# Exclude bad markers
mean_sce &lt;- mean_sce[rowData(sce)$good_marker,]

# Transform and scale
assay(mean_sce, &quot;arcsinh&quot;) &lt;- asinh(assay(mean_sce, &quot;meanCounts&quot;)) 
assay(mean_sce, &quot;arcsinh_scaled&quot;) &lt;- t(scale(t(asinh(assay(mean_sce, &quot;meanCounts&quot;)))))

plotHeatmap(mean_sce, exprs_values = &quot;arcsinh_scaled&quot;, 
            features = rownames(sce)[rowData(sce)$good_marker],
            colour_columns_by = c(&quot;celltype&quot;), 
            labels_col = mean_sce$celltype,
            show_colnames = TRUE, annotation_legend = TRUE, borders_color = NA,
            color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            zlim = c(-4, 4),legend = TRUE)</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/names%20for%20clusters-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="clustering-quality-control" class="section level1">
<h1>Clustering Quality Control</h1>
<div id="marker-expression-density-of-clusters-vs.-manually-gated---cd8-t-cell" class="section level2">
<h2>Marker expression density of clusters vs. manually gated - CD8+ T cell</h2>
<pre class="r"><code># Density of Expression in different CD8+ T cell clusters vs. manually gated Lag3+ cells
cur_sce &lt;- data.frame(colData(sce))
cur_exprs &lt;- data.frame(t(assays(sce)[[2]]))
cur_exprs &lt;- cbind(cur_exprs, cur_sce[,c(&quot;celltype&quot;, &quot;celltype_rf&quot;, &quot;celltype_clustered&quot;)])
cur_exprs$cellID &lt;- rownames(cur_exprs)

cur_exprs &lt;- cur_exprs %&gt;%
  reshape2::melt(id.vars = c(&quot;cellID&quot;, &quot;celltype&quot;, &quot;celltype_rf&quot;, &quot;celltype_clustered&quot;), 
                 variable.name = &quot;marker&quot;, value.name = &quot;asinh&quot;)

# clustering
a &lt;- cur_exprs %&gt;%
  filter(celltype == &quot;CD8+ T cell&quot; &amp; marker %in% marker_list$`CD8+ T cell`) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)

# manual gating
b &lt;- cur_exprs %&gt;%
  filter(celltype == &quot;CD8+ T cell&quot; &amp; marker %in% marker_list$`CD8+ T cell`) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_rf)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)

grid.arrange(a, b, ncol = 2)</code></pre>
<pre><code>Warning: Removed 17847 rows containing non-finite values (stat_density).
Removed 17847 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_Tcytotoxic-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="cd8--t-cell" class="section level2">
<h2>CD8- T cell</h2>
<pre class="r"><code># clustering
a &lt;- cur_exprs %&gt;%
  filter(celltype == &quot;CD8- T cell&quot; &amp; marker %in% marker_list$`CD8- T cell`) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)

# manual gating
b &lt;- cur_exprs %&gt;%
  filter(celltype == &quot;CD8- T cell&quot; &amp; marker %in% marker_list$`CD8- T cell`) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_rf)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)

grid.arrange(a, b, ncol = 2)</code></pre>
<pre><code>Warning: Removed 33619 rows containing non-finite values (stat_density).
Removed 33619 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_Tcells-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="tumor" class="section level2">
<h2>Tumor</h2>
<pre class="r"><code># clustering
cur_exprs %&gt;%
  filter(celltype == &quot;Tumor&quot; &amp; marker %in% marker_list$Tumor) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)</code></pre>
<pre><code>Warning: Removed 124307 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_Tumor-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="neutrophil" class="section level2">
<h2>Neutrophil</h2>
<pre class="r"><code># clustering
cur_exprs %&gt;%
  filter(celltype == &quot;Neutrophil&quot; &amp; marker %in% marker_list$Neutrophil) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)</code></pre>
<pre><code>Warning: Removed 1029 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_Neutrophil-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="macrophage" class="section level2">
<h2>Macrophage</h2>
<pre class="r"><code># clustering
cur_exprs %&gt;%
  filter(celltype == &quot;Macrophage&quot; &amp; marker %in% marker_list$Macrophage) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)</code></pre>
<pre><code>Warning: Removed 47900 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_Macrophage-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cd38" class="section level2">
<h2>CD38</h2>
<pre class="r"><code># clustering
cur_exprs %&gt;%
  filter(celltype == &quot;CD38&quot; &amp; marker %in% marker_list$CD38) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)</code></pre>
<pre><code>Warning: Removed 2477 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_CD38-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="unknown" class="section level2">
<h2>unknown</h2>
<pre class="r"><code># clustering
cur_exprs %&gt;%
  filter(celltype == &quot;unknown&quot; &amp; marker %in% marker_list$unknown) %&gt;%
  ggplot(data=., aes(asinh, color = celltype_clustered)) + 
  geom_density() +
  xlim(0,4) + 
  facet_wrap(~marker)</code></pre>
<pre><code>Warning: Removed 36 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/04_2_RNA_classification_subclustering.rmd/density_marker_unknown-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sce-object" class="section level1">
<h1>SCE object</h1>
<div id="save-the-single-cell-object" class="section level2">
<h2>Save the single cell object</h2>
<pre class="r"><code>saveRDS(sce, &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gridExtra_2.3               CATALYST_1.18.1            
 [3] igraph_1.2.11               viridis_0.6.2              
 [5] viridisLite_0.4.0           scater_1.22.0              
 [7] scuttle_1.4.0               ggplot2_3.3.5              
 [9] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0
[11] Biobase_2.54.0              GenomicRanges_1.46.1       
[13] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[15] S4Vectors_0.32.3            BiocGenerics_0.40.0        
[17] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[19] LSD_4.1-0                   dplyr_1.0.7                
[21] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                  tidyselect_1.1.1           
  [3] grid_4.1.2                  BiocParallel_1.28.3        
  [5] Rtsne_0.15                  aws.signature_0.6.0        
  [7] flowCore_2.6.0              munsell_0.5.0              
  [9] ScaledMatrix_1.2.0          codetools_0.2-18           
 [11] withr_2.4.3                 colorspace_2.0-2           
 [13] highr_0.9                   knitr_1.37                 
 [15] rstudioapi_0.13             ggsignif_0.6.3             
 [17] labeling_0.4.2              git2r_0.29.0               
 [19] GenomeInfoDbData_1.2.7      polyclip_1.10-0            
 [21] farver_2.1.0                pheatmap_1.0.12            
 [23] flowWorkspace_4.6.0         rprojroot_2.0.2            
 [25] vctrs_0.3.8                 generics_0.1.2             
 [27] TH.data_1.1-0               xfun_0.29                  
 [29] R6_2.5.1                    doParallel_1.0.16          
 [31] ggbeeswarm_0.6.0            clue_0.3-60                
 [33] rsvd_1.0.5                  bitops_1.0-7               
 [35] DelayedArray_0.20.0         assertthat_0.2.1           
 [37] promises_1.2.0.1            scales_1.1.1               
 [39] multcomp_1.4-18             beeswarm_0.4.0             
 [41] gtable_0.3.0                beachmat_2.10.0            
 [43] processx_3.5.2              RProtoBufLib_2.6.0         
 [45] sandwich_3.0-1              rlang_1.0.0                
 [47] GlobalOptions_0.1.2         splines_4.1.2              
 [49] rstatix_0.7.0               hexbin_1.28.2              
 [51] broom_0.7.12                reshape2_1.4.4             
 [53] yaml_2.2.2                  abind_1.4-5                
 [55] backports_1.4.1             httpuv_1.6.5               
 [57] RBGL_1.70.0                 tools_4.1.2                
 [59] ellipsis_0.3.2              jquerylib_0.1.4            
 [61] RColorBrewer_1.1-2          ggridges_0.5.3             
 [63] Rcpp_1.0.8                  plyr_1.8.6                 
 [65] base64enc_0.1-3             sparseMatrixStats_1.6.0    
 [67] zlibbioc_1.40.0             purrr_0.3.4                
 [69] RCurl_1.98-1.5              ps_1.6.0                   
 [71] FlowSOM_2.2.0               ggpubr_0.4.0               
 [73] GetoptLong_1.0.5            cowplot_1.1.1              
 [75] zoo_1.8-9                   ggrepel_0.9.1              
 [77] cluster_2.1.2               colorRamps_2.3             
 [79] fs_1.5.2                    magrittr_2.0.2             
 [81] ncdfFlow_2.40.0             data.table_1.14.2          
 [83] scattermore_0.7             circlize_0.4.13            
 [85] mvtnorm_1.1-3               whisker_0.4                
 [87] ggnewscale_0.4.5            evaluate_0.14              
 [89] XML_3.99-0.8                jpeg_0.1-9                 
 [91] shape_1.4.6                 ggcyto_1.22.0              
 [93] compiler_4.1.2              tibble_3.1.6               
 [95] crayon_1.4.2                ggpointdensity_0.1.0       
 [97] htmltools_0.5.2             later_1.3.0                
 [99] tidyr_1.2.0                 RcppParallel_5.1.5         
[101] aws.s3_0.3.21               DBI_1.1.2                  
[103] tweenr_1.0.2                ComplexHeatmap_2.10.0      
[105] MASS_7.3-55                 Matrix_1.4-0               
[107] car_3.0-12                  cli_3.1.1                  
[109] parallel_4.1.2              pkgconfig_2.0.3            
[111] getPass_0.2-2               xml2_1.3.3                 
[113] foreach_1.5.2               vipor_0.4.5                
[115] bslib_0.3.1                 XVector_0.34.0             
[117] drc_3.0-1                   stringr_1.4.0              
[119] callr_3.7.0                 digest_0.6.29              
[121] ConsensusClusterPlus_1.58.0 graph_1.72.0               
[123] rmarkdown_2.11              DelayedMatrixStats_1.16.0  
[125] curl_4.3.2                  gtools_3.9.2               
[127] rjson_0.2.21                lifecycle_1.0.1            
[129] jsonlite_1.7.3              carData_3.0-5              
[131] BiocNeighbors_1.12.0        fansi_1.0.2                
[133] pillar_1.7.0                lattice_0.20-45            
[135] plotrix_3.8-2               fastmap_1.1.0              
[137] httr_1.4.2                  survival_3.2-13            
[139] glue_1.6.1                  png_0.1-7                  
[141] iterators_1.0.13            Rgraphviz_2.38.0           
[143] nnls_1.4                    ggforce_0.3.3              
[145] stringi_1.7.6               sass_0.4.0                 
[147] BiocSingular_1.10.0         CytoML_2.6.0               
[149] latticeExtra_0.6-29         cytolib_2.6.1              
[151] irlba_2.3.5                </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
