<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tobias Hoch" />

<meta name="date" content="2020-07-28" />

<title>05_RNA_chemokine_expressing_cells</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Score.html">Bcell Score</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_1.html">Supp-Figure 1</a>
    </li>
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="XX_hazard_ratio.html">Hazard Ratio Analysis</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">05_RNA_chemokine_expressing_cells</h1>
<h4 class="author">Tobias Hoch</h4>
<h4 class="date">2020-07-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-18
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>melanoma_publication_old_data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtreeee1595de9ec17ee414a5beafb9ed08d174e12139targetblankee1595da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/ee1595de9ec17ee414a5beafb9ed08d174e12139" target="_blank">ee1595d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtreeee1595de9ec17ee414a5beafb9ed08d174e12139targetblankee1595da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/ee1595de9ec17ee414a5beafb9ed08d174e12139" target="_blank">ee1595d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/._clinical metadata preparation.Rmd
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._protein_neutrophil.png
    Ignored:    output/._rna_neutrophil.png
    Ignored:    output/PSOCKclusterOut/
    Ignored:    output/bcell_grouping.png
    Ignored:    output/dysfunction_correlation.pdf

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/04_1_Protein_celltype_classification.rmd
    Modified:   analysis/04_1_RNA_celltype_classification.rmd
    Modified:   analysis/04_2_RNA_classification_subclustering.rmd
    Modified:   analysis/04_2_protein_classification_subclustering.rmd
    Modified:   analysis/07_TCF7_PD1_gating.rmd
    Modified:   analysis/08_color_vectors.rmd
    Modified:   analysis/09_Tcell_Score.Rmd
    Modified:   analysis/10_Dysfunction_Score.rmd
    Modified:   analysis/11_Bcell_Score.Rmd
    Modified:   analysis/Figure_1.rmd
    Modified:   analysis/Figure_2.rmd
    Modified:   analysis/Figure_3.rmd
    Modified:   analysis/Figure_4.rmd
    Modified:   analysis/Figure_5.rmd
    Modified:   analysis/Supp-Figure_1.rmd
    Modified:   analysis/Supp-Figure_2.rmd
    Modified:   analysis/Supp-Figure_3.rmd
    Modified:   analysis/Supp-Figure_4.rmd
    Modified:   analysis/Supp-Figure_5.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/05_RNA_chemokine_expressing_cells.rmd</code>) and HTML (<code>docs/05_RNA_chemokine_expressing_cells.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/ee1595de9ec17ee414a5beafb9ed08d174e12139/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/ee1595de9ec17ee414a5beafb9ed08d174e12139/docs/05_RNA_chemokine_expressing_cells.html" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/3f5af3f59358eebb33aa2a8a06fddca34dad6a6b/docs/05_RNA_chemokine_expressing_cells.html" target="_blank">3f5af3f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
add .html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/9442cb9048bc92f3320eda95d2b40ce49d09d10e/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">9442cb9</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-12-22
</td>
<td>
add all new files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/77466b7a723aa4c2dec15a4d54d42b02670cc448/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">77466b7</a>
</td>
<td>
Tobias Hoch
</td>
<td>
2020-10-22
</td>
<td>
work on subfigures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/d8819f2b618d0620606a5e5b181d3b5340469763/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">d8819f2</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-08
</td>
<td>
read new data (nuclei expansion) and adapt scripts
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/73881bb7aa187ca60764b60dd8e7b530965b7655/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">73881bb</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-07-30
</td>
<td>
add code to files from old repo
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/cf46cfa8b12f4f0a91b191b1049f877eac74db54/analysis/05_RNA_chemokine_expressing_cells.rmd" target="_blank">cf46cfa</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-07-28
</td>
<td>
create files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here, we define chemokine-expressing cells</p>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message= FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())</code></pre>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>source(&quot;code/helper_functions/detect_mRNA_expression.R&quot;)
source(&quot;code/helper_functions/validityChecks.R&quot;)
library(SingleCellExperiment)
library(dplyr)
library(ggplot2)
library(scater)
library(CATALYST)
library(reshape2)
library(LSD)
library(data.table)
library(ComplexHeatmap)
library(corrplot)
library(pheatmap)
library(grid)
library(gridExtra)
library(tidyr)
library(colorRamps)</code></pre>
</div>
<div id="load-the-single-cell-experiment-object-and-the-image-metadata" class="section level2">
<h2>Load the single cell experiment object and the image metadata</h2>
<pre class="r"><code>sce &lt;- readRDS(file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
</div>
</div>
<div id="detect-chemokine-expressing-cells" class="section level1">
<h1>Detect Chemokine Expressing Cells</h1>
<div id="detection-of-chemokine-expressing-cells" class="section level2">
<h2>Detection of chemokine expressing cells</h2>
<p>for the detection of chemokine expressing cells we make use of the fact that we also measured a negative control (DapB).</p>
<pre class="r"><code># get the names of the chemokine channels without the negative control channel
chemokine_channels = rownames(sce[which(grepl(&quot;T\\d+_&quot;,rownames(sce)) &amp; ! grepl(&quot;DapB&quot;,rownames(sce))),])

# run function to define chemokine expressing cells 
output_list &lt;- compute_difference(sce, 
                          cellID = &quot;cellID&quot;, 
                          assay_name = &quot;asinh&quot;, 
                          threshold = 0.01, 
                          mRNA_channels = chemokine_channels, 
                          negative_control = &quot;T6_DapB&quot;, 
                          return_calc_metrics = TRUE)

# overwrite SCE object
sce &lt;- output_list$output_sce</code></pre>
</div>
<div id="plot-results-from-chemokine-detection" class="section level2">
<h2>Plot results from chemokine detection</h2>
<pre class="r"><code># check difference between DapB and signal (histogram)
for(i in chemokine_channels){
  
  # subset whole data set for visualization purposes
  diff_chemo &lt;- output_list[[i]]
  diff_chemo_sub &lt;- diff_chemo[sample(nrow(diff_chemo), nrow(diff_chemo)*0.1), ]
  
  a = ggplot(data = diff_chemo_sub, aes(x=diff)) + 
  geom_histogram(binwidth = 0.01) + 
  xlab(paste(&quot;DapB mRNA signal subtracted from&quot;, i, sep = &quot; &quot;)) + ggtitle(&quot;Raw distribution&quot;) + 
    theme(plot.title = element_text(hjust = 0.5, size = 35), 
          axis.title = element_text(size = 25), 
          axis.text =  element_text(size = 15))

  b = ggplot(data = diff_chemo_sub, aes(x=scaled_diff)) + 
  geom_histogram(binwidth = 0.1, aes(fill = 
                                       ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
  xlab(paste(&quot;DapB mRNA signal subtracted from&quot;, i, sep = &quot; &quot;)) + ggtitle(&quot;Scaled distribution&quot;) + 
    labs(fill = &quot;legend&quot;) +
    xlim(-5,7) +
    scale_fill_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;)) +
        theme(plot.title = element_text(hjust = 0.5, size = 35), 
          axis.title = element_text(size = 25), 
          legend.text = element_text(size = 20), 
          legend.title = element_text(size=20),
          axis.text =  element_text(size = 15))
  
  # significant cells defined by subtraction
  c = ggplot(data=diff_chemo_sub, aes(x=mean_negative_control, y=mean_chemokine)) + 
    geom_point(alpha=0.2, aes(col = 
                                ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
    scale_color_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;)) +
    xlim(0,5.5) + ylim(0,5.5) +
    ylab(paste(&quot;Mean expression of&quot;, i, sep=&quot; &quot;)) +
    xlab(&quot;Mean DapB mRNAexpression&quot;) +
    ggtitle(paste(&quot;DapB mRNA vs.&quot;, i, sep = &quot; &quot;)) + 
    theme(plot.title = element_text(hjust = 0.5, size = 35), 
          axis.title = element_text(size = 25), 
          legend.position = &quot;none&quot;,
          axis.text =  element_text(size = 15))
  #png(file = paste(&quot;~/Daniel_volume/Rout_RNA/chemokine_detection_method_comparison/&quot;,i,&quot;difference_distribution_BH_0.01.png&quot;, sep=&quot;_&quot;), height = 1000, width = 1600)
  grid.arrange(a,b,c, nrow = 3, ncol=1)
  #dev.off()
}</code></pre>
<pre><code>Warning: Removed 170 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-1.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 163 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-2.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 182 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-3.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 382 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-4.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 139 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-5.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 269 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-6.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 273 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-7.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 68 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<pre><code>Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-8.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 169 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-9.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 10 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-10.png" width="1344" style="display: block; margin: auto;" /></p>
<pre><code>Warning: Removed 187 rows containing non-finite values (stat_bin).

Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/different_chemokine_detection_results-11.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="basic-stats" class="section level1">
<h1>Basic Stats</h1>
<div id="basic-numbers-on-the-chemokine-expressing-cells" class="section level2">
<h2>Basic numbers on the chemokine expressing cells</h2>
<pre class="r"><code>chemokines &lt;- data.frame(colData(sce))
chemokines &lt;- chemokines[, chemokine_channels]

# calculate the amount of cells that are positive for 1, 2 and multiple combinations. exclude column containing the ids (12)
single_combinations = chemokines[rowSums(chemokines[,-1]) == 1,-1]

# number of single chemokines positive cells 
nrow(single_combinations)</code></pre>
<pre><code>[1] 45168</code></pre>
<pre class="r"><code>double_combinations = chemokines[rowSums(chemokines[,-1]) == 2,-1]
# number of single chemokines positive cells 
nrow(double_combinations)</code></pre>
<pre><code>[1] 8656</code></pre>
<pre class="r"><code>multiple_combinations = chemokines[rowSums(chemokines[,-1]) &gt;= 3,-1]
# number of cells that express 3 or more chemokines
nrow(multiple_combinations)</code></pre>
<pre><code>[1] 2427</code></pre>
<pre class="r"><code># number of double positives per chemokine
double_counts &lt;- colSums(double_combinations)

# frequency matrix and corrplot for frequency matrix
double_combinations[double_combinations == 0] &lt;- NA
count_matrix = psych::pairwiseCount(x=double_combinations)

# normalize the frequency matrix by the amount of double combinations that occur for each chemokine
frequency_matrix &lt;- count_matrix
for (i in colnames(count_matrix)){
  frequency_matrix[,i] &lt;- frequency_matrix[,i]/double_counts[i]
}</code></pre>
</div>
<div id="frequency-double-positive-cells" class="section level2">
<h2>Frequency double-positive cells</h2>
<p>The next plot shows the frequencies of all double positive cell occurences. e.g. of all T4_CXCL10 expressing cells that also express another chemokine more than 50% express T9_CXCL9.</p>
<pre class="r"><code>corrplot(frequency_matrix, is.corr = FALSE, tl.col = &#39;black&#39;, method = &#39;pie&#39;, type = &#39;full&#39;, 
         tl.srt = 45, tl.cex = 0.8, tl.offset = 0.5, cl.length = 2, cl.cex = 1, cl.align.text = &quot;l&quot;, cl.ratio = 0.3,
         diag=TRUE, order = &quot;hclust&quot;)</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/frequency_of_double_positive_chemokines-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="corrplot-of-frequency-matrix" class="section level2">
<h2>Corrplot of Frequency matrix</h2>
<p>Now we normalize the numbers of double positives by the numbers of all respective positive chemokines. this shows that usually between 20-40 percent of chemokine expressing cells are double positive expressors.</p>
<pre class="r"><code>single_counts &lt;- colSums(single_combinations)
frequency_matrix &lt;- count_matrix
for (i in colnames(count_matrix)){
  frequency_matrix[,i] &lt;- frequency_matrix[,i]/single_counts[i]
}

corrplot(frequency_matrix, is.corr = FALSE, tl.col = &#39;black&#39;, method = &#39;pie&#39;, type = &#39;full&#39;, 
         tl.srt = 45, tl.cex = 0.8, tl.offset = 0.5, cl.length = 2, cl.cex = 1, cl.align.text = &quot;l&quot;, cl.ratio = 0.3,
         cl.lim = c(0,1), diag=TRUE, order = &quot;hclust&quot;)</code></pre>
<p><img src="figure/05_RNA_chemokine_expressing_cells.rmd/freq_corrplot-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sce-object" class="section level1">
<h1>SCE object</h1>
<div id="add-data-to-sce-object" class="section level2">
<h2>Add data to SCE object</h2>
<pre class="r"><code># general chemokine producer tag for every cell (logical binary)
sce$chemokine &lt;- ifelse(rowSums(data.frame(colData(sce)[,chemokine_channels])) &gt; 0, TRUE, FALSE)

# rename colData entry names
idx &lt;- match(chemokine_channels, colnames(colData(sce)))
for(i in idx){
  colnames(colData(sce))[i] &lt;- strsplit(colnames(colData(sce))[i], split = &quot;_&quot;)[[1]][2]
}</code></pre>
</div>
<div id="add-expressor-info-and-colour_vector-with-control-samples" class="section level2">
<h2>Add expressor info and colour_vector (with control samples)</h2>
<pre class="r"><code>cur_df &lt;- colData(sce)
cur_df &lt;- as_tibble(cur_df)
cur_df &lt;- cur_df[,grepl(&quot;CCL|CXCL|Dap&quot;,colnames(cur_df))]

for(i in colnames(cur_df)){
  cur_df[[i]] &lt;- ifelse(cur_df[[i]]== 1,i,&quot;NA&quot;)
}

cur_df &lt;- cur_df %&gt;%
  unite(expressor,sep = &quot;_&quot;,na.rm =TRUE,remove=FALSE)

cur_df$expressor &lt;- gsub(&quot;NA_&quot;,&quot;&quot;,cur_df$expressor)
cur_df$expressor &lt;- gsub(&quot;_NA&quot;,&quot;&quot;,cur_df$expressor)

# summary table of all combinations
summary_cur_df &lt;- table(cur_df$expressor)

# order table according to abundance of combinations
summary_cur_df&lt;- summary_cur_df[order(-as.numeric(summary_cur_df))]

# combinations with more than 600 occurrences
targets &lt;- names(which(summary_cur_df &gt; 600))
targets &lt;- targets[!targets == &quot;NA&quot;]

# add expressor info to sce object
sce$expressor &lt;- cur_df$expressor

# add target names to metadata
metadata(sce)$chemokines_morethan600_withcontrol &lt;- targets</code></pre>
</div>
<div id="save-sce-object" class="section level2">
<h2>Save SCE object</h2>
<pre class="r"><code>saveRDS(object = sce,file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] colorRamps_2.3              tidyr_1.1.2                
 [3] gridExtra_2.3               pheatmap_1.0.12            
 [5] corrplot_0.84               ComplexHeatmap_2.4.3       
 [7] data.table_1.13.6           LSD_4.1-0                  
 [9] reshape2_1.4.4              CATALYST_1.12.2            
[11] scater_1.16.2               ggplot2_3.3.3              
[13] dplyr_1.0.2                 SingleCellExperiment_1.12.0
[15] SummarizedExperiment_1.20.0 Biobase_2.50.0             
[17] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
[19] IRanges_2.24.1              S4Vectors_0.28.1           
[21] BiocGenerics_0.36.0         MatrixGenerics_1.2.0       
[23] matrixStats_0.57.0          workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                circlize_0.4.12            
  [3] drc_3.0-1                   plyr_1.8.6                 
  [5] igraph_1.2.6                ConsensusClusterPlus_1.52.0
  [7] splines_4.0.3               flowCore_2.0.1             
  [9] BiocParallel_1.22.0         TH.data_1.0-10             
 [11] digest_0.6.27               htmltools_0.5.0            
 [13] viridis_0.5.1               magrittr_2.0.1             
 [15] CytoML_2.0.5                cluster_2.1.0              
 [17] openxlsx_4.2.3              RcppParallel_5.0.2         
 [19] sandwich_3.0-0              flowWorkspace_4.0.6        
 [21] cytolib_2.0.3               jpeg_0.1-8.1               
 [23] colorspace_2.0-0            ggrepel_0.9.0              
 [25] haven_2.3.1                 xfun_0.20                  
 [27] crayon_1.3.4                RCurl_1.98-1.2             
 [29] jsonlite_1.7.2              hexbin_1.28.2              
 [31] graph_1.66.0                survival_3.2-7             
 [33] zoo_1.8-8                   glue_1.4.2                 
 [35] gtable_0.3.0                nnls_1.4                   
 [37] zlibbioc_1.36.0             XVector_0.30.0             
 [39] GetoptLong_1.0.5            DelayedArray_0.16.0        
 [41] ggcyto_1.16.0               car_3.0-10                 
 [43] BiocSingular_1.4.0          Rgraphviz_2.32.0           
 [45] shape_1.4.5                 abind_1.4-5                
 [47] scales_1.1.1                mvtnorm_1.1-1              
 [49] Rcpp_1.0.5                  plotrix_3.7-8              
 [51] viridisLite_0.3.0           tmvnsim_1.0-2              
 [53] clue_0.3-58                 foreign_0.8-81             
 [55] rsvd_1.0.3                  FlowSOM_1.20.0             
 [57] tsne_0.1-3                  RColorBrewer_1.1-2         
 [59] ellipsis_0.3.1              farver_2.0.3               
 [61] pkgconfig_2.0.3             XML_3.99-0.5               
 [63] labeling_0.4.2              tidyselect_1.1.0           
 [65] rlang_0.4.10                later_1.1.0.1              
 [67] munsell_0.5.0               cellranger_1.1.0           
 [69] tools_4.0.3                 generics_0.1.0             
 [71] ggridges_0.5.3              evaluate_0.14              
 [73] stringr_1.4.0               yaml_2.2.1                 
 [75] knitr_1.30                  fs_1.5.0                   
 [77] zip_2.1.1                   purrr_0.3.4                
 [79] nlme_3.1-151                RBGL_1.64.0                
 [81] whisker_0.4                 xml2_1.3.2                 
 [83] compiler_4.0.3              rstudioapi_0.13            
 [85] beeswarm_0.2.3              curl_4.3                   
 [87] png_0.1-7                   tibble_3.0.4               
 [89] stringi_1.5.3               forcats_0.5.0              
 [91] lattice_0.20-41             Matrix_1.3-2               
 [93] psych_2.0.12                vctrs_0.3.6                
 [95] pillar_1.4.7                lifecycle_0.2.0            
 [97] GlobalOptions_0.1.2         BiocNeighbors_1.6.0        
 [99] cowplot_1.1.1               bitops_1.0-6               
[101] irlba_2.3.3                 httpuv_1.5.4               
[103] R6_2.5.0                    latticeExtra_0.6-29        
[105] promises_1.1.1              RProtoBufLib_2.0.0         
[107] rio_0.5.16                  vipor_0.4.5                
[109] codetools_0.2-18            MASS_7.3-53                
[111] gtools_3.8.2                rprojroot_2.0.2            
[113] rjson_0.2.20                withr_2.3.0                
[115] mnormt_2.0.2                multcomp_1.4-15            
[117] GenomeInfoDbData_1.2.4      hms_0.5.3                  
[119] ncdfFlow_2.34.0             rmarkdown_2.6              
[121] DelayedMatrixStats_1.10.1   carData_3.0-4              
[123] Rtsne_0.15                  git2r_0.28.0               
[125] base64enc_0.1-3             ggbeeswarm_0.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
