<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Daniel" />


<title>05 cytotoxic T cell quantification</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Score.html">Bcell Score</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_1.html">Supp-Figure 1</a>
    </li>
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="XX_hazard_ratio.html">Hazard Ratio Analysis</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">05 cytotoxic T cell quantification</h1>
<h4 class="author">Daniel</h4>
<h4 class="date">July 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>melanoma_publication_old_data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree2e443a5cda5affb9de9ca1eb678438325319a857targetblank2e443a5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/2e443a5cda5affb9de9ca1eb678438325319a857" target="_blank">2e443a5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree2e443a5cda5affb9de9ca1eb678438325319a857targetblank2e443a5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/2e443a5cda5affb9de9ca1eb678438325319a857" target="_blank">2e443a5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/._clinical metadata preparation.Rmd
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._protein_neutrophil.png
    Ignored:    output/._rna_neutrophil.png
    Ignored:    output/PSOCKclusterOut/
    Ignored:    output/bcell_grouping.png
    Ignored:    output/dysfunction_correlation.pdf

Untracked files:
    Untracked:  analysis/00_prepare_clinical_dat.rmd
    Untracked:  code/helper_functions/findMilieu.R
    Untracked:  code/helper_functions/findPatch.R

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/01_Protein_read_data.rmd
    Modified:   analysis/01_RNA_read_data.rmd
    Modified:   analysis/02_Protein_annotations.rmd
    Modified:   analysis/02_RNA_annotations.rmd
    Modified:   analysis/03_Protein_quality_control.rmd
    Modified:   analysis/03_RNA_quality_control.rmd
    Modified:   analysis/04_1_Protein_celltype_classification.rmd
    Modified:   analysis/04_1_RNA_celltype_classification.rmd
    Modified:   analysis/04_2_RNA_classification_subclustering.rmd
    Modified:   analysis/04_2_protein_classification_subclustering.rmd
    Modified:   analysis/05_RNA_chemokine_expressing_cells.rmd
    Modified:   analysis/06_RNA_chemokine_patch_detection.rmd
    Modified:   analysis/07_TCF7_PD1_gating.rmd
    Modified:   analysis/08_color_vectors.rmd
    Modified:   analysis/09_Tcell_Score.Rmd
    Modified:   analysis/10_Dysfunction_Score.rmd
    Modified:   analysis/11_Bcell_Score.Rmd
    Modified:   analysis/Figure_1.rmd
    Modified:   analysis/Figure_2.rmd
    Modified:   analysis/Figure_3.rmd
    Modified:   analysis/Figure_4.rmd
    Modified:   analysis/Figure_5.rmd
    Modified:   analysis/Summary_Statistics.rmd
    Modified:   analysis/Supp-Figure_1.rmd
    Modified:   analysis/Supp-Figure_2.rmd
    Modified:   analysis/Supp-Figure_3.rmd
    Modified:   analysis/Supp-Figure_4.rmd
    Modified:   analysis/Supp-Figure_5.rmd
    Modified:   analysis/XX_hazard_ratio.rmd
    Deleted:    code/findPackages.R
    Deleted:    code/helper_functions/findClusters.R
    Deleted:    code/helper_functions/findCommunity.R
    Deleted:    code/helper_functions/getCellCount.R
    Deleted:    code/helper_functions/plotBarFracCluster.R
    Deleted:    code/helper_functions/plotCellFrac.R
    Deleted:    code/helper_functions/plotCellFracGroups.R
    Deleted:    code/helper_functions/plotCellFracGroupsSubset.R
    Modified:   code/helper_functions/scatter_function.R
    Modified:   code/helper_functions/validityChecks.R
    Deleted:    data/mask_comparison/20190809_ZTMA256.1_slide2_TH_s1_p1_r15_a15_ac_full.tiff
    Deleted:    data/mask_comparison/20190809_ZTMA256.1_slide2_TH_s1_p1_r15_a15_ac_ilastik_s2_Probabilities_equalized_cellmask.tiff
    Deleted:    data/mask_comparison/20191023_ZTMA256.1_slide3_TH_s0_p10_r4_a4_ac_full.tiff
    Deleted:    data/mask_comparison/20191023_ZTMA256.1_slide3_TH_s0_p10_r4_a4_ac_ilastik_s2_Probabilities_equalized_cellmask.tiff

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/09_Tcell_Score.Rmd</code>) and HTML (<code>docs/09_Tcell_Score.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/2e443a5cda5affb9de9ca1eb678438325319a857/analysis/09_Tcell_Score.Rmd" target="_blank">2e443a5</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
remove files that are not needed
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message= FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())</code></pre>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions/&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions//calculateSummary.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//censor_dat.R
value   ?                                  
visible FALSE                              
        code/helper_functions//detect_mRNA_expression.R
value   ?                                              
visible FALSE                                          
        code/helper_functions//DistanceToClusterCenter.R
value   ?                                               
visible FALSE                                           
        code/helper_functions//findMilieu.R code/helper_functions//findPatch.R
value   ?                                   ?                                 
visible FALSE                               FALSE                             
        code/helper_functions//getInfoFromString.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//getSpotnumber.R
value   ?                                     
visible FALSE                                 
        code/helper_functions//plotCellCounts.R
value   ?                                      
visible FALSE                                  
        code/helper_functions//plotCellFractions.R
value   ?                                         
visible FALSE                                     
        code/helper_functions//plotDist.R
value   ?                                
visible FALSE                            
        code/helper_functions//scatter_function.R
value   ?                                        
visible FALSE                                    
        code/helper_functions//sceChecks.R
value   ?                                 
visible FALSE                             
        code/helper_functions//validityChecks.R
value   ?                                      
visible FALSE                                  </code></pre>
<pre class="r"><code>library(LSD)
library(SingleCellExperiment)
library(ggplot2)
library(scater)
library(viridis)
library(igraph)
library(CATALYST)
library(reshape2)
library(cowplot)
library(ggridges)
library(pheatmap)
library(tidyverse)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>Load Data</h2>
</div>
<div id="cytotoxic-t-cell-scoring" class="section level1">
<h1>Cytotoxic T cell Scoring</h1>
<div id="t-cell-density" class="section level2">
<h2>T Cell Density</h2>
<p>We want to calculate the T cell density. we calculate this per mm2. We use the count for cytotxic T cells as defined under “03_cell_type_definition” script</p>
<pre class="r"><code>cur_df &lt;- data.frame(celltype = sce_prot$celltype,
                             ImageNumber = sce_prot$ImageNumber)

# count cell types per images
cellcount &lt;- (t(table(cur_df)))

# here we get the imagesize from the image metadata
im_size &lt;- (image_mat$Height_cellmask * image_mat$Width_cellmask)/1000000

# data frame
cellcount &lt;- data.frame(cellcount)

# we calculate the density for each celltype for 1 mm2
cellcount$density &lt;- cellcount$Freq/im_size[cellcount$ImageNumber]
cellcount &lt;- cellcount[cellcount$celltype == &quot;Tcytotoxic&quot;,]

# there are roughly 60 images with 50 or less cytotoxic T cells
hist(cellcount$density,breaks = 300)</code></pre>
<p><img src="figure/09_Tcell_Score.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># add cyotoxic T cell density to sce
cellcount$ImageNumber &lt;- as.integer(cellcount$ImageNumber)
cur_sce &lt;- data.frame(colData(sce_prot))
cur_sce &lt;- left_join(cur_sce, cellcount[,c(&quot;ImageNumber&quot;, &quot;density&quot;)])
sce_prot$cyotoxic_density_image &lt;- cur_sce$density</code></pre>
</div>
<div id="t-cell-density-score-per-image" class="section level2">
<h2>T cell density score per image</h2>
<ul>
<li>absent: up to 40 cytotoxic T cells/mm2</li>
<li>low: 40-100 cytotoxic T cells/mm2</li>
<li>med: 100-400 cytotoxic T cells/mm2</li>
<li>high: &gt;400 cytotoxic T cells/mm2</li>
</ul>
<pre class="r"><code># define a vector with all ImgeNumbers
T_density_scores &lt;- c(1:length(unique(cellcount$ImageNumber)))

T_density_scores  &lt;- rep(&quot;unassigned&quot;,length(unique(cellcount$ImageNumber)))

T_absent &lt;- which(cellcount$density &lt;= 40)
T_low &lt;- which(cellcount$density &gt; 40 &amp; cellcount[cellcount$celltype == &quot;Tcytotoxic&quot;,]$density &lt;= 100)
T_med &lt;- which(cellcount$density &gt; 100 &amp; cellcount$density &lt;= 400)
T_high &lt;- which(cellcount$density &gt; 400)

T_density_scores[T_absent] &lt;- &quot;absent&quot;
T_density_scores[T_low] &lt;- &quot;low&quot;
T_density_scores[T_med] &lt;- &quot;med&quot;
T_density_scores[T_high] &lt;- &quot;high&quot;

# now we add the information to the single cell experiment
sce_prot$Tcell_density_score_image &lt;- T_density_scores[sce_prot$ImageNumber]
sce_prot$Tcell_density_score_image &lt;- factor(sce_prot$Tcell_density_score_image, levels = c(&quot;absent&quot;, &quot;low&quot;, &quot;med&quot;, &quot;high&quot;))

# number of samples per group
data.frame(colData(sce_prot)) %&gt;%
  distinct(Description, .keep_all = T) %&gt;%
  group_by(Tcell_density_score_image) %&gt;%
  summarise(n=n())</code></pre>
<pre><code># A tibble: 4 x 2
  Tcell_density_score_image     n
  &lt;fct&gt;                     &lt;int&gt;
1 absent                       51
2 low                          25
3 med                          46
4 high                         45</code></pre>
</div>
<div id="e_i_d-score-compared-to-t_frac_score_per_imagenumber" class="section level2">
<h2>E_I_D score compared to T_frac_score_per_ImageNumber</h2>
<pre class="r"><code>cur_df &lt;- data.frame(T_density = sce_prot$Tcell_density_score_image,
                     E_I_D = sce_prot$E_I_D)

table(cur_df)</code></pre>
<pre><code>         E_I_D
T_density      D      E    E/D      I    I/E
   absent  36924 116043  18583  50857  30611
   low      8541  53869   4971  25955  34393
   med         0  77110      0  94479  69812
   high        0  59707      0 142200 137295</code></pre>
</div>
<div id="add-scores-to-rna-data-set" class="section level2">
<h2>Add Scores to RNA data set</h2>
<pre class="r"><code>sce_rna$infiltration &lt;- NULL
sce_rna$T_frac_score_per_BlockID &lt;- NULL
sce_rna$T_frac_score_per_ImageNumber &lt;- NULL
sce_rna$T_frac_score_per_PatientID &lt;- NULL
sce_rna$cyotoxic_density_image &lt;- NULL

description_data &lt;- data.frame(colData(sce_prot)) %&gt;%
  distinct(Description, .keep_all = TRUE)

col_rna &lt;- data.frame(colData(sce_rna))

# left_join
col_rna &lt;- left_join(col_rna, description_data[,c(&quot;Description&quot;, &quot;Tcell_density_score_image&quot;, &quot;cyotoxic_density_image&quot;)])

# add to sce (attention: cytotoxic density is calculated on protein data set!)
sce_rna$Tcell_density_score_image &lt;- col_rna$Tcell_density_score_image
sce_rna$cyotoxic_density_image &lt;- col_rna$density</code></pre>
</div>
<div id="save-updated-sce" class="section level2">
<h2>Save updated SCE</h2>
<pre class="r"><code>saveRDS(sce_prot,file = &quot;data/data_for_analysis/sce_protein.rds&quot;)
saveRDS(sce_rna,file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] forcats_0.5.0               stringr_1.4.0              
 [3] dplyr_1.0.2                 purrr_0.3.4                
 [5] readr_1.4.0                 tidyr_1.1.2                
 [7] tibble_3.0.4                tidyverse_1.3.0            
 [9] pheatmap_1.0.12             ggridges_0.5.3             
[11] cowplot_1.1.1               reshape2_1.4.4             
[13] CATALYST_1.12.2             igraph_1.2.6               
[15] viridis_0.5.1               viridisLite_0.3.0          
[17] scater_1.16.2               ggplot2_3.3.3              
[19] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
[21] Biobase_2.50.0              GenomicRanges_1.42.0       
[23] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[25] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[27] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[29] LSD_4.1-0                   workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.2.1             readxl_1.3.1               
  [3] circlize_0.4.12             drc_3.0-1                  
  [5] plyr_1.8.6                  ConsensusClusterPlus_1.52.0
  [7] splines_4.0.3               flowCore_2.0.1             
  [9] BiocParallel_1.22.0         TH.data_1.0-10             
 [11] digest_0.6.27               htmltools_0.5.0            
 [13] fansi_0.4.1                 magrittr_2.0.1             
 [15] CytoML_2.0.5                cluster_2.1.0              
 [17] openxlsx_4.2.3              ComplexHeatmap_2.4.3       
 [19] modelr_0.1.8                RcppParallel_5.0.2         
 [21] sandwich_3.0-0              flowWorkspace_4.0.6        
 [23] cytolib_2.0.3               jpeg_0.1-8.1               
 [25] colorspace_2.0-0            rvest_0.3.6                
 [27] ggrepel_0.9.0               haven_2.3.1                
 [29] xfun_0.20                   crayon_1.3.4               
 [31] RCurl_1.98-1.2              jsonlite_1.7.2             
 [33] hexbin_1.28.2               graph_1.66.0               
 [35] survival_3.2-7              zoo_1.8-8                  
 [37] glue_1.4.2                  gtable_0.3.0               
 [39] nnls_1.4                    zlibbioc_1.36.0            
 [41] XVector_0.30.0              GetoptLong_1.0.5           
 [43] DelayedArray_0.16.0         ggcyto_1.16.0              
 [45] car_3.0-10                  BiocSingular_1.4.0         
 [47] Rgraphviz_2.32.0            shape_1.4.5                
 [49] abind_1.4-5                 scales_1.1.1               
 [51] mvtnorm_1.1-1               DBI_1.1.0                  
 [53] Rcpp_1.0.5                  plotrix_3.7-8              
 [55] clue_0.3-58                 foreign_0.8-81             
 [57] rsvd_1.0.3                  FlowSOM_1.20.0             
 [59] tsne_0.1-3                  httr_1.4.2                 
 [61] RColorBrewer_1.1-2          ellipsis_0.3.1             
 [63] pkgconfig_2.0.3             XML_3.99-0.5               
 [65] dbplyr_2.0.0                utf8_1.1.4                 
 [67] tidyselect_1.1.0            rlang_0.4.10               
 [69] later_1.1.0.1               munsell_0.5.0              
 [71] cellranger_1.1.0            tools_4.0.3                
 [73] cli_2.2.0                   generics_0.1.0             
 [75] broom_0.7.3                 evaluate_0.14              
 [77] yaml_2.2.1                  knitr_1.30                 
 [79] fs_1.5.0                    zip_2.1.1                  
 [81] RBGL_1.64.0                 whisker_0.4                
 [83] xml2_1.3.2                  compiler_4.0.3             
 [85] rstudioapi_0.13             beeswarm_0.2.3             
 [87] curl_4.3                    png_0.1-7                  
 [89] reprex_0.3.0                stringi_1.5.3              
 [91] lattice_0.20-41             Matrix_1.3-2               
 [93] vctrs_0.3.6                 pillar_1.4.7               
 [95] lifecycle_0.2.0             GlobalOptions_0.1.2        
 [97] BiocNeighbors_1.6.0         data.table_1.13.6          
 [99] bitops_1.0-6                irlba_2.3.3                
[101] httpuv_1.5.4                R6_2.5.0                   
[103] latticeExtra_0.6-29         promises_1.1.1             
[105] gridExtra_2.3               RProtoBufLib_2.0.0         
[107] rio_0.5.16                  vipor_0.4.5                
[109] codetools_0.2-18            assertthat_0.2.1           
[111] MASS_7.3-53                 gtools_3.8.2               
[113] rprojroot_2.0.2             rjson_0.2.20               
[115] withr_2.3.0                 multcomp_1.4-15            
[117] GenomeInfoDbData_1.2.4      hms_0.5.3                  
[119] ncdfFlow_2.34.0             grid_4.0.3                 
[121] rmarkdown_2.6               DelayedMatrixStats_1.10.1  
[123] carData_3.0-4               Rtsne_0.15                 
[125] git2r_0.28.0                lubridate_1.7.9.2          
[127] base64enc_0.1-3             ggbeeswarm_0.6.0           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
