<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tobias Hoch and Daniel Schulz" />

<meta name="date" content="2021-12-10" />

<title>Supplementary Figure 9</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Patch_Detection.html">Bcell Patch Detection</a>
    </li>
    <li>
      <a href="12_Bcell_Score.html">Bcell Score</a>
    </li>
    <li>
      <a href="13_Dysfunction_Stain.html">Dysfunction Validation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplementary Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
    <li>
      <a href="Supp-Figure_6.html">Supp-Figure 6</a>
    </li>
    <li>
      <a href="Supp-Figure_7.html">Supp-Figure 7</a>
    </li>
    <li>
      <a href="Supp-Figure_8.html">Supp-Figure 8</a>
    </li>
    <li>
      <a href="Supp-Figure_9.html">Supp-Figure 9</a>
    </li>
    <li>
      <a href="Supp-Figure_10.html">Supp-Figure 10</a>
    </li>
    <li>
      <a href="Supp-Figure_11.html">Supp-Figure 11</a>
    </li>
    <li>
      <a href="Supp-Figure_12.html">Supp-Figure 12</a>
    </li>
    <li>
      <a href="Supp-Figure_13.html">Supp-Figure 13</a>
    </li>
    <li>
      <a href="Supp-Figure_14.html">Supp-Figure 14</a>
    </li>
    <li>
      <a href="Table_ClinicalData.html">Clinical Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Supplementary Figure 9</h1>
<h4 class="author">Tobias Hoch and Daniel Schulz</h4>
<h4 class="date">2021-12-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MelanomaIMC/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtree1affd7bda3b8da020e017c0a21251bf17844d32dtargetblank1affd7ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/1affd7bda3b8da020e017c0a21251bf17844d32d" target="_blank">1affd7b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtree1affd7bda3b8da020e017c0a21251bf17844d32dtargetblank1affd7ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/1affd7bda3b8da020e017c0a21251bf17844d32d" target="_blank">1affd7b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    Table_S4.csv
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/Supp-Figure_6.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Supp-Figure_9.rmd</code>) and HTML (<code>docs/Supp-Figure_9.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/64e5fde08d082d625c8c93c1d31d949814f2fa47/analysis/Supp-Figure_9.rmd" target="_blank">64e5fde</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-16
</td>
<td>
change order and naming of supp fig files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/b20b6fbc2a2e0ee6e1e3b73f3ace0d3c11cf084d/analysis/Supp-Figure_9.rmd" target="_blank">b20b6fb</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-02
</td>
<td>
update code for Supp Figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/3da15db29f412e8bbd5b3c941ecc9bab416135f9/analysis/Supp-Figure_9.rmd" target="_blank">3da15db</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-11-24
</td>
<td>
changes for revision
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This script generates plots for Supplementary Figure 9.</p>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions/calculateSummary.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/censor_dat.R
value   ?                                 
visible FALSE                             
        code/helper_functions/detect_mRNA_expression.R
value   ?                                             
visible FALSE                                         
        code/helper_functions/DistanceToClusterCenter.R
value   ?                                              
visible FALSE                                          
        code/helper_functions/findMilieu.R code/helper_functions/findPatch.R
value   ?                                  ?                                
visible FALSE                              FALSE                            
        code/helper_functions/getInfoFromString.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/getSpotnumber.R
value   ?                                    
visible FALSE                                
        code/helper_functions/plotCellCounts.R
value   ?                                     
visible FALSE                                 
        code/helper_functions/plotCellFractions.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/plotDist.R code/helper_functions/read_Data.R
value   ?                                ?                                
visible FALSE                            FALSE                            
        code/helper_functions/scatter_function.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/sceChecks.R
value   ?                                
visible FALSE                            
        code/helper_functions/validityChecks.R
value   ?                                     
visible FALSE                                 </code></pre>
<pre class="r"><code>library(SingleCellExperiment)
library(dplyr)
library(ggplot2)
library(ggbeeswarm)
library(tidyr)
library(scater)
library(dittoSeq)
library(gridExtra)
library(cowplot)
library(data.table)
library(ggpmisc)
library(ggpubr)
library(ComplexHeatmap)
library(rstatix)
library(dendextend)
library(parallel)
library(neighbouRhood)
library(unix)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>Load Data</h2>
<pre class="r"><code># SCE object
sce_rna = readRDS(file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)
sce_prot = readRDS(file = &quot;data/data_for_analysis/sce_protein.rds&quot;)

sce_rna &lt;- sce_rna[,sce_rna$Location != &quot;CTRL&quot;]
sce_prot &lt;- sce_prot[,sce_prot$Location != &quot;CTRL&quot;]

# meta data
dat_relation = fread(file = &quot;data/data_for_analysis/protein/Object relationships.csv&quot;,stringsAsFactors = FALSE)
dat_relation_rna = fread(file = &quot;data/data_for_analysis/RNA/Object relationships.csv&quot;,stringsAsFactors = FALSE)

targets &lt;- metadata(sce_rna)$chemokines_morethan600_withcontrol</code></pre>
</div>
<div id="prepare-relation-table" class="section level2">
<h2>Prepare Relation Table</h2>
<pre class="r"><code># prepare data 
dat_relation$cellID_first &lt;- paste(&quot;RNA&quot;, paste(dat_relation$`First Image Number`, dat_relation$`First Object Number`, sep = &quot;_&quot;), sep = &quot;_&quot;)
dat_relation$cellID_second &lt;- paste(&quot;RNA&quot;, paste(dat_relation$`Second Image Number`, dat_relation$`Second Object Number`, sep = &quot;_&quot;), sep = &quot;_&quot;)</code></pre>
</div>
</div>
<div id="supp-figure-9a" class="section level1">
<h1>Supp Figure 9A</h1>
<div id="expressor-in-tcell-groups" class="section level2">
<h2>Expressor in Tcell groups</h2>
<pre class="r"><code>targets &lt;- metadata(sce_rna)$chemokines_morethan600_withcontrol

frac &lt;- data.frame(colData(sce_rna)) %&gt;%
  filter(Location != &quot;CTRL&quot;) %&gt;%
  group_by(Description, Tcell_density_score_image, expressor) %&gt;%
  summarise(n=n()) %&gt;%
  mutate(fraction = n / sum(n)) %&gt;%
  filter(expressor %in% targets) %&gt;%
  reshape2::dcast(Description + Tcell_density_score_image ~ expressor, value.var = &quot;fraction&quot;, fill = 0) %&gt;%
  reshape2::melt(id.vars = c(&quot;Description&quot;, &quot;Tcell_density_score_image&quot;), variable.name = &quot;expressor&quot;, value.name = &quot;fraction&quot;)

stat.test &lt;- frac %&gt;%
  group_by(expressor) %&gt;%
  kruskal_test(data = ., fraction ~ Tcell_density_score_image) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  add_significance(&quot;p.adj&quot;,cutpoints = c(0, 1e-04, 0.001, 0.01, 0.1, 1)) %&gt;%
  arrange(p.adj) %&gt;%
  mutate(group1 = expressor, group2 = expressor) %&gt;%
  add_x_position()

frac$expressor &lt;- factor(frac$expressor, levels = stat.test$expressor)

ggplot(frac, aes(x=expressor, y = fraction)) + 
  geom_boxplot(alpha=.75, outlier.size = 0.5, aes(fill = Tcell_density_score_image)) +
  stat_pvalue_manual(x = &quot;group1&quot;, y.position = 0.055, stat.test, size = 4) +
  scale_color_discrete(guide = FALSE) +
  theme_bw() +
  theme(text = element_text(size = 15),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  guides(fill=guide_legend(title=&quot;T cell Score&quot;)) +
  xlab(&quot;&quot;) + 
  ylab(&quot;Fractions&quot;)  +
  coord_cartesian(ylim = c(0,0.06))</code></pre>
<pre><code>Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
use `guide = &quot;none&quot;` instead.</code></pre>
<embed src="figure/Supp-Figure_9.rmd/Supp_Fig_9A_expressor_cd8_grouping-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="supp-figure-9b" class="section level1">
<h1>Supp Figure 9B</h1>
<div id="grouping-from-figure-1e" class="section level2">
<h2>Grouping from Figure 1E</h2>
<pre class="r"><code># Create table with celltype fractions
cur_df &lt;- data.frame(celltype = sce_prot$celltype,
                     Description = sce_prot$Description,
                     Location = sce_prot$Location)

# remove control samples 
cur_df &lt;- cur_df %&gt;%
  filter(Location != &quot;CTRL&quot;) %&gt;%
  group_by(Description, celltype) %&gt;%
  summarise(n=n()) %&gt;% 
  group_by(Description) %&gt;%
  mutate(fraction = n / sum(n)) %&gt;%
  reshape2::dcast(Description ~ celltype, value.var = &quot;fraction&quot;, fill=0)

matrixrownames &lt;- cur_df$Description 

# now we create a matrix from the data and cluster the data based on the cell fractions
hm_dat = as.matrix(cur_df[,-1])
rownames(hm_dat) &lt;- as.character(matrixrownames)

# calculate distance and then cluster images based on cluster fraction
dd &lt;- dist((hm_dat), method = &quot;euclidean&quot;)
hc &lt;- hclust(dd, method = &quot;ward.D2&quot;)
row_sorted &lt;- hc$labels

dend &lt;- as.dendrogram(hc)
clusters &lt;- data.frame(cutree(dend, k=4)) #### order_clusters_as_data = FALSE?? 
clusters_1E &lt;- color_branches(dend, k = 4, col = c(&quot;gray50&quot;, &quot;blue&quot;, &quot;green&quot;, &quot;red&quot;), groupLabels = TRUE)

# get labels from dend
dend_labels &lt;- clusters_1E %&gt;%
  labels()

# change colnames
colnames(clusters) &lt;- &quot;dend_cluster&quot;
clusters$Description &lt;- rownames(clusters)

# same orientation as in 1E
clusters &lt;- clusters[match(dend_labels, clusters$Description),]

# change cluster names
clusters$cluster_name &lt;- &quot;&quot;
clusters[clusters$dend_cluster == 3,]$cluster_name &lt;- &quot;Grey Branch&quot; 
clusters[clusters$dend_cluster == 4,]$cluster_name &lt;- &quot;Blue Branch&quot; 
clusters[clusters$dend_cluster == 2,]$cluster_name &lt;- &quot;Green Branch&quot; 
clusters[clusters$dend_cluster == 1,]$cluster_name &lt;- &quot;Red Branch&quot; 

# add cluster to sce_rna object
all_dat &lt;- data.frame(colData(sce_rna))[,c(&quot;Description&quot;, &quot;ImageNumber&quot;)]
all_dat &lt;- left_join(all_dat, clusters)
sce_rna$cluster_name &lt;- as.character(all_dat$cluster_name)

sce_rna$cluster_name &lt;- factor(sce_rna$cluster_name, levels = c(&quot;Red Branch&quot;, &quot;Green Branch&quot;, &quot;Blue Branch&quot;, &quot;Grey Branch&quot;))


targets &lt;- metadata(sce_rna)$chemokines_morethan600_withcontrol

frac &lt;- data.frame(colData(sce_rna)) %&gt;%
  filter(Location != &quot;CTRL&quot;) %&gt;%
  group_by(Description, cluster_name, expressor) %&gt;%
  summarise(n=n()) %&gt;%
  mutate(fraction = n / sum(n)) %&gt;%
  filter(expressor %in% targets) %&gt;%
  reshape2::dcast(Description + cluster_name ~ expressor, value.var = &quot;fraction&quot;, fill = 0) %&gt;%
  reshape2::melt(id.vars = c(&quot;Description&quot;, &quot;cluster_name&quot;), variable.name = &quot;expressor&quot;, value.name = &quot;fraction&quot;)

stat.test &lt;- frac %&gt;%
  group_by(expressor) %&gt;%
  kruskal_test(data = ., fraction ~ cluster_name) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  add_significance(&quot;p.adj&quot;,cutpoints = c(0, 1e-04, 0.001, 0.01, 0.1, 1)) %&gt;%
  arrange(p.adj) %&gt;%
  mutate(group1 = expressor, group2 = expressor) %&gt;%
  add_x_position()

frac$expressor &lt;- factor(frac$expressor, levels = stat.test$expressor)

ggplot(frac, aes(x=expressor, y = fraction)) + 
  geom_boxplot(alpha=.75, outlier.size = 0.5, aes(fill = cluster_name)) +
  stat_pvalue_manual(x = &quot;group1&quot;, y.position = 0.055, stat.test, size = 4) +
  scale_color_discrete(guide = FALSE) +
  theme_bw() +
  theme(text = element_text(size = 15),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1)) + 
  guides(fill=guide_legend(title=&quot;Dendrogram Cluster&quot;)) +
  xlab(&quot;&quot;) + 
  scale_fill_manual(values=c(&quot;red&quot;, &quot;green&quot;, &quot;blue&quot;, &quot;grey&quot;)) +
  ylab(&quot;Fractions&quot;)  +
  coord_cartesian(ylim = c(0,0.06))</code></pre>
<pre><code>Warning: It is deprecated to specify `guide = FALSE` to remove a guide. Please
use `guide = &quot;none&quot;` instead.</code></pre>
<embed src="figure/Supp-Figure_9.rmd/Supp_Fig_9B_expressor_dend_cluster-1.pdf" width="1344" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="supp-figure-9c" class="section level1">
<h1>Supp Figure 9C</h1>
<div id="cp-output" class="section level2">
<h2>CP output</h2>
<pre class="r"><code># Load and prepare
dat_cells = fread(file = &quot;data/data_for_analysis/rna/cell.csv&quot;,stringsAsFactors = FALSE)
dat_relation = fread(file = &quot;data/data_for_analysis/rna/Object relationships.csv&quot;,stringsAsFactors = FALSE)

# Number of cores used for multicore:
if(detectCores() &gt;= 12){
  ncores = round(detectCores()/1.25,0)
}
if(detectCores() &gt; 1 &amp; detectCores() &lt; 12){
  ncores = round(detectCores()/2,0)
}
if(detectCores() == 1){
  ncores = 1
}

n_perm = 100 </code></pre>
</div>
<div id="start-the-analysis" class="section level2">
<h2>Start the analysis</h2>
<pre class="r"><code>start = Sys.time()
cur_sce &lt;- as.data.frame(colData(sce_rna))

# add same cellID to dat_cells as in sce object
dat_cells$cellID &lt;- paste(&quot;RNA_&quot;, paste(dat_cells$ImageNumber, dat_cells$ObjectNumber, sep = &quot;_&quot;), sep = &quot;&quot;)

image_df &lt;- data.frame()
gc()</code></pre>
<pre><code>            used   (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells   9388034  501.4   14603417  780.0  14603417  780.0
Vcells 747115554 5700.1 1086952781 8292.8 905727318 6910.2</code></pre>
<pre class="r"><code>rlimit_as(Inf)</code></pre>
<pre><code>$cur
[1] Inf

$max
[1] Inf</code></pre>
<pre class="r"><code>for(size in c(2,3,4,5,6)) {
  images &lt;- data.frame()
  for(i in colnames(cur_sce[,grepl(&quot;CCL|CXCL&quot;,colnames(cur_sce))])){
    # add chemokine info to celltype
    sce_info &lt;- cur_sce[,c(&quot;cellID&quot;, i , &quot;Description&quot;)]
    
    # add celltype information
    dat_cells_tmp &lt;- left_join(as.data.frame(dat_cells), sce_info, by = &quot;cellID&quot;)
    
    #assign labels and groups
    dat_cells_tmp$label &lt;- dat_cells_tmp[,i]
    dat_cells_tmp$group &lt;- dat_cells_tmp$Description
    dat_cells_tmp &lt;- as.data.table(dat_cells_tmp)
    
    # subset dat_relation and dat_cells
    dat_cells_sub &lt;- dat_cells_tmp
    dat_relation_sub &lt;- dat_relation
    
    # Prepare the data
    d = neighbouRhood::prepare_tables(dat_cells_sub, dat_relation_sub)
    
    # Calculate the baseline statistics
    dat_baseline = neighbouRhood::apply_labels(d[[1]], d[[2]]) %&gt;%
      neighbouRhood::aggregate_classic_patch(., patch_size = size)
    
    # Calculate the permutation statistics
    # This will run the test using parallel computing. The name of the idcol does actually not matter.
    
    set.seed(12312)
    dat_perm = rbindlist(mclapply(1:n_perm, function(x){
      dat_labels = neighbouRhood::shuffle_labels(d[[1]])
      neighbouRhood::apply_labels(dat_labels, d[[2]]) %&gt;%
        neighbouRhood::aggregate_classic_patch(., patch_size = size)
    },mc.cores = ncores
    ), idcol = &#39;run&#39;)
    
    # calc p values
    dat_p &lt;- neighbouRhood::calc_p_vals(dat_baseline, dat_perm, n_perm = n_perm, p_tresh = 0.01) 
    
    # select interactions between chemokine+ cells
    dat_p$interaction &lt;- paste(dat_p$FirstLabel, dat_p$SecondLabel, sep = &quot;_&quot;)
    
    dat_p_wide &lt;- dat_p %&gt;%
      reshape2::dcast(group ~ interaction, value.var = &quot;sigval&quot;, fill = 0) %&gt;%
      select(group, `1_1`)
    
    summary &lt;- as.data.frame(dat_p_wide) %&gt;%
      group_by(`1_1`) %&gt;%
      summarise(n=n(),.groups = &#39;drop&#39;) %&gt;%
      ungroup() %&gt;%
      mutate(percentage_sig = (n/sum(n)) * 100)
    
    images &lt;- rbind(images, cbind(summary[1,], i))
    gc()
  }
  
  # calculate percentage of images with significant patches
  images$percentage_sig &lt;- 100 - images$percentage_sig
  images$patch_size &lt;- size
  images &lt;- select(images, percentage_sig, i, patch_size)
  colnames(images) &lt;- c(&quot;significant_images&quot;, &quot;chemokine&quot;, &quot;patch_size&quot;)
  
  # add to data.frame
  image_df &lt;- rbind(image_df, images)
}
end = Sys.time()

print(end-start)</code></pre>
<pre><code>Time difference of 27.89348 mins</code></pre>
</div>
<div id="visualize" class="section level2">
<h2>Visualize</h2>
<pre class="r"><code>dat &lt;- image_df %&gt;%
  reshape2::dcast(chemokine ~ patch_size, value.var = &quot;significant_images&quot;, fill = 0)

rownames(dat) &lt;- dat$chemokine
dat$chemokine &lt;- NULL

m &lt;- t(as.matrix(dat))

col_fun = viridis::inferno(100)

Heatmap(m,
        cluster_rows = FALSE,
        col = col_fun,
        column_title = &quot;Self-Interaction&quot;,
        column_title_side = &quot;bottom&quot;,
        show_row_names = TRUE,
        cell_fun = function(j, i, x, y, width, height, fill) {
          grid.text(sprintf(&quot;%.1f&quot;, m[i, j]), x, y, gp = gpar(fontsize = 15, col = &quot;grey&quot;))
          },
        heatmap_legend_param = list(
          title = &quot;% Significant\nImages&quot;, at = c(0, 10, 20, 30, 40, 50),
          labels = c(&quot;0%&quot;, &quot;10%&quot;, &quot;20%&quot;, &quot;30%&quot;,&quot;40%&quot;, &quot;50%&quot;)),
        row_title = &quot;Motif Size&quot;,
        row_names_side = &quot;left&quot;,
        width = unit(15, &quot;cm&quot;),
        height = unit(8, &quot;cm&quot;))</code></pre>
<embed src="figure/Supp-Figure_9.rmd/Supp_Fig_9C_patch_analysis-1.pdf" width="768" style="display: block; margin: auto;" type="application/pdf" />
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  grid      stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] unix_1.5.4                  neighbouRhood_0.4          
 [3] magrittr_2.0.2              dtplyr_1.2.1               
 [5] dendextend_1.15.2           rstatix_0.7.0              
 [7] ComplexHeatmap_2.10.0       ggpubr_0.4.0               
 [9] ggpmisc_0.4.5               ggpp_0.4.3                 
[11] data.table_1.14.2           cowplot_1.1.1              
[13] gridExtra_2.3               dittoSeq_1.6.0             
[15] scater_1.22.0               scuttle_1.4.0              
[17] tidyr_1.2.0                 ggbeeswarm_0.6.0           
[19] ggplot2_3.3.5               SingleCellExperiment_1.16.0
[21] SummarizedExperiment_1.24.0 Biobase_2.54.0             
[23] GenomicRanges_1.46.1        GenomeInfoDb_1.30.1        
[25] IRanges_2.28.0              S4Vectors_0.32.3           
[27] BiocGenerics_0.40.0         MatrixGenerics_1.6.0       
[29] matrixStats_0.61.0          dplyr_1.0.7                
[31] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] backports_1.4.1           circlize_0.4.13          
  [3] plyr_1.8.6                BiocParallel_1.28.3      
  [5] digest_0.6.29             foreach_1.5.2            
  [7] htmltools_0.5.2           magick_2.7.3             
  [9] viridis_0.6.2             fansi_1.0.2              
 [11] ScaledMatrix_1.2.0        cluster_2.1.2            
 [13] doParallel_1.0.16         colorspace_2.0-2         
 [15] ggrepel_0.9.1             xfun_0.29                
 [17] callr_3.7.0               crayon_1.4.2             
 [19] RCurl_1.98-1.5            jsonlite_1.7.3           
 [21] iterators_1.0.13          glue_1.6.1               
 [23] gtable_0.3.0              zlibbioc_1.40.0          
 [25] XVector_0.34.0            MatrixModels_0.5-0       
 [27] GetoptLong_1.0.5          DelayedArray_0.20.0      
 [29] car_3.0-12                BiocSingular_1.10.0      
 [31] shape_1.4.6               abind_1.4-5              
 [33] SparseM_1.81              scales_1.1.1             
 [35] pheatmap_1.0.12           DBI_1.1.2                
 [37] Rcpp_1.0.8                viridisLite_0.4.0        
 [39] clue_0.3-60               rsvd_1.0.5               
 [41] httr_1.4.2                RColorBrewer_1.1-2       
 [43] ellipsis_0.3.2            pkgconfig_2.0.3          
 [45] farver_2.1.0              sass_0.4.0               
 [47] utf8_1.2.2                tidyselect_1.1.1         
 [49] labeling_0.4.2            rlang_1.0.0              
 [51] reshape2_1.4.4            later_1.3.0              
 [53] munsell_0.5.0             tools_4.1.2              
 [55] cli_3.1.1                 generics_0.1.2           
 [57] broom_0.7.12              ggridges_0.5.3           
 [59] evaluate_0.14             stringr_1.4.0            
 [61] fastmap_1.1.0             yaml_2.2.2               
 [63] processx_3.5.2            knitr_1.37               
 [65] fs_1.5.2                  purrr_0.3.4              
 [67] sparseMatrixStats_1.6.0   whisker_0.4              
 [69] quantreg_5.87             compiler_4.1.2           
 [71] rstudioapi_0.13           beeswarm_0.4.0           
 [73] png_0.1-7                 ggsignif_0.6.3           
 [75] tibble_3.1.6              bslib_0.3.1              
 [77] stringi_1.7.6             highr_0.9                
 [79] ps_1.6.0                  lattice_0.20-45          
 [81] Matrix_1.4-0              vctrs_0.3.8              
 [83] pillar_1.7.0              lifecycle_1.0.1          
 [85] jquerylib_0.1.4           GlobalOptions_0.1.2      
 [87] BiocNeighbors_1.12.0      bitops_1.0-7             
 [89] irlba_2.3.5               httpuv_1.6.5             
 [91] R6_2.5.1                  promises_1.2.0.1         
 [93] vipor_0.4.5               codetools_0.2-18         
 [95] assertthat_0.2.1          rprojroot_2.0.2          
 [97] rjson_0.2.21              withr_2.4.3              
 [99] GenomeInfoDbData_1.2.7    beachmat_2.10.0          
[101] rmarkdown_2.11            DelayedMatrixStats_1.16.0
[103] carData_3.0-5             git2r_0.29.0             
[105] getPass_0.2-2            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
