<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tobias Hoch and Daniel Schulz" />

<meta name="date" content="2021-01-30" />

<title>Supplementary Figure 6</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Patch_Detection.html">Bcell Patch Detection</a>
    </li>
    <li>
      <a href="12_Bcell_Score.html">Bcell Score</a>
    </li>
    <li>
      <a href="13_Dysfunction_Stain.html">Dysfunction Validation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplementary Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
    <li>
      <a href="Supp-Figure_6.html">Supp-Figure 6</a>
    </li>
    <li>
      <a href="Supp-Figure_7.html">Supp-Figure 7</a>
    </li>
    <li>
      <a href="Supp-Figure_8.html">Supp-Figure 8</a>
    </li>
    <li>
      <a href="Supp-Figure_9.html">Supp-Figure 9</a>
    </li>
    <li>
      <a href="Supp-Figure_10.html">Supp-Figure 10</a>
    </li>
    <li>
      <a href="Supp-Figure_11.html">Supp-Figure 11</a>
    </li>
    <li>
      <a href="Supp-Figure_12.html">Supp-Figure 12</a>
    </li>
    <li>
      <a href="Supp-Figure_13.html">Supp-Figure 13</a>
    </li>
    <li>
      <a href="Supp-Figure_14.html">Supp-Figure 14</a>
    </li>
    <li>
      <a href="Table_ClinicalData.html">Clinical Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Supplementary Figure 6</h1>
<h4 class="author">Tobias Hoch and Daniel Schulz</h4>
<h4 class="date">2021-01-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-22
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>MelanomaIMC/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtreed246c15656639107e108bbb2fd118635b156f4a3targetblankd246c15a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/d246c15656639107e108bbb2fd118635b156f4a3" target="_blank">d246c15</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupMelanomaIMCtreed246c15656639107e108bbb2fd118635b156f4a3targetblankd246c15a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/MelanomaIMC/tree/d246c15656639107e108bbb2fd118635b156f4a3" target="_blank">d246c15</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    Table_S4.csv
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/Supp-Figure_10.rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/license.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Supp-Figure_6.rmd</code>) and HTML (<code>docs/Supp-Figure_6.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/5418dcda61311e628e9bcf8cdf2a179555526546/analysis/Supp-Figure_6.rmd" target="_blank">5418dcd</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
<td>
add remaining pngs and new .htmls
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/MelanomaIMC/5418dcda61311e628e9bcf8cdf2a179555526546/docs/Supp-Figure_6.html" target="_blank">5418dcd</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
<td>
add remaining pngs and new .htmls
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/64e5fde08d082d625c8c93c1d31d949814f2fa47/analysis/Supp-Figure_6.rmd" target="_blank">64e5fde</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-16
</td>
<td>
change order and naming of supp fig files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/f9a3a83c8bd4e7c866f7199a6bc59723b5f11a35/analysis/Supp-Figure_6.rmd" target="_blank">f9a3a83</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-08
</td>
<td>
clean repo for release
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/fa0f6012c8c2f39948a9ebe93c23cc4dccf462c9/analysis/Supp-Figure_6.rmd" target="_blank">fa0f601</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-06
</td>
<td>
clean Supp Fig code
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/b20b6fbc2a2e0ee6e1e3b73f3ace0d3c11cf084d/analysis/Supp-Figure_6.rmd" target="_blank">b20b6fb</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-02
</td>
<td>
update code for Supp Figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/d6a945a660564f53ecbeb8495e2c3f2dbc1e6f60/analysis/Supp-Figure_6.rmd" target="_blank">d6a945a</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-12-06
</td>
<td>
updated figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/3da15db29f412e8bbd5b3c941ecc9bab416135f9/analysis/Supp-Figure_6.rmd" target="_blank">3da15db</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-11-24
</td>
<td>
changes for revision
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/434eee41eeccd323199a2c2911928aa5ddc4298f/analysis/Supp-Figure_6.rmd" target="_blank">434eee4</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-09-23
</td>
<td>
Figure adaptions and new Supp Figure with gates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/545c2074122dd74d6f848462358773367f0a9024/analysis/Supp-Figure_6.rmd" target="_blank">545c207</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-12-22
</td>
<td>
clean up branch
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/9442cb9048bc92f3320eda95d2b40ce49d09d10e/analysis/Supp-Figure_6.rmd" target="_blank">9442cb9</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-12-22
</td>
<td>
add all new files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/d8c7699e00688b90d8d48c656dc4a296d55dc4c0/analysis/Supp-Figure_6.rmd" target="_blank">d8c7699</a>
</td>
<td>
Tobias Hoch
</td>
<td>
2020-10-23
</td>
<td>
adapt figure 1 and supp fig 6
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/77466b7a723aa4c2dec15a4d54d42b02670cc448/analysis/Supp-Figure_6.rmd" target="_blank">77466b7</a>
</td>
<td>
Tobias Hoch
</td>
<td>
2020-10-22
</td>
<td>
work on subfigures
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This script generates plots for Supplementary Figure 6.</p>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<pre class="r"><code>knitr::opts_chunk$set(echo = TRUE, message= FALSE)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())</code></pre>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions/calculateSummary.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/censor_dat.R
value   ?                                 
visible FALSE                             
        code/helper_functions/detect_mRNA_expression.R
value   ?                                             
visible FALSE                                         
        code/helper_functions/DistanceToClusterCenter.R
value   ?                                              
visible FALSE                                          
        code/helper_functions/findMilieu.R code/helper_functions/findPatch.R
value   ?                                  ?                                
visible FALSE                              FALSE                            
        code/helper_functions/getInfoFromString.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/getSpotnumber.R
value   ?                                    
visible FALSE                                
        code/helper_functions/plotCellCounts.R
value   ?                                     
visible FALSE                                 
        code/helper_functions/plotCellFractions.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/plotDist.R code/helper_functions/read_Data.R
value   ?                                ?                                
visible FALSE                            FALSE                            
        code/helper_functions/scatter_function.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/sceChecks.R
value   ?                                
visible FALSE                            
        code/helper_functions/validityChecks.R
value   ?                                     
visible FALSE                                 </code></pre>
<pre class="r"><code>library(cytomapper)
library(SingleCellExperiment)
library(reshape2)
library(tidyverse)
library(dplyr)
library(data.table) 
library(ggrastr)
library(ggplot2)
library(colorRamps)
library(RColorBrewer)
library(gridExtra)
library(ggpmisc)
library(ComplexHeatmap)
library(scater)
library(dittoSeq)
library(ggbeeswarm)
library(corrplot)
library(ggpubr)
library(cowplot)
library(circlize)
library(ggrepel)
library(rstatix)
library(ape)
library(biomaRt)</code></pre>
</div>
</div>
<div id="load-data" class="section level1">
<h1>Load Data</h1>
<pre class="r"><code># SCE object
sce_prot = readRDS(file = &quot;data/data_for_analysis/sce_protein.rds&quot;)
sce_rna = readRDS(file = &quot;data/data_for_analysis/sce_RNA.rds&quot;)

sce_rna &lt;- sce_rna[,sce_rna$Location != &quot;CTRL&quot;]
sce_prot &lt;- sce_prot[,sce_prot$Location != &quot;CTRL&quot;]</code></pre>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
</div>
<div id="supp-fig-5a" class="section level1">
<h1>Supp Fig 5A</h1>
<pre class="r"><code># sample 300 cells per cell type
sce_prot_sub &lt;- sce_prot[,sce_prot$layer_1_gated != &quot;unlabelled&quot;]

set.seed(2345)
# sub-sample 500 cells
sample &lt;- data.frame(colData(sce_prot_sub)) %&gt;%
  group_by(celltype) %&gt;%
  slice_sample(n=300)

# unique cellIDs
sample &lt;- sample[sample$cellID %in% unique(sample$cellID),]

cur_sce &lt;- sce_prot[,sce_prot$cellID %in% sample$cellID]

good_markers &lt;- rownames(sce_prot)[rowData(sce_prot)$good_marker]

colors &lt;- metadata(sce_prot)$colour_vector$celltype
colors &lt;- colors[c(&quot;B cell&quot;, &quot;BnT cell&quot;, &quot;CD4+ T cell&quot;, &quot;CD8+ T cell&quot;, &quot;FOXP3+ T cell&quot;, &quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;pDC&quot;, &quot;Stroma&quot;, &quot;Tumor&quot;, &quot;unknown&quot;)]

dittoHeatmap(cur_sce,
             genes = good_markers, 
             assay = &quot;asinh&quot;,
            annot.by = c(&quot;celltype&quot;),
            show_colnames = FALSE,
            cluster_rows = TRUE,
            annot.colors = colors,
            heatmap.colors = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            breaks = seq(-3,3, length.out = 101),
            use_raster=TRUE)</code></pre>
<p><img src="figure/Supp-Figure_6.rmd/Supp_Fig_6A_gatedCells_Protein-1.png" width="720" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6A_gatedCells_Protein-1">
Past versions of Supp_Fig_6A_gatedCells_Protein-1.png
</button>
</p>
<div id="fig-Supp_Fig_6A_gatedCells_Protein-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6A_gatedCells_Protein-1.png" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="supp-figure-6b" class="section level1">
<h1>Supp Figure 6B</h1>
<pre class="r"><code>set.seed(2345)

sce_rna_sub &lt;- sce_rna[,sce_rna$layer_1_gated != &quot;unlabelled&quot;]

# remove unkown - not enough cells
sce_rna_sub &lt;- sce_rna_sub[,sce_rna_sub$celltype != &quot;unknown&quot;]

# sub-sample 500 cells
sample &lt;- data.frame(colData(sce_rna_sub)) %&gt;%
  group_by(celltype) %&gt;%
  slice_sample(n=300)

# unique cellIDs
sample &lt;- sample[sample$cellID %in% unique(sample$cellID),]

cur_sce &lt;- sce_rna[,sce_rna$cellID %in% unique(sample$cellID)]

good_markers &lt;- rownames(sce_rna)[rowData(sce_rna)$good_marker]
colors &lt;- metadata(sce_rna)$colour_vector$celltype[]
colors &lt;- colors[c(&quot;CD38&quot;, &quot;CD8- T cell&quot;, &quot;CD8+ T cell&quot;, &quot;HLA-DR&quot;, &quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;Stroma&quot;, &quot;Tumor&quot;, &quot;Vasculature&quot;)]

dittoHeatmap(cur_sce, genes = good_markers, assay = &quot;asinh&quot;,
            annot.by = c(&quot;celltype&quot;),
            show_colnames = FALSE,
            cluster_rows = TRUE,
            annot.colors = colors,
            heatmap.colors = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100),
            breaks = seq(-3,3, length.out = 101),
            use_raster=TRUE)</code></pre>
<p><img src="figure/Supp-Figure_6.rmd/Supp_Fig_6B_gatedCells_RNA-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6B_gatedCells_RNA-1">
Past versions of Supp_Fig_6B_gatedCells_RNA-1.png
</button>
</p>
<div id="fig-Supp_Fig_6B_gatedCells_RNA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6B_gatedCells_RNA-1.png" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="supp-figure-6c-and-5d" class="section level1">
<h1>Supp Figure 6C and 5D</h1>
<p>These plots are generated after the randomForest classification. For this, see files 04_1_RNA_celltype_classification.rmd and 04_1_Protein_celltype_classification.rmd</p>
</div>
<div id="supp-figure-6e" class="section level1">
<h1>Supp Figure 6E</h1>
<pre class="r"><code># rna data
cur_rna &lt;- data.frame(colData(sce_rna))

# protein data
cur_prot &lt;- data.frame(colData(sce_prot))

# rna data set - cell type fractions
rna_sum &lt;- cur_rna %&gt;%
  group_by(Description, celltype) %&gt;%
  summarise(n = n()) %&gt;%
  group_by(Description) %&gt;%
  mutate(fraction = n/sum(n)) %&gt;%
  reshape2::dcast(Description ~ celltype, value.var = &quot;fraction&quot;, fill = 0)

# protein data set - cell type fractions
prot_sum &lt;- cur_prot %&gt;%
  group_by(Description, celltype) %&gt;%
  summarise(n = n()) %&gt;%
  group_by(Description) %&gt;%
  mutate(fraction = n/sum(n)) %&gt;%
  reshape2::dcast(Description ~ celltype, value.var = &quot;fraction&quot;, fill = 0)

# equal images
all(rna_sum$Description == prot_sum$Description)</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># correlation
cor &lt;- cor(rna_sum[,-1], prot_sum[,-1], method = &quot;pearson&quot;)

# reorder cor matrix
cor &lt;- cor[c(&quot;CD38&quot;, &quot;HLA-DR&quot;, &quot;Stroma&quot;, &quot;Vasculature&quot;, &quot;unknown&quot;, &quot;CD8- T cell&quot;, &quot;CD8+ T cell&quot;, &quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;Tumor&quot;),
           c(&quot;B cell&quot;, &quot;BnT cell&quot;, &quot;pDC&quot;, &quot;Stroma&quot;, &quot;unknown&quot;, &quot;FOXP3+ T cell&quot;, &quot;CD4+ T cell&quot;, &quot;CD8+ T cell&quot;, &quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;Tumor&quot;) ]

corrplot(cor, 
         addCoef.col = &quot;black&quot;,
         method = &quot;circle&quot;,
         tl.col=&quot;black&quot;,
         tl.cex = 1.5)</code></pre>
<embed src="figure/Supp-Figure_6.rmd/Supp_Fig_6E_cross_correlation-1.pdf" width="864" style="display: block; margin: auto;" type="application/pdf" />
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6E_cross_correlation-1">
Past versions of Supp_Fig_6E_cross_correlation-1.pdf
</button>
</p>
<div id="fig-Supp_Fig_6E_cross_correlation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6E_cross_correlation-1.pdf" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># correlation and p-value
common_cells &lt;- c(&quot;Macrophage&quot;, &quot;Neutrophil&quot;, &quot;Tumor&quot;, &quot;CD8+ T cell&quot;)

# mean correlation for celltype specific correlation
round(mean(cor(rna_sum[,&quot;Macrophage&quot;], prot_sum[,&quot;Macrophage&quot;]),
     cor(rna_sum[,&quot;Neutrophil&quot;], prot_sum[,&quot;Neutrophil&quot;]),
     cor(rna_sum[,&quot;Tumor&quot;], prot_sum[,&quot;Tumor&quot;]),
     cor(rna_sum[,&quot;CD8+ T cell&quot;], prot_sum[,&quot;CD8+ T cell&quot;])),2)</code></pre>
<pre><code>[1] 0.93</code></pre>
<pre class="r"><code># p-values
cor.test(rna_sum[,&quot;Macrophage&quot;], prot_sum[,&quot;Macrophage&quot;])</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  rna_sum[, &quot;Macrophage&quot;] and prot_sum[, &quot;Macrophage&quot;]
t = 31.629, df = 157, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9050165 0.9481476
sample estimates:
      cor 
0.9297027 </code></pre>
<pre class="r"><code>cor.test(rna_sum[,&quot;Neutrophil&quot;], prot_sum[,&quot;Neutrophil&quot;])</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  rna_sum[, &quot;Neutrophil&quot;] and prot_sum[, &quot;Neutrophil&quot;]
t = 61.685, df = 157, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9727088 0.9853381
sample estimates:
      cor 
0.9799866 </code></pre>
<pre class="r"><code>cor.test(rna_sum[,&quot;Tumor&quot;], prot_sum[,&quot;Tumor&quot;])</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  rna_sum[, &quot;Tumor&quot;] and prot_sum[, &quot;Tumor&quot;]
t = 47.778, df = 157, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9554987 0.9759952
sample estimates:
      cor 
0.9672899 </code></pre>
<pre class="r"><code>cor.test(rna_sum[,&quot;CD8+ T cell&quot;], prot_sum[,&quot;CD8+ T cell&quot;])</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  rna_sum[, &quot;CD8+ T cell&quot;] and prot_sum[, &quot;CD8+ T cell&quot;]
t = 26.889, df = 157, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.8740854 0.9307515
sample estimates:
      cor 
0.9064157 </code></pre>
</div>
<div id="supp-figure-6f" class="section level1">
<h1>Supp Figure 6F</h1>
<div id="detection-of-chemokine-expressing-cells" class="section level2">
<h2>Detection of chemokine expressing cells</h2>
<p>for the detection of chemokine expressing cells we make use of the fact that we also measured a negative control (DapB).</p>
<pre class="r"><code># get the names of the chemokine channels without the negative control channel
chemokine_channels = rownames(sce_rna[which(grepl(&quot;T\\d+_&quot;,rownames(sce_rna)) &amp; ! grepl(&quot;DapB&quot;,rownames(sce_rna))),])
chemokine_channels_sub &lt;- c(&quot;T2_CCL22&quot;)

# run function to define chemokine expressing cells 
output_list &lt;- compute_difference(sce_rna, 
                          cellID = &quot;cellID&quot;, 
                          assay_name = &quot;asinh&quot;, 
                          threshold = 0.01, 
                          mRNA_channels = chemokine_channels_sub, 
                          negative_control = &quot;T6_DapB&quot;, 
                          return_calc_metrics = TRUE)</code></pre>
</div>
<div id="plot-results-from-chemokine-detection" class="section level2">
<h2>Plot Results from Chemokine Detection</h2>
<pre class="r"><code># check difference between DapB and signal (histogram)
plot_list = list()
for(i in chemokine_channels_sub){
  
  # subset whole data set for visualization purposes
  diff_chemo &lt;- output_list[[i]]
  diff_chemo_sub &lt;- diff_chemo[sample(nrow(diff_chemo), nrow(diff_chemo)*0.5), ]

  a &lt;- ggplot(data = diff_chemo_sub, aes(x=scaled_diff)) + 
  geom_histogram(binwidth = 0.05, aes(fill = 
                                       ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
  xlab(paste(paste(&quot;Scaled Difference &quot;, i, sep = &quot; &quot;), &quot; - DapB&quot;, sep = &quot;&quot;)) + 
    xlim(-5,7) +
    theme_minimal() + 
    theme(text = element_text(size=20),
          legend.position = &quot;none&quot;) + 
    scale_fill_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;))
  
  # significant cells defined by subtraction
  b &lt;- ggplot(data=diff_chemo_sub, aes(x=mean_negative_control, y=mean_chemokine)) + 
    geom_point_rast(alpha=0.2, aes(col = 
                                ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
    scale_color_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;), 
                       name = &quot;Legend&quot;) +
    guides(color = guide_legend(override.aes = list(alpha=1, size=3))) +
    xlim(0,5.5) + ylim(0,5.5) +
    ylab(paste(&quot;Mean expression of&quot;, i, sep=&quot; &quot;)) +
    xlab(&quot;Mean DapB mRNA expression&quot;) +
    theme_minimal() + 
    theme(text = element_text(size=20)) 

  
  grid.arrange(a,b, nrow = 1, ncol=2)
}</code></pre>
<pre><code>Warning: Removed 327 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<embed src="figure/Supp-Figure_6.rmd/Supp_Fig_6F_chemokine_detection-1.pdf" width="425" style="display: block; margin: auto;" type="application/pdf" />
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6F_chemokine_detection-1">
Past versions of Supp_Fig_6F_chemokine_detection-1.pdf
</button>
</p>
<div id="fig-Supp_Fig_6F_chemokine_detection-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6F_chemokine_detection-1.pdf" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="supp-figure-6g" class="section level1">
<h1>Supp Figure 6G</h1>
<p>This script reproduces the homology analysis between the different chemokines. We downloaded the data from www.ncbi.nlm.nih.gov and saved the transcript sequences. <a href="https://www.ebi.ac.uk/Tools/msa/clustalo/" class="uri">https://www.ebi.ac.uk/Tools/msa/clustalo/</a> was used to align all-vs-all transcripts using the following call:</p>
<pre><code>$APPBIN/clustal-omega-1.2.4/bin/clustalo --infile clustalo-E20210914-122047-0397-8283578-p2m.sequence --threads 8 --MAC-RAM 8000 --verbose --guidetree-out clustalo-E20210914-122047-0397-8283578-p2m.dnd --outfmt clustal --resno --outfile clustalo-E20210914-122047-0397-8283578-p2m.clustal_num --output-order tree-order --seqtype dna</code></pre>
<p>We will first look at the identity matrix, inidcating the percentage of sequence similarity.</p>
<pre class="r"><code>seq_similarity &lt;- read.table(&quot;data/data_for_analysis/ClustalW_results/identity_matrix.txt&quot;)
rownames(seq_similarity) &lt;- seq_similarity$V2
seq_similarity &lt;- seq_similarity[,-c(1,2)]

# Remove non coding transcript variants
seq_similarity &lt;- seq_similarity[!grepl(&quot;XR_|NR_&quot;, rownames(seq_similarity)),
                                 !grepl(&quot;XR_|NR_&quot;, rownames(seq_similarity))]
rownames(seq_similarity) &lt;- sub(&quot;\\.[0-9]*&quot;, &quot;&quot;, rownames(seq_similarity))


phylogenetic_tree &lt;- read.tree(&quot;data/data_for_analysis/ClustalW_results/phylogenetic_tree.txt&quot;)</code></pre>
<p>Now, we will map between RefSeq transcript ids, ensemble transcript ids and gene names.</p>
<pre class="r"><code># workaround: useMart error: SSL certificate problem: unable to get local issuer certificate
httr::set_config(httr::config(ssl_verifypeer = FALSE), override = FALSE)

ensembl &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
cur_tab &lt;- getBM(attributes=c(&quot;refseq_mrna&quot;, &quot;ensembl_gene_id&quot;, &quot;hgnc_symbol&quot;), 
      filters = &quot;refseq_mrna&quot;, values = rownames(seq_similarity), 
      mart = ensembl, uniqueRows = FALSE)
cur_tab &lt;- cur_tab[match(rownames(seq_similarity), cur_tab$refseq_mrna),]

rownames(seq_similarity)  &lt;- colnames(seq_similarity) &lt;- 
    paste0(cur_tab$refseq_mrna, &quot;_&quot;, cur_tab$hgnc_symbol)</code></pre>
<p>And we compare it to correlation in expression across all cells.</p>
<pre class="r"><code>final_sce &lt;- sce_rna
# select only the cells that express chemokines
for_analysis &lt;- final_sce[,final_sce$expressor != &quot;NA&quot;]

cur_cor &lt;- cor(t(assay(for_analysis, &quot;asinh&quot;)[c(&quot;T5_CCL4&quot;, &quot;T7_CCL18&quot;, &quot;T1_CXCL8&quot;,
                                             &quot;T4_CXCL10&quot;, &quot;T3_CXCL12&quot;, &quot;T8_CXCL13&quot;,
                                             &quot;T12_CCL2&quot;, &quot;T2_CCL22&quot;,
                                             &quot;T9_CXCL9&quot;, &quot;T11_CCL8&quot;, &quot;T10_CCL19&quot;),]), 
                                           method = &quot;spearman&quot;)

pheatmap(cur_cor, color = colorRampPalette(c(&quot;dark blue&quot;, &quot;white&quot;, &quot;dark red&quot;))(100), 
         breaks = seq(-1, 1, length.out = 100))</code></pre>
<p><img src="figure/Supp-Figure_6.rmd/cor-expression-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cor-expression-1">
Past versions of cor-expression-1.png
</button>
</p>
<div id="fig-cor-expression-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/cor-expression-1.png" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor_tibbble &lt;- cur_cor %&gt;%
    as_tibble() %&gt;%
    mutate(probe = rownames(cur_cor)) %&gt;% 
    pivot_longer(cols = 1:ncol(cur_cor)) %&gt;%
    mutate(probe = str_split(probe, pattern = &quot;_&quot;, simplify = TRUE)[,2],
           name = str_split(name, pattern = &quot;_&quot;, simplify = TRUE)[,2]) %&gt;%
    arrange(probe, name) %&gt;%
    filter(probe != name)

sim_tibble &lt;- seq_similarity %&gt;%
    as_tibble() %&gt;% 
    mutate(probe = rownames(seq_similarity)) %&gt;% 
    pivot_longer(cols = 1:ncol(seq_similarity)) %&gt;%
    mutate(from_chemo = str_split(probe, pattern = &quot;_&quot;, simplify = TRUE)[,3],
           to_chemo = str_split(name, pattern = &quot;_&quot;, simplify = TRUE)[,3]) %&gt;%
    group_by(from_chemo, to_chemo) %&gt;%
    dplyr::summarize(mean_similarity = mean(value, na.rm=TRUE)) %&gt;%
    arrange(from_chemo, to_chemo) %&gt;%
    filter(from_chemo != to_chemo)

all.equal(paste(cor_tibbble$probe, cor_tibbble$name),
          paste(sim_tibble$from_chemo, sim_tibble$to_chemo))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>ggplot(data.frame(similarity = sim_tibble$mean_similarity,
                  correlation = cor_tibbble$value)) +
    geom_point(aes(similarity, correlation))</code></pre>
<pre><code>Warning: Removed 20 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/Supp-Figure_6.rmd/viz-correlation-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-viz-correlation-1">
Past versions of viz-correlation-1.png
</button>
</p>
<div id="fig-viz-correlation-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/viz-correlation-1.png" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor.test(sim_tibble$mean_similarity,
         cor_tibbble$value)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  sim_tibble$mean_similarity and cor_tibbble$value
t = 4.9613, df = 88, p-value = 3.387e-06
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.2883482 0.6150626
sample estimates:
      cor 
0.4675216 </code></pre>
<div id="number-of-co-expressors" class="section level2">
<h2>Number of co-expressors</h2>
<p>Finally, we will compare the sequence similarity to the jaccard index of chemokine expressors.</p>
<pre class="r"><code># Define the jaccard similarity
jaccard &lt;- function(x,y){
    intersection &lt;- length(intersect(x,y))
    union = length(x) + length(y) - intersection
    return (intersection/union)
}

# We will pass the unique cell ids into this function
cur_out &lt;- lapply(seq_len(nrow(sim_tibble)),
       function(x){
           from_chemo_cells &lt;- colnames(final_sce)[colData(final_sce)[[sim_tibble$from_chemo[x]]] != 0]
           to_chemo_cells &lt;- colnames(final_sce)[colData(final_sce)[[sim_tibble$to_chemo[x]]] != 0]
           return(jaccard(from_chemo_cells, to_chemo_cells))
       })
sim_tibble$jaccard_sim &lt;- unlist(cur_out)

ggplot(data.frame(similarity = sim_tibble$mean_similarity,
                  jaccard_sim = sim_tibble$jaccard_sim)) +
    geom_point(aes(similarity, jaccard_sim)) +
    geom_smooth(method = &quot;lm&quot;, aes(similarity, jaccard_sim)) +  stat_cor(method = &quot;pearson&quot;,
           aes(x = similarity, y = jaccard_sim, label = paste0(&quot;atop(&quot;, ..r.label..,  &quot;,&quot;, ..p.label.. ,&quot;)&quot;)),
           size = 7, cor.coef.name = &quot;R&quot;, label.sep=&quot;\n&quot;, label.y.npc = &quot;top&quot;) + 
    theme_bw() + theme(text=element_text(size=15)) +
    xlab(&quot;Sequence similarity&quot;) + ylab(&quot;Chemokine co-expression [Jaccard index]&quot;)</code></pre>
<pre><code>Warning: Removed 20 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>Warning: Removed 20 rows containing non-finite values (stat_cor).</code></pre>
<pre><code>Warning: Removed 20 rows containing missing values (geom_point).</code></pre>
<embed src="figure/Supp-Figure_6.rmd/Supp_Fig_6G_JaccardPlot-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6G_JaccardPlot-1">
Past versions of Supp_Fig_6G_JaccardPlot-1.pdf
</button>
</p>
<div id="fig-Supp_Fig_6G_JaccardPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6G_JaccardPlot-1.pdf" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cor.test(sim_tibble$mean_similarity,
         sim_tibble$jaccard_sim)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  sim_tibble$mean_similarity and sim_tibble$jaccard_sim
t = 1.6191, df = 88, p-value = 0.109
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.03836368  0.36433718
sample estimates:
      cor 
0.1700787 </code></pre>
</div>
</div>
<div id="supp-figure-6h" class="section level1">
<h1>Supp Figure 6H</h1>
<div id="fraction-of-tumor-cells-that-express-chemokines" class="section level2">
<h2>Fraction of Tumor Cells that express chemokines</h2>
<pre class="r"><code>cur_dat &lt;- data.frame(colData(sce_rna))
cur_dat &lt;- cur_dat %&gt;%
  filter(celltype == &quot;Tumor&quot;) %&gt;%
  filter(Location != &quot;CTRL&quot;)

cur_dat &lt;- cur_dat[,c(&quot;ImageNumber&quot;, &quot;Mutation&quot;, colnames(cur_dat)[grepl(glob2rx(&quot;C*L*&quot;),names(cur_dat))])]

# colSums of Chemokines in Tumor Cells (Multiple Producer count more than once)
cur_dat &lt;- cur_dat %&gt;%
  group_by(ImageNumber, Mutation) %&gt;%
  mutate(cells = n()) %&gt;%
  group_by(ImageNumber, cells, Mutation) %&gt;%
  summarise_each(funs(sum))</code></pre>
<pre><code>Warning: `summarise_each_()` was deprecated in dplyr 0.7.0.
Please use `across()` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<pre><code>Warning: `funs()` was deprecated in dplyr 0.8.0.
Please use a list of either functions or lambdas: 

  # Simple named list: 
  list(mean = mean, median = median)

  # Auto named with `tibble::lst()`: 
  tibble::lst(mean, median)

  # Using lambdas
  list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<pre class="r"><code># compute fractions
cur_dat[,4:14] &lt;- cur_dat[,4:14] / t(cur_dat$cells)

cur_dat &lt;- cur_dat %&gt;%
  filter(cells &gt; 200) %&gt;%
  reshape2::melt(id.vars=c(&quot;ImageNumber&quot;, &quot;cells&quot;, &quot;Mutation&quot;), variable.name=&quot;chemokine&quot;, value.name=&quot;fraction&quot;)

ggplot(cur_dat,aes(x=fct_reorder(chemokine, fraction, .fun = median, .desc = TRUE), y=fraction+0.001)) + 
  geom_boxplot(alpha=.5) +
  geom_quasirandom(alpha=.2) +
  theme_bw() + 
  theme(text=element_text(size=16)) +
  ylab(&quot;Fraction of Expressing Tumor Cells\n(fraction + 0.001)&quot;) +
  xlab(&quot;&quot;) +
  scale_y_log10() +
  annotation_logticks(sides = &quot;l&quot;) +
  geom_hline(yintercept = median(cur_dat$fraction+0.001), linetype = 2) </code></pre>
<embed src="figure/Supp-Figure_6.rmd/Supp_Fig_6H_tumor_cells_chemokine-1.pdf" width="864" style="display: block; margin: auto;" type="application/pdf" />
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Supp_Fig_6H_tumor_cells_chemokine-1">
Past versions of Supp_Fig_6H_tumor_cells_chemokine-1.pdf
</button>
</p>
<div id="fig-Supp_Fig_6H_tumor_cells_chemokine-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/BodenmillerGroup/MelanomaIMC/blob/235386f58a01f8dc1a9defcf3deb6434e6b464af/docs/figure/Supp-Figure_6.rmd/Supp_Fig_6H_tumor_cells_chemokine-1.pdf" target="_blank">235386f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2022-02-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.3 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] biomaRt_2.50.3              ape_5.6-1                  
 [3] rstatix_0.7.0               ggrepel_0.9.1              
 [5] circlize_0.4.13             cowplot_1.1.1              
 [7] ggpubr_0.4.0                corrplot_0.92              
 [9] ggbeeswarm_0.6.0            dittoSeq_1.6.0             
[11] scater_1.22.0               scuttle_1.4.0              
[13] ComplexHeatmap_2.10.0       ggpmisc_0.4.5              
[15] ggpp_0.4.3                  gridExtra_2.3              
[17] RColorBrewer_1.1-2          colorRamps_2.3             
[19] ggrastr_1.0.1               data.table_1.14.2          
[21] forcats_0.5.1               stringr_1.4.0              
[23] purrr_0.3.4                 readr_2.1.2                
[25] tidyr_1.2.0                 tibble_3.1.6               
[27] ggplot2_3.3.5               tidyverse_1.3.1            
[29] reshape2_1.4.4              cytomapper_1.6.0           
[31] SingleCellExperiment_1.16.0 SummarizedExperiment_1.24.0
[33] Biobase_2.54.0              GenomicRanges_1.46.1       
[35] GenomeInfoDb_1.30.1         IRanges_2.28.0             
[37] S4Vectors_0.32.3            BiocGenerics_0.40.0        
[39] MatrixGenerics_1.6.0        matrixStats_0.61.0         
[41] EBImage_4.36.0              dplyr_1.0.7                
[43] workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                shinydashboard_0.7.2     
  [3] tidyselect_1.1.1          RSQLite_2.2.9            
  [5] AnnotationDbi_1.56.2      htmlwidgets_1.5.4        
  [7] BiocParallel_1.28.3       munsell_0.5.0            
  [9] ScaledMatrix_1.2.0        codetools_0.2-18         
 [11] withr_2.4.3               colorspace_2.0-2         
 [13] filelock_1.0.2            highr_0.9                
 [15] knitr_1.37                rstudioapi_0.13          
 [17] ggsignif_0.6.3            labeling_0.4.2           
 [19] git2r_0.29.0              GenomeInfoDbData_1.2.7   
 [21] farver_2.1.0              bit64_4.0.5              
 [23] pheatmap_1.0.12           rhdf5_2.38.0             
 [25] rprojroot_2.0.2           vctrs_0.3.8              
 [27] generics_0.1.2            xfun_0.29                
 [29] BiocFileCache_2.2.1       R6_2.5.1                 
 [31] doParallel_1.0.16         clue_0.3-60              
 [33] rsvd_1.0.5                locfit_1.5-9.4           
 [35] cachem_1.0.6              bitops_1.0-7             
 [37] rhdf5filters_1.6.0        DelayedArray_0.20.0      
 [39] assertthat_0.2.1          promises_1.2.0.1         
 [41] scales_1.1.1              beeswarm_0.4.0           
 [43] gtable_0.3.0              beachmat_2.10.0          
 [45] Cairo_1.5-14              processx_3.5.2           
 [47] rlang_1.0.0               MatrixModels_0.5-0       
 [49] systemfonts_1.0.3         splines_4.1.2            
 [51] GlobalOptions_0.1.2       broom_0.7.12             
 [53] yaml_2.2.2                abind_1.4-5              
 [55] modelr_0.1.8              backports_1.4.1          
 [57] httpuv_1.6.5              tools_4.1.2              
 [59] ellipsis_0.3.2            raster_3.5-15            
 [61] jquerylib_0.1.4           ggridges_0.5.3           
 [63] Rcpp_1.0.8                plyr_1.8.6               
 [65] progress_1.2.2            sparseMatrixStats_1.6.0  
 [67] zlibbioc_1.40.0           RCurl_1.98-1.5           
 [69] prettyunits_1.1.1         ps_1.6.0                 
 [71] GetoptLong_1.0.5          viridis_0.6.2            
 [73] haven_2.4.3               cluster_2.1.2            
 [75] fs_1.5.2                  magrittr_2.0.2           
 [77] magick_2.7.3              SparseM_1.81             
 [79] reprex_2.0.1              whisker_0.4              
 [81] hms_1.1.1                 mime_0.12                
 [83] fftwtools_0.9-11          evaluate_0.14            
 [85] xtable_1.8-4              XML_3.99-0.8             
 [87] jpeg_0.1-9                readxl_1.3.1             
 [89] shape_1.4.6               compiler_4.1.2           
 [91] crayon_1.4.2              htmltools_0.5.2          
 [93] mgcv_1.8-38               later_1.3.0              
 [95] tzdb_0.2.0                tiff_0.1-11              
 [97] lubridate_1.8.0           DBI_1.1.2                
 [99] dbplyr_2.1.1              rappdirs_0.3.3           
[101] Matrix_1.4-0              car_3.0-12               
[103] cli_3.1.1                 parallel_4.1.2           
[105] pkgconfig_2.0.3           getPass_0.2-2            
[107] sp_1.4-6                  terra_1.5-17             
[109] xml2_1.3.3                foreach_1.5.2            
[111] svglite_2.0.0             vipor_0.4.5              
[113] bslib_0.3.1               XVector_0.34.0           
[115] rvest_1.0.2               callr_3.7.0              
[117] digest_0.6.29             Biostrings_2.62.0        
[119] rmarkdown_2.11            cellranger_1.1.0         
[121] DelayedMatrixStats_1.16.0 curl_4.3.2               
[123] shiny_1.7.1               quantreg_5.87            
[125] rjson_0.2.21              nlme_3.1-155             
[127] lifecycle_1.0.1           jsonlite_1.7.3           
[129] Rhdf5lib_1.16.0           carData_3.0-5            
[131] BiocNeighbors_1.12.0      viridisLite_0.4.0        
[133] fansi_1.0.2               pillar_1.7.0             
[135] lattice_0.20-45           KEGGREST_1.34.0          
[137] fastmap_1.1.0             httr_1.4.2               
[139] glue_1.6.1                png_0.1-7                
[141] iterators_1.0.13          svgPanZoom_0.3.4         
[143] bit_4.0.4                 stringi_1.7.6            
[145] sass_0.4.0                HDF5Array_1.22.1         
[147] blob_1.2.2                BiocSingular_1.10.0      
[149] memoise_2.0.1             irlba_2.3.5              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
