<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tobias Hoch and Daniel Schulz" />

<meta name="date" content="2020-08-04" />

<title>Supplementary Figure 1</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Melanoma Publication</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    RNA TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_RNA_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_RNA_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_RNA_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_RNA_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_RNA_classification_subclustering.html">Subclustering</a>
    </li>
    <li>
      <a href="05_RNA_chemokine_expressing_cells.html">Chemokine-Expressing Cells</a>
    </li>
    <li>
      <a href="06_RNA_chemokine_patch_detection.html">Chemokine Patches</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Protein TMA
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_Protein_read_data.html">Read Data</a>
    </li>
    <li>
      <a href="02_Protein_annotations.html">Annotations</a>
    </li>
    <li>
      <a href="03_Protein_quality_control.html">Quality Control</a>
    </li>
    <li>
      <a href="04_1_Protein_celltype_classification.html">Automated Celltype Classification</a>
    </li>
    <li>
      <a href="04_2_Protein_classification_subclustering.html">Subclustering</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Misc.
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_prepare_clinical_dat.html">Prep Clinical Table</a>
    </li>
    <li>
      <a href="07_TCF7_PD1_gating.html">TCF7/PD1 Gating</a>
    </li>
    <li>
      <a href="08_color_vectors.html">Color Vectors</a>
    </li>
    <li>
      <a href="09_Tcell_Score.html">Tcell Score</a>
    </li>
    <li>
      <a href="10_Dysfunction_Score.html">Dysfunction Score</a>
    </li>
    <li>
      <a href="11_Bcell_Score.html">Bcell Score</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Main Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Supplemental Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Supp-Figure_1.html">Supp-Figure 1</a>
    </li>
    <li>
      <a href="Supp-Figure_2.html">Supp-Figure 2</a>
    </li>
    <li>
      <a href="Supp-Figure_3.html">Supp-Figure 3</a>
    </li>
    <li>
      <a href="Supp-Figure_4.html">Supp-Figure 4</a>
    </li>
    <li>
      <a href="Supp-Figure_5.html">Supp-Figure 5</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Additional Material
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Summary_Statistics.html">Summary Statistics</a>
    </li>
    <li>
      <a href="XX_hazard_ratio.html">Hazard Ratio Analysis</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Supplementary Figure 1</h1>
<h4 class="author">Tobias Hoch and Daniel Schulz</h4>
<h4 class="date">2020-08-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>melanoma_publication_old_data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200728code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200728)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200728code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200728)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree4affda4895cc0b25d4336d4649dce69baf935b77targetblank4affda4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/4affda4895cc0b25d4336d4649dce69baf935b77" target="_blank">4affda4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomBodenmillerGroupmelanomapublicationtree4affda4895cc0b25d4336d4649dce69baf935b77targetblank4affda4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/BodenmillerGroup/melanoma_publication/tree/4affda4895cc0b25d4336d4649dce69baf935b77" target="_blank">4affda4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    analysis/._clinical metadata preparation.Rmd
    Ignored:    code/.DS_Store
    Ignored:    code/._.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/data_for_analysis/
    Ignored:    data/full_data/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._protein_neutrophil.png
    Ignored:    output/._rna_neutrophil.png
    Ignored:    output/PSOCKclusterOut/
    Ignored:    output/bcell_grouping.png
    Ignored:    output/dysfunction_correlation.pdf

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/Figure_4.rmd
    Modified:   analysis/Supp-Figure_1.rmd
    Modified:   analysis/Supp-Figure_4.rmd
    Modified:   analysis/Supp-Figure_5.rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Supp-Figure_1.rmd</code>) and HTML (<code>docs/Supp-Figure_1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/4affda4895cc0b25d4336d4649dce69baf935b77/analysis/Supp-Figure_1.rmd" target="_blank">4affda4</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-04-14
</td>
<td>
figure adaptations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/4affda4895cc0b25d4336d4649dce69baf935b77/docs/Supp-Figure_1.html" target="_blank">4affda4</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-04-14
</td>
<td>
figure adaptations
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/32038914d8c7a9d59102c5e037cad3afae9ac319/analysis/Supp-Figure_1.rmd" target="_blank">3203891</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-19
</td>
<td>
change celltype names
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/32038914d8c7a9d59102c5e037cad3afae9ac319/docs/Supp-Figure_1.html" target="_blank">3203891</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-19
</td>
<td>
change celltype names
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/ee1595de9ec17ee414a5beafb9ed08d174e12139/analysis/Supp-Figure_1.rmd" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/ee1595de9ec17ee414a5beafb9ed08d174e12139/docs/Supp-Figure_1.html" target="_blank">ee1595d</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-12
</td>
<td>
clean repo and adapt files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/BodenmillerGroup/melanoma_publication/3f5af3f59358eebb33aa2a8a06fddca34dad6a6b/docs/Supp-Figure_1.html" target="_blank">3f5af3f</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-09
</td>
<td>
add .html files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/20a1458a4a84f38a314ea39a12a78aad3595e51d/analysis/Supp-Figure_1.rmd" target="_blank">20a1458</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-04
</td>
<td>
adapt figure order
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/f9bb33a83d519d86072a7e716214ff78bce022a0/analysis/Supp-Figure_1.rmd" target="_blank">f9bb33a</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2021-02-04
</td>
<td>
new Figure 5 and minor changes in figure order
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/9442cb9048bc92f3320eda95d2b40ce49d09d10e/analysis/Supp-Figure_1.rmd" target="_blank">9442cb9</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-12-22
</td>
<td>
add all new files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/1af3353530babcfa8b447823d242ec1948959f96/analysis/Supp-Figure_1.rmd" target="_blank">1af3353</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-16
</td>
<td>
add stuff
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/a6b51cdba16c8bff44aeee86c92f092f34b980cc/analysis/Supp-Figure_1.rmd" target="_blank">a6b51cd</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-14
</td>
<td>
clean scripts, add new subfigures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/BodenmillerGroup/melanoma_publication/blob/d8819f2b618d0620606a5e5b181d3b5340469763/analysis/Supp-Figure_1.rmd" target="_blank">d8819f2</a>
</td>
<td>
toobiwankenobi
</td>
<td>
2020-10-08
</td>
<td>
read new data (nuclei expansion) and adapt scripts
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This script generates plots for Supplementary Figure 1.</p>
</div>
<div id="preparations" class="section level1">
<h1>Preparations</h1>
<div id="load-libraries" class="section level2">
<h2>Load libraries</h2>
<pre class="r"><code>sapply(list.files(&quot;code/helper_functions&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions/calculateSummary.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/censor_dat.R
value   ?                                 
visible FALSE                             
        code/helper_functions/detect_mRNA_expression.R
value   ?                                             
visible FALSE                                         
        code/helper_functions/DistanceToClusterCenter.R
value   ?                                              
visible FALSE                                          
        code/helper_functions/findMilieu.R code/helper_functions/findPatch.R
value   ?                                  ?                                
visible FALSE                              FALSE                            
        code/helper_functions/getInfoFromString.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/getSpotnumber.R
value   ?                                    
visible FALSE                                
        code/helper_functions/plotCellCounts.R
value   ?                                     
visible FALSE                                 
        code/helper_functions/plotCellFractions.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/plotDist.R
value   ?                               
visible FALSE                           
        code/helper_functions/scatter_function.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/sceChecks.R
value   ?                                
visible FALSE                            
        code/helper_functions/validityChecks.R
value   ?                                     
visible FALSE                                 </code></pre>
<pre class="r"><code>library(SingleCellExperiment)</code></pre>
<pre><code>Loading required package: SummarizedExperiment</code></pre>
<pre><code>Loading required package: MatrixGenerics</code></pre>
<pre><code>Loading required package: matrixStats</code></pre>
<pre><code>
Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>Loading required package: parallel</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:parallel&#39;:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:base&#39;:

    expand.grid</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Loading required package: Biobase</code></pre>
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    &#39;browseVignettes()&#39;. To cite Bioconductor, see
    &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>
Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:MatrixGenerics&#39;:

    rowMedians</code></pre>
<pre><code>The following objects are masked from &#39;package:matrixStats&#39;:

    anyMissing, rowMedians</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:SummarizedExperiment&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:GenomicRanges&#39;:

    shift</code></pre>
<pre><code>The following object is masked from &#39;package:IRanges&#39;:

    shift</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, second</code></pre>
<pre class="r"><code>library(ggplot2)
library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following objects are masked from &#39;package:GenomicRanges&#39;:

    intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:GenomeInfoDb&#39;:

    intersect</code></pre>
<pre><code>The following objects are masked from &#39;package:IRanges&#39;:

    collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>The following objects are masked from &#39;package:S4Vectors&#39;:

    first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>The following objects are masked from &#39;package:BiocGenerics&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following object is masked from &#39;package:matrixStats&#39;:

    count</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(stringr)
library(ggpubr)
library(ComplexHeatmap)</code></pre>
<pre><code>Loading required package: grid</code></pre>
<pre><code>========================================
ComplexHeatmap version 2.4.3
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite:
Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
  genomic data. Bioinformatics 2016.

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
<pre class="r"><code>library(circlize)</code></pre>
<pre><code>========================================
circlize version 0.4.12
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
<pre class="r"><code>library(gridExtra)</code></pre>
<pre><code>
Attaching package: &#39;gridExtra&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    combine</code></pre>
<pre><code>The following object is masked from &#39;package:Biobase&#39;:

    combine</code></pre>
<pre><code>The following object is masked from &#39;package:BiocGenerics&#39;:

    combine</code></pre>
<pre class="r"><code>library(ggbeeswarm)
library(ggrepel)
library(ggpmisc)</code></pre>
<pre><code>
Attaching package: &#39;ggpmisc&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggplot2&#39;:

    annotate</code></pre>
<pre class="r"><code>library(ggrastr)</code></pre>
</div>
<div id="load-and-process-the-data" class="section level2">
<h2>Load and process the data</h2>
<pre class="r"><code># load data
cells1 = fread(&quot;data/data_for_analysis/12plex_validation/overexpression/20190305/cell.csv&quot;, header = T,sep=&quot;,&quot;)
cells2 = fread(&quot;data/data_for_analysis/12plex_validation/overexpression/20190306/cell.csv&quot;, header = T,sep=&quot;,&quot;)
meta1 = fread(&quot;data/data_for_analysis/12plex_validation/overexpression/20190305/Image.csv&quot;,header = T,sep=&quot;,&quot;)</code></pre>
<pre><code>Warning in require_bit64_if_needed(ans): Some columns are type &#39;integer64&#39;
but package bit64 is not installed. Those columns will print as strange
looking floating point data. There is no need to reload the data. Simply
install.packages(&#39;bit64&#39;) to obtain the integer64 print method and print the
data again.</code></pre>
<pre class="r"><code>meta2 = fread(&quot;data/data_for_analysis/12plex_validation/overexpression/20190306/Image.csv&quot;,header = T,sep=&quot;,&quot;)</code></pre>
<pre><code>Warning in require_bit64_if_needed(ans): Some columns are type &#39;integer64&#39;
but package bit64 is not installed. Those columns will print as strange
looking floating point data. There is no need to reload the data. Simply
install.packages(&#39;bit64&#39;) to obtain the integer64 print method and print the
data again.</code></pre>
<pre class="r"><code>panel = fread( &quot;data/data_for_analysis/12plex_validation/overexpression/20190305/20190305_A431_overexpression.csv&quot;)

# extract replicate and stain info
meta1 = meta1[,.(ImageNumber,FileName_CellImage, Group_Number, Metadata_Target)]
meta1[,&#39;:=&#39; (replicate = getInfoFromString(FileName_CellImage,&quot;_&quot;,1)), by=ImageNumber]
meta2 = meta2[,.(ImageNumber,FileName_CellImage, Group_Number, Metadata_Target)]
meta2[,&#39;:=&#39; (replicate = getInfoFromString(FileName_CellImage,&quot;_&quot;,1)), by=ImageNumber]

# sort Metal Number in same order than in the Image.csv file
panel$`Metal Number` = str_extract(panel$`Metal Tag`, &quot;[0-9]+&quot;)
panel = panel[order(panel$`Metal Number`),]
panel$channel = c(1:nrow(panel))

cells_long1 = melt.data.table(cells1,id.vars = c(&quot;ImageNumber&quot;, &quot;ObjectNumber&quot;),variable.factor = F)
cells_long2 = melt.data.table(cells2,id.vars = c(&quot;ImageNumber&quot;, &quot;ObjectNumber&quot;),variable.factor = F)

# create unique ID with Image and ObjectNumber
cells_long1[,id := paste(ImageNumber,ObjectNumber,sep =&quot;_&quot;)]
cells_long2[,id := paste(ImageNumber,ObjectNumber,sep =&quot;_&quot;)]

# multiply value by 2E16 since it divided by this number in CellProfiler
cells_long1$value = cells_long1$value * 65535
cells_long2$value = cells_long2$value * 65535

# calculate counts_asinh
cells_long1[,&#39;:=&#39; (channel = as.integer(getInfoFromFileList(variable,sep=&quot;_&quot;,strPos = 4,censorStr = &quot;c&quot;)),
                 counts_asinh = asinh(value/1)),
                 by=id]
cells_long2[,&#39;:=&#39; (channel = as.integer(getInfoFromFileList(variable,sep=&quot;_&quot;,strPos = 4,censorStr = &quot;c&quot;)),
                 counts_asinh = asinh(value/1)),
                 by=id]

# take only FullStack entries and not FullStackFiltered
cells_long1[,measurement:=getInfoFromString(variable,&quot;_&quot;,3),by=variable]
cells_long1[,signal_type:=getInfoFromString(variable,&quot;_&quot;,2),by=variable]
cells_long2[,measurement:=getInfoFromString(variable,&quot;_&quot;,3),by=variable]
cells_long2[,signal_type:=getInfoFromString(variable,&quot;_&quot;,2),by=variable]

unique(cells_long1$signal_type)</code></pre>
<pre><code>[1] &quot;MeanIntensityCorrectedLS&quot; &quot;MeanIntensityCorrected&quot;  
[3] &quot;MeanIntensity&quot;           </code></pre>
<pre class="r"><code>unique(cells_long2$signal_type)</code></pre>
<pre><code>[1] &quot;MeanIntensityCorrectedLS&quot; &quot;MeanIntensityCorrected&quot;  
[3] &quot;MeanIntensity&quot;           </code></pre>
</div>
<div id="merge-meta-data-with-cells-data" class="section level2">
<h2>Merge meta data with cells data</h2>
<pre class="r"><code>cells1 = merge(cells_long1, meta1,by=&quot;ImageNumber&quot;)
cells2 = merge(cells_long2, meta2,by=&quot;ImageNumber&quot;)

cells = rbind(cells1, cells2)</code></pre>
</div>
<div id="exclude-dna-histone-and-pancytokeratin-data-and-then-get-the-info-of-ppib-staining" class="section level2">
<h2>Exclude DNA, Histone and panCytokeratin data and then get the info of PPIB staining</h2>
<pre class="r"><code># exclude some channels, make sure to exclude the right ones!
cells_panel = merge(cells,panel[,.(channel,Target, `Metal Tag`, `Metal Number`)],by=&quot;channel&quot;)
cells_panel = cells_panel[Target %in% c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;,&quot;T9&quot;,&quot;T10&quot;,&quot;T11&quot;,&quot;T12&quot;),]
cells_panel = cells_panel[measurement != &quot;FullStack&quot;,]
cells_panel = cells_panel[signal_type != &quot;MeanIntensityCorrectedLS&quot;,]
unique(cells_panel$signal_type)</code></pre>
<pre><code>[1] &quot;MeanIntensityCorrected&quot; &quot;MeanIntensity&quot;         </code></pre>
<pre class="r"><code>cells_panel_LScorrected = cells_panel[cells_panel$signal_type == &quot;MeanIntensityCorrectedLS&quot;,]
cells_panel_uncorrected = cells_panel[cells_panel$signal_type == &quot;MeanIntensity&quot;,]
cells_panel_corrected = cells_panel[cells_panel$signal_type == &quot;MeanIntensityCorrected&quot;,]</code></pre>
</div>
</div>
<div id="supp-figure-1e" class="section level1">
<h1>Supp Figure 1E</h1>
<div id="unspecific-amplifier-binding" class="section level2">
<h2>Unspecific Amplifier Binding</h2>
<pre class="r"><code>results = matrix(nrow=12, ncol = 12)
rownames(results) = unique(cells_panel$Target)
colnames(results) = unique(cells_panel$Target)
results = as.data.frame(results)

cells_panel_corrected = as.data.frame(cells_panel_corrected)

# calculate percentage of unspecific binding
for(i in unique(cells_panel$Target)){
  mean_target = mean(cells_panel_corrected[cells_panel_corrected[&quot;Metadata_Target&quot;] == i &amp; cells_panel_corrected[&quot;Target&quot;] == i, &quot;value&quot;])
  for(j in unique(cells_panel$Target)){
    row_index = which(rownames(results) %in% i)
    col_index = which(colnames(results) %in% j)
    results[row_index, col_index] = mean(cells_panel_corrected[cells_panel_corrected[&quot;Metadata_Target&quot;] == j &amp; cells_panel_corrected[&quot;Target&quot;] == i, &quot;value&quot;]) / mean_target * 100
  }
}

# set crosstalk to 0 from one channel to same channel
results[results == 100] &lt;- 0

a = Heatmap(as.matrix(results),
        col = colorRamp2(c(0,1,3), c(&quot;white&quot;,&quot;blue&quot;, &quot;red&quot;)),
        row_order = order(as.numeric(gsub(&quot;T&quot;, &quot;&quot;, unique(cells_panel_corrected$Target)))),
        column_order =  order(as.numeric(gsub(&quot;T&quot;, &quot;&quot;, unique(cells_panel_corrected$Target)))),
        heatmap_legend_param = list(title = &quot;% cross-talk&quot;, size =15),
        column_names_side = &quot;top&quot;,
        column_title = &quot;Stained Channel&quot;,
        row_title = &quot;Other Channels&quot;,
        row_names_side = &quot;left&quot;,
        column_names_rot = 0,
        column_names_gp = gpar(fontsize = 15),
        row_names_gp = gpar(fontsize = 15),
        column_names_centered = T,
        cell_fun = function(j, i, x, y, width, height, fill) {
        grid.text(sprintf(&quot;%.1f&quot;, as.matrix(results)[i, j]), x, y, gp = gpar(fontsize = 10))
})
draw(a)</code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1E_heatmap_unspecific_binding-1.pdf" width="672" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="supp-figure-1d" class="section level1">
<h1>Supp Figure 1D</h1>
<div id="boxplot-with-uncorrected-and-corrected-signal" class="section level2">
<h2>Boxplot with uncorrected and corrected signal</h2>
<pre class="r"><code>cells_panel2 &lt;- cells_panel
cells_panel2[cells_panel2$signal_type == &quot;MeanIntensity&quot;, ]$signal_type = &quot;uncorrected&quot;
cells_panel2[cells_panel2$signal_type == &quot;MeanIntensityCorrected&quot;, ]$signal_type = &quot;spill-over corrected&quot;

ggplot(cells_panel2[cells_panel2$Metadata_Target == &quot;T9&quot;,], aes(x=`Metal Tag`, y=counts_asinh, fill=signal_type)) +
  geom_boxplot(outlier.shape = NA, lwd=0.5) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1),
        text = element_text(size=18)) +
  guides(fill=guide_legend(title=&quot;Signal&quot;)) +
  ylab(&quot;Mean Count (asinh)&quot;) +
  xlab(&quot;&quot;) +
  facet_wrap(~Metadata_Target,labeller = as_labeller(c(T9 = &quot;mRNA Probe for Channel 9 (Er168)&quot;)))</code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1D_corrected signal-1.pdf" width="768" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="remove-previous-data" class="section level2">
<h2>Remove previous data</h2>
<pre class="r"><code>rm(list = ls())

# Reload helper functions
sapply(list.files(&quot;code/helper_functions&quot;, full.names = TRUE), source)</code></pre>
<pre><code>        code/helper_functions/calculateSummary.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/censor_dat.R
value   ?                                 
visible FALSE                             
        code/helper_functions/detect_mRNA_expression.R
value   ?                                             
visible FALSE                                         
        code/helper_functions/DistanceToClusterCenter.R
value   ?                                              
visible FALSE                                          
        code/helper_functions/findMilieu.R code/helper_functions/findPatch.R
value   ?                                  ?                                
visible FALSE                              FALSE                            
        code/helper_functions/getInfoFromString.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/getSpotnumber.R
value   ?                                    
visible FALSE                                
        code/helper_functions/plotCellCounts.R
value   ?                                     
visible FALSE                                 
        code/helper_functions/plotCellFractions.R
value   ?                                        
visible FALSE                                    
        code/helper_functions/plotDist.R
value   ?                               
visible FALSE                           
        code/helper_functions/scatter_function.R
value   ?                                       
visible FALSE                                   
        code/helper_functions/sceChecks.R
value   ?                                
visible FALSE                            
        code/helper_functions/validityChecks.R
value   ?                                     
visible FALSE                                 </code></pre>
</div>
<div id="load-the-data" class="section level2">
<h2>Load the data</h2>
<pre class="r"><code>cells1 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190208/cell.csv&quot;, header = T,sep=&quot;,&quot;)
cells3 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190215/cell.csv&quot;, header = T,sep=&quot;,&quot;)
cells4 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190222/cell.csv&quot;, header = T,sep=&quot;,&quot;)

meta1 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190208/Image.csv&quot;,header = T,sep=&quot;,&quot;)
meta3 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190215/Image.csv&quot;,header = T,sep=&quot;,&quot;)
meta4 = fread(&quot;data/data_for_analysis/12plex_validation/HeLa/20190222/Image.csv&quot;,header = T,sep=&quot;,&quot;)

meta1 = meta1[,.(ImageNumber,FileName_CellImage)]
meta3 = meta3[,.(ImageNumber,FileName_CellImage)]
meta4 = meta4[,.(ImageNumber,FileName_CellImage)]

meta1[,&#39;:=&#39; (replicate = getInfoFromString(FileName_CellImage,&quot;_&quot;,1),stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,6), 
             secondary_stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,7)),by=ImageNumber]
meta3[,&#39;:=&#39; (replicate = getInfoFromString(FileName_CellImage,&quot;_&quot;,1),stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,6), 
             secondary_stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,7)),by=ImageNumber]
meta4[,&#39;:=&#39; (replicate = getInfoFromString(FileName_CellImage,&quot;_&quot;,1),stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,7), 
             secondary_stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,8), 
             another_stain = getInfoFromString(FileName_CellImage,&quot;_&quot;,6)),by=ImageNumber]

# rename positive stain 
meta1[meta1$stain==&quot;positive&quot; &amp; meta1$secondary_stain == &quot;noAb&quot;,]$stain = &quot;positive without Ab&quot;
meta1[meta1$stain==&quot;positive&quot; &amp; meta1$secondary_stain == &quot;withAb&quot;,]$stain = &quot;positive with Ab&quot;
#meta2[meta2$stain==&quot;positive&quot; &amp; meta2$secondary_stain == &quot;noAb&quot;,]$stain = &quot;positive without Ab&quot;
#meta2[meta2$stain==&quot;positive&quot; &amp; meta2$secondary_stain == &quot;withAb&quot;,]$stain = &quot;positive with Ab&quot;
meta3[meta3$stain==&quot;positive&quot; &amp; meta3$secondary_stain == &quot;noAb&quot;,]$stain = &quot;positive without Ab&quot;
meta3[meta3$stain==&quot;positive&quot; &amp; meta3$secondary_stain == &quot;withAb&quot;,]$stain = &quot;positive with Ab&quot;
meta3[meta3$stain==&quot;negative&quot; &amp; meta3$secondary_stain == &quot;s0&quot;,]$stain = &quot;negative 1st&quot;
meta3[meta3$stain==&quot;negative&quot; &amp; meta3$secondary_stain == &quot;new&quot;,]$stain = &quot;negative&quot;
meta4[meta4$stain==&quot;noAb&quot;,]$stain = &quot;positive without Ab&quot;
meta4[meta4$stain==&quot;withAb&quot;,]$stain = &quot;positive with Ab&quot;
meta4[meta4$another_stain==&quot;negative&quot;,]$stain = &quot;negative&quot;
meta4[meta4$another_stain==&quot;T12&quot;,]$stain = &quot;T12&quot;

# change stain name
meta3[meta3$stain==&quot;T7new&quot;,]$stain = &quot;T7&quot;
meta3[meta3$stain==&quot;T5new&quot;,]$stain = &quot;T5&quot;
meta3[meta3$stain==&quot;T11new&quot;,]$stain = &quot;T11&quot;
meta3[meta3$replicate ==&quot;20190221&quot;,]$replicate = &quot;20190215&quot;

panel = fread( &quot;data/data_for_analysis/12plex_validation/HeLa/20190212_HeLa_12plex_validation.csv&quot;)

# sort Metal Number in same order than in the Image.csv file!!
panel$`Metal Number` = str_extract(panel$`Metal Tag`, &quot;[0-9]+&quot;)
panel = panel[order(panel$`Metal Number`),]
panel$channel = c(1:nrow(panel))

# melt table
cells1_long = melt.data.table(cells1,id.vars = c(&quot;ImageNumber&quot;, &quot;ObjectNumber&quot;),variable.factor = F)
cells3_long = melt.data.table(cells3,id.vars = c(&quot;ImageNumber&quot;, &quot;ObjectNumber&quot;),variable.factor = F)
cells4_long = melt.data.table(cells4,id.vars = c(&quot;ImageNumber&quot;, &quot;ObjectNumber&quot;),variable.factor = F)

# multiply value by 2E16 since it divided by this number in CellProfiler
cells1_long$value = cells1_long$value * 65535
cells3_long$value = cells3_long$value * 65535
cells4_long$value = cells4_long$value * 65535

cells1_long[,id := paste(ImageNumber,ObjectNumber,sep =&quot;_&quot;)]
cells1_long[,&#39;:=&#39; (channel = as.integer(getInfoFromFileList(variable,sep=&quot;_&quot;,strPos = 4,censorStr = &quot;c&quot;)),
                 counts_asinh = asinh(value/1)),
                 by=id]

cells3_long[,id := paste(ImageNumber,ObjectNumber,sep =&quot;_&quot;)]
cells3_long[,&#39;:=&#39; (channel = as.integer(getInfoFromFileList(variable,sep=&quot;_&quot;,strPos = 4,censorStr = &quot;c&quot;)),
                 counts_asinh = asinh(value/1)),
                 by=id]

cells4_long[,id := paste(ImageNumber,ObjectNumber,sep =&quot;_&quot;)]
cells4_long[,&#39;:=&#39; (channel = as.integer(getInfoFromFileList(variable,sep=&quot;_&quot;,strPos = 4,censorStr = &quot;c&quot;)),
                 counts_asinh = asinh(value/1)),
                 by=id]

# take only FullStack entries and not FullStackFiltered (not possible with RNA measurement, only with Ab&#39;s)
cells1_long[,measurement:=getInfoFromString(variable,&quot;_&quot;,3),by=variable]
cells3_long[,measurement:=getInfoFromString(variable,&quot;_&quot;,3),by=variable]
cells4_long[,measurement:=getInfoFromString(variable,&quot;_&quot;,3),by=variable]

cells1_long = cells1_long[measurement==&quot;FullStack&quot;,]
cells3_long = cells3_long[measurement==&quot;FullStack&quot;,]
cells4_long = cells4_long[measurement==&quot;FullStack&quot;,]</code></pre>
</div>
<div id="merge-meta-dat-with-cells-data-and-then-then-merge-all-the-files-to-have-one-file-with-all-three-replicates" class="section level2">
<h2>Merge meta dat with cells data and then then merge all the files to have one file with all three replicates</h2>
<pre class="r"><code>cells1_long = merge(cells1_long,meta1,by=&quot;ImageNumber&quot;)
cells3_long = merge(cells3_long,meta3,by=&quot;ImageNumber&quot;)
cells4_long = merge(cells4_long,meta4,by=&quot;ImageNumber&quot;)

# select only 1 negative measuremenet in 3th measurement
cells3_long = cells3_long[cells3_long$stain != &quot;negative 1st&quot;, ]

# remove additional columns which are not needed
cells1_long = cells1_long[,!(&quot;secondary_stain&quot;)]
cells3_long = cells3_long[,!(&quot;secondary_stain&quot;)]
cells4_long = cells4_long[,!(&quot;secondary_stain&quot;)]
cells4_long = cells4_long[,!(&quot;another_stain&quot;)]


cells = rbind(cells1_long, cells3_long)
cells = rbind(cells, cells4_long)</code></pre>
</div>
<div id="merge-cells-and-panel-data-and-exclude-non-relevant-channels" class="section level2">
<h2>Merge cells and panel data and exclude non-relevant channels</h2>
<pre class="r"><code># exclude certain channels, make sure to exclude the right ones!
cells_panel &lt;- merge(cells,panel[,.(channel,Target)],by=&quot;channel&quot;)
cells_panel &lt;- cells_panel[Target %in% c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;,&quot;T9&quot;,&quot;T10&quot;,&quot;T11&quot;,&quot;T12&quot;),]</code></pre>
</div>
<div id="load-data-frolm-8plex-validation-and-make-it-comparable" class="section level2">
<h2>Load data frolm 8plex validation and make it comparable</h2>
<pre class="r"><code># import data from 8plex validation
cells_panel_8plex = readRDS(file = &quot;data/data_for_analysis/12plex_validation/HeLa/2018_data_8plex_validation_DS.rds&quot;)

# make both data.frames same so they can be merged
cells_panel = cells_panel[, !(&quot;measurement&quot;)]
cells_panel_8plex = cells_panel_8plex[, !(&quot;AreaShape_Area&quot;)]

# rename the stains in the 8plex data
unique(cells_panel_8plex$stain)</code></pre>
<pre><code> [1] &quot;C1&quot;     &quot;C2&quot;     &quot;C3&quot;     &quot;C4&quot;     &quot;C5&quot;     &quot;C6&quot;     &quot;C7&quot;     &quot;C8&quot;    
 [9] &quot;all&quot;    &quot;oldMix&quot; &quot;neg&quot;   </code></pre>
<pre class="r"><code>cells_panel_8plex = cells_panel_8plex[!(stain %in% (&quot;oldMix&quot;)), ]
cells_panel_8plex[stain == &quot;C1&quot;, ]$stain = &quot;T1&quot;
cells_panel_8plex[stain == &quot;C2&quot;, ]$stain = &quot;T4&quot;
cells_panel_8plex[stain == &quot;C3&quot;, ]$stain = &quot;T3&quot;
cells_panel_8plex[stain == &quot;C4&quot;, ]$stain = &quot;T8&quot;
cells_panel_8plex[stain == &quot;C5&quot;, ]$stain = &quot;T12&quot;
cells_panel_8plex[stain == &quot;C6&quot;, ]$stain = &quot;T2&quot;
cells_panel_8plex[stain == &quot;C7&quot;, ]$stain = &quot;T9&quot;
cells_panel_8plex[stain == &quot;C8&quot;, ]$stain = &quot;T6&quot;
cells_panel_8plex[stain == &quot;all&quot;, ]$stain = &quot;positive with Ab&quot;
cells_panel_8plex[stain == &quot;neg&quot;, ]$stain = &quot;negative&quot;
cells_panel_8plex[Target == &quot;C1&quot;, ]$Target = &quot;T1&quot;
cells_panel_8plex[Target == &quot;C2&quot;, ]$Target = &quot;T4&quot;
cells_panel_8plex[Target == &quot;C3&quot;, ]$Target = &quot;T3&quot;
cells_panel_8plex[Target == &quot;C4&quot;, ]$Target = &quot;T8&quot;
cells_panel_8plex[Target == &quot;C5&quot;, ]$Target = &quot;T12&quot;
cells_panel_8plex[Target == &quot;C6&quot;, ]$Target = &quot;T2&quot;
cells_panel_8plex[Target == &quot;C7&quot;, ]$Target = &quot;T9&quot;
cells_panel_8plex[Target == &quot;C8&quot;, ]$Target = &quot;T6&quot;

# check
unique(cells_panel_8plex$stain)</code></pre>
<pre><code> [1] &quot;T1&quot;               &quot;T4&quot;               &quot;T3&quot;               &quot;T8&quot;              
 [5] &quot;T12&quot;              &quot;T2&quot;               &quot;T9&quot;               &quot;T6&quot;              
 [9] &quot;positive with Ab&quot; &quot;negative&quot;        </code></pre>
<pre class="r"><code>unique(cells_panel_8plex$Target)</code></pre>
<pre><code>[1] &quot;T1&quot;  &quot;T4&quot;  &quot;T3&quot;  &quot;T8&quot;  &quot;T12&quot; &quot;T2&quot;  &quot;T9&quot;  &quot;T6&quot; </code></pre>
<pre class="r"><code>all(colnames(cells_panel) == colnames(cells_panel_8plex))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code># merge both files
cells_panel_12plex = rbind(cells_panel, cells_panel_8plex)</code></pre>
</div>
</div>
<div id="supp-figure-1a" class="section level1">
<h1>Supp Figure 1A</h1>
<div id="throw-out-the-dna-histone-and-pancytokeratin-data-and-then-get-the-info-of-ppib-staining" class="section level2">
<h2>Throw out the DNA, histone and panCytokeratin data and then get the info of PPIB staining</h2>
<pre class="r"><code># subset
cells_panel_12plex_sub = cells_panel_12plex[stain == Target,]

# relevel factor
cells_panel_12plex_sub$Target &lt;- factor(cells_panel_12plex_sub$Target, levels = c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;,&quot;T9&quot;,&quot;T10&quot;,&quot;T11&quot;,&quot;T12&quot;))

# check out difference in signal intensity for PPIB in the different channels
ggplot(data = cells_panel_12plex_sub[Target %in%c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;,&quot;T9&quot;,&quot;T10&quot;,&quot;T11&quot;,&quot;T12&quot;),], aes(x=as.factor(Target),y=counts_asinh)) +
  geom_boxplot(fill=&quot;deepskyblue1&quot;)+
  theme_minimal()+
  theme(text = element_text(size=25), axis.text.x = element_text(angle = 90, hjust = 1)) +
  ylab(&quot;Mean Intensity Count per Cell [asinh]&quot;) + 
  xlab(&quot;Target Channels&quot;) + 
  geom_hline(yintercept = mean(cells_panel_12plex_sub$counts_asinh), linetype = 2) +
  scale_x_discrete(breaks=c(&quot;T1&quot;, &quot;T2&quot;, &quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;, &quot;T9&quot;, &quot;T10&quot;,&quot;T11&quot;, &quot;T12&quot;),
                      labels= c(&quot;T1 PPIB&quot;,&quot;T2 PPIB&quot;,&quot;T3 PPIB&quot;, &quot;T4 PPIB&quot;, &quot;T5 PPIB&quot;,&quot;T6 PPIB&quot;,&quot;T7 PPIB&quot;,&quot;T8 PPIB&quot;,&quot;T9 PPIB&quot;,&quot;T10 PPIB&quot;, &quot;T11 PPIB&quot;, &quot;T12 PPIB&quot;)) </code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1A_merged_comparison_per_channel-1.pdf" width="960" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="signal-range" class="section level2">
<h2>Signal range</h2>
<pre class="r"><code>cells_panel_12plex_sub %&gt;%
  group_by(stain) %&gt;%
  summarize(mean_per_channel = mean(counts_asinh)) %&gt;%
  mutate(sd = sd(mean_per_channel), mean = mean(mean_per_channel), min = min(mean_per_channel), max = max(mean_per_channel))</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre><code># A tibble: 12 x 6
   stain mean_per_channel    sd  mean   min   max
   &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
 1 T1                1.58 0.195  1.48  1.14  1.77
 2 T10               1.44 0.195  1.48  1.14  1.77
 3 T11               1.14 0.195  1.48  1.14  1.77
 4 T12               1.68 0.195  1.48  1.14  1.77
 5 T2                1.68 0.195  1.48  1.14  1.77
 6 T3                1.38 0.195  1.48  1.14  1.77
 7 T4                1.64 0.195  1.48  1.14  1.77
 8 T5                1.21 0.195  1.48  1.14  1.77
 9 T6                1.37 0.195  1.48  1.14  1.77
10 T7                1.77 0.195  1.48  1.14  1.77
11 T8                1.48 0.195  1.48  1.14  1.77
12 T9                1.39 0.195  1.48  1.14  1.77</code></pre>
</div>
</div>
<div id="supp-figure-1c" class="section level1">
<h1>Supp Figure 1C</h1>
<div id="influence-of-antibody-incubation" class="section level2">
<h2>Influence of Antibody Incubation</h2>
<pre class="r"><code>dd = cells_panel[stain == &quot;positive without Ab&quot; | stain == &quot;positive with Ab&quot;,]

# relevel factor
dd$Target &lt;- factor(dd$Target, levels = c(&quot;T1&quot;,&quot;T2&quot;,&quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;,&quot;T9&quot;,&quot;T10&quot;,&quot;T11&quot;,&quot;T12&quot;))
dd$stain &lt;- ifelse(dd$stain == &quot;positive with Ab&quot;, &quot;+ Antibodies&quot;, &quot;- Antibodies&quot;)

# plot
ggplot(data=dd, aes(x=Target, y=counts_asinh, fill = stain)) + 
  geom_boxplot() +
  scale_x_discrete(breaks=c(&quot;T1&quot;, &quot;T2&quot;, &quot;T3&quot;,&quot;T4&quot;,&quot;T5&quot;,&quot;T6&quot;,&quot;T7&quot;,&quot;T8&quot;, &quot;T9&quot;, &quot;T11&quot;,&quot;T10&quot;, &quot;T12&quot;),
                      labels= c(&quot;T1 POLR2A&quot;,&quot;T2 PPIB&quot;,&quot;T3 UBC&quot;, &quot;T4 HPRT1&quot;, &quot;T5 TUB&quot;,&quot;T6 RPL28&quot;,&quot;T7 RPL5&quot;,&quot;T8 B2M&quot;,&quot;T9 ACTB&quot;,&quot;T10 LDHA&quot;, &quot;T11 RPLP0&quot;, &quot;T12 GAPDH&quot;)) +
  theme_minimal()+
  theme(text = element_text(size=25), axis.text.x = element_text(angle = 90, hjust = 1)) + 
  ylab(&quot;Mean Intensity Count per Cell [asinh]&quot;) +
  xlab(&quot;Target Channels&quot;) + 
  scale_fill_discrete(name = &quot;Protocol Type&quot;) </code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1C_overnight_Ab_incubation-1.pdf" width="960" style="display: block; margin: auto;" type="application/pdf" />
</div>
</div>
<div id="supp-figure-1b" class="section level1">
<h1>Supp Figure 1B</h1>
<div id="compare-positive-controls-with-rnaseq-data" class="section level2">
<h2>Compare Positive Controls with RNAseq data</h2>
<pre class="r"><code># add gene name
cells_panel_12plex$gene &lt;- &quot;&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T1&quot;,]$gene &lt;- &quot;POLR2A&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T2&quot;,]$gene &lt;- &quot;PPIB&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T3&quot;,]$gene &lt;- &quot;UBC&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T4&quot;,]$gene &lt;- &quot;HPRT1&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T5&quot;,]$gene &lt;- &quot;TUB&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T6&quot;,]$gene &lt;- &quot;RPL28&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T7&quot;,]$gene &lt;- &quot;RPL5&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T8&quot;,]$gene &lt;- &quot;B2M&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T9&quot;,]$gene &lt;- &quot;ACTB&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T10&quot;,]$gene &lt;- &quot;LDHA&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T11&quot;,]$gene &lt;- &quot;RPLP0&quot;
cells_panel_12plex[cells_panel_12plex$Target == &quot;T12&quot;,]$gene &lt;- &quot;GAPDH&quot;

median_per_gene &lt;- cells_panel_12plex %&gt;%
  filter(stain == &quot;positive with Ab&quot;) %&gt;%
  filter(replicate %in% c(&quot;20190208&quot;, &quot;20190215&quot;, &quot;20190222&quot;)) %&gt;%
  group_by(gene) %&gt;%
  summarise(expression_RNAScope = median(counts_asinh))</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code># data from human protein atlas
rna_seq &lt;- data.frame(median_per_gene$gene)
colnames(rna_seq) &lt;- &quot;gene&quot;

# Normalized Expression Data from Human Protein Atlas 
rna_seq$expression_RNAseq &lt;- c(145.3, 89.2, 217.1, 21.3, 74.9, 1.2, 54.7, 87.9, 83.6, 72.4, 23.9, 62.9)

# merge
sum &lt;- left_join(median_per_gene, rna_seq)</code></pre>
<pre><code>Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code># plot 
ggplot(sum, aes(x=log2(expression_RNAseq), y=expression_RNAScope, label=gene)) + 
  geom_label_repel(size=7, fill=&quot;deepskyblue1&quot;, alpha=1,
                   min.segment.length = unit(0, &#39;lines&#39;),
                   force_pull = 0) +
  geom_point(size=3) +
  geom_smooth(method = &quot;lm&quot;) +
  xlab(&quot;Normalized Expression RNA-Seq [log2]&quot;) +
  ylab(&quot;Median Expression RNAScope [asinh]&quot;) + 
  stat_cor(method = &quot;pearson&quot;,
           aes(label = paste0(&quot;atop(&quot;, ..r.label..,  &quot;,&quot;, ..rr.label.. ,&quot;)&quot;)),
           size = 10, cor.coef.name = &quot;R&quot;, label.sep=&quot;\n&quot;, label.y.npc = &quot;top&quot;) + 
  theme_minimal() + 
  theme(text = element_text(size=20)) </code></pre>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1B_correlation_with_RNAseq-1.pdf" width="768" style="display: block; margin: auto;" type="application/pdf" />
</div>
<div id="correlation" class="section level2">
<h2>Correlation</h2>
<pre class="r"><code># correlation of the two technologies
cor(sum$expression_RNAScope, sum$expression_RNAseq, method = c(&quot;spearman&quot;))</code></pre>
<pre><code>[1] 0.8391608</code></pre>
</div>
</div>
<div id="supp-figure-1f" class="section level1">
<h1>Supp Figure 1F</h1>
<div id="load-data" class="section level2">
<h2>Load Data</h2>
<pre class="r"><code>sce_rna &lt;- readRDS(&quot;data/data_for_analysis/sce_RNA.rds&quot;)</code></pre>
</div>
<div id="detection-of-chemokine-expressing-cells" class="section level2">
<h2>Detection of chemokine expressing cells</h2>
<p>for the detection of chemokine expressing cells we make use of the fact that we also measured a negative control (DapB).</p>
<pre class="r"><code># get the names of the chemokine channels without the negative control channel
chemokine_channels = rownames(sce_rna[which(grepl(&quot;T\\d+_&quot;,rownames(sce_rna)) &amp; ! grepl(&quot;DapB&quot;,rownames(sce_rna))),])
chemokine_channels_sub &lt;- c(&quot;T2_CCL22&quot;)

# run function to define chemokine expressing cells 
output_list &lt;- compute_difference(sce_rna, 
                          cellID = &quot;cellID&quot;, 
                          assay_name = &quot;asinh&quot;, 
                          threshold = 0.01, 
                          mRNA_channels = chemokine_channels_sub, 
                          negative_control = &quot;T6_DapB&quot;, 
                          return_calc_metrics = TRUE)</code></pre>
</div>
<div id="plot-results-from-chemokine-detection" class="section level2">
<h2>Plot Results from Chemokine Detection</h2>
<pre class="r"><code># check difference between DapB and signal (histogram)
plot_list = list()
for(i in chemokine_channels_sub){
  
  # subset whole data set for visualization purposes
  diff_chemo &lt;- output_list[[i]]
  diff_chemo_sub &lt;- diff_chemo[sample(nrow(diff_chemo), nrow(diff_chemo)*0.5), ]

  a &lt;- ggplot(data = diff_chemo_sub, aes(x=scaled_diff)) + 
  geom_histogram(binwidth = 0.05, aes(fill = 
                                       ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
  xlab(paste(paste(&quot;Scaled Difference &quot;, i, sep = &quot; &quot;), &quot; - DapB&quot;, sep = &quot;&quot;)) + 
    xlim(-5,7) +
    theme_minimal() + 
    theme(text = element_text(size=20),
          legend.position = &quot;none&quot;) + 
    scale_fill_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;))
  
  # significant cells defined by subtraction
  b &lt;- ggplot(data=diff_chemo_sub, aes(x=mean_negative_control, y=mean_chemokine)) + 
    geom_point_rast(alpha=0.2, aes(col = 
                                ifelse(padj &lt;= 0.01 &amp; scaled_diff &gt; 0, &#39;p&lt;0.01&#39;, &#39;n.s.&#39;))) + 
    scale_color_manual(values = c(&quot;black&quot;, &quot;deepskyblue1&quot;), 
                       name = &quot;Legend&quot;) +
    guides(color = guide_legend(override.aes = list(alpha=1, size=3))) +
    xlim(0,5.5) + ylim(0,5.5) +
    ylab(paste(&quot;Mean expression of&quot;, i, sep=&quot; &quot;)) +
    xlab(&quot;Mean DapB mRNA expression&quot;) +
    theme_minimal() + 
    theme(text = element_text(size=20)) 

  
  grid.arrange(a,b, nrow = 1, ncol=2)
}</code></pre>
<pre><code>Warning: Removed 330 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<pre><code>Warning: Removed 2 rows containing missing values (geom_point).</code></pre>
<embed src="figure/Supp-Figure_1.rmd/Supp_Fig_1F_chemokine_detection-1.pdf" width="425" style="display: block; margin: auto;" type="application/pdf" />
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04 LTS

Matrix products: default
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] ggrastr_0.2.1               ggpmisc_0.3.7              
 [3] ggrepel_0.9.0               ggbeeswarm_0.6.0           
 [5] gridExtra_2.3               circlize_0.4.12            
 [7] ComplexHeatmap_2.4.3        ggpubr_0.4.0               
 [9] stringr_1.4.0               dplyr_1.0.2                
[11] ggplot2_3.3.3               data.table_1.13.6          
[13] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
[15] Biobase_2.50.0              GenomicRanges_1.42.0       
[17] GenomeInfoDb_1.26.2         IRanges_2.24.1             
[19] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[21] MatrixGenerics_1.2.0        matrixStats_0.57.0         
[23] workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] nlme_3.1-151           bitops_1.0-6           fs_1.5.0              
 [4] RColorBrewer_1.1-2     rprojroot_2.0.2        tools_4.0.3           
 [7] backports_1.2.1        utf8_1.1.4             R6_2.5.0              
[10] vipor_0.4.5            mgcv_1.8-33            colorspace_2.0-0      
[13] GetoptLong_1.0.5       withr_2.3.0            tidyselect_1.1.0      
[16] curl_4.3               compiler_4.0.3         git2r_0.28.0          
[19] cli_2.2.0              Cairo_1.5-12.2         DelayedArray_0.16.0   
[22] labeling_0.4.2         scales_1.1.1           digest_0.6.27         
[25] foreign_0.8-81         rmarkdown_2.6          rio_0.5.16            
[28] XVector_0.30.0         pkgconfig_2.0.3        htmltools_0.5.1.1     
[31] rlang_0.4.10           GlobalOptions_0.1.2    readxl_1.3.1          
[34] rstudioapi_0.13        farver_2.0.3           shape_1.4.5           
[37] generics_0.1.0         zip_2.1.1              car_3.0-10            
[40] RCurl_1.98-1.2         magrittr_2.0.1         GenomeInfoDbData_1.2.4
[43] Matrix_1.3-2           fansi_0.4.1            Rcpp_1.0.5            
[46] munsell_0.5.0          abind_1.4-5            lifecycle_0.2.0       
[49] stringi_1.5.3          whisker_0.4            yaml_2.2.1            
[52] carData_3.0-4          zlibbioc_1.36.0        promises_1.1.1        
[55] forcats_0.5.0          crayon_1.3.4           lattice_0.20-41       
[58] splines_4.0.3          haven_2.3.1            hms_0.5.3             
[61] knitr_1.30             pillar_1.4.7           rjson_0.2.20          
[64] ggsignif_0.6.0         glue_1.4.2             evaluate_0.14         
[67] vctrs_0.3.6            png_0.1-7              httpuv_1.5.4          
[70] cellranger_1.1.0       gtable_0.3.0           purrr_0.3.4           
[73] tidyr_1.1.2            assertthat_0.2.1       clue_0.3-58           
[76] xfun_0.20              openxlsx_4.2.3         broom_0.7.3           
[79] rstatix_0.6.0          later_1.1.0.1          tibble_3.0.4          
[82] beeswarm_0.2.3         cluster_2.1.0          ellipsis_0.3.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
